// @aws-sdk/client-eks/dist-es/schemas/schemas_0.js
const _A = "Addon";
const _AAP = "AssociatedAccessPolicy";
const _AAPL = "AssociatedAccessPoliciesList";
const _AAPR = "AssociateAccessPolicyRequest";
const _AAPRs = "AssociateAccessPolicyResponse";
const _AAPs = "AssociateAccessPolicy";
const _ACD = "AddonCompatibilityDetail";
const _ACDd = "AddonCompatibilityDetails";
const _ACR = "AccessConfigResponse";
const _ADE = "AccessDeniedException";
const _AE = "AccessEntry";
const _AEC = "AssociateEncryptionConfig";
const _AECR = "AssociateEncryptionConfigRequest";
const _AECRs = "AssociateEncryptionConfigResponse";
const _AH = "AddonHealth";
const _AI = "AddonInfo";
const _AIL = "AddonIssueList";
const _AIPC = "AssociateIdentityProviderConfig";
const _AIPCR = "AssociateIdentityProviderConfigRequest";
const _AIPCRs = "AssociateIdentityProviderConfigResponse";
const _AId = "AddonIssue";
const _ANCR = "AddonNamespaceConfigRequest";
const _ANCRd = "AddonNamespaceConfigResponse";
const _AP = "AccessPolicy";
const _APIA = "AddonPodIdentityAssociations";
const _APIAL = "AddonPodIdentityAssociationsList";
const _APIC = "AddonPodIdentityConfiguration";
const _APICL = "AddonPodIdentityConfigurationList";
const _APL = "AccessPoliciesList";
const _AS = "AccessScope";
const _ASG = "AutoScalingGroup";
const _ASGL = "AutoScalingGroupList";
const _AVI = "AddonVersionInfo";
const _AVIL = "AddonVersionInfoList";
const _Ad = "Addons";
const _BRE = "BadRequestException";
const _BS = "BlockStorage";
const _C = "Certificate";
const _CA = "CreateAddon";
const _CACR = "CreateAccessConfigRequest";
const _CAE = "CreateAccessEntry";
const _CAER = "CreateAccessEntryRequest";
const _CAERr = "CreateAccessEntryResponse";
const _CAR = "CreateAddonRequest";
const _CARr = "CreateAddonResponse";
const _CC = "CreateCluster";
const _CCR = "ComputeConfigRequest";
const _CCRo = "ComputeConfigResponse";
const _CCRon = "ConnectorConfigRequest";
const _CCRonn = "ConnectorConfigResponse";
const _CCRr = "CreateClusterRequest";
const _CCRre = "CreateClusterResponse";
const _CE = "ClientException";
const _CEAS = "CreateEksAnywhereSubscription";
const _CEASR = "CreateEksAnywhereSubscriptionRequest";
const _CEASRr = "CreateEksAnywhereSubscriptionResponse";
const _CFP = "CreateFargateProfile";
const _CFPR = "CreateFargateProfileRequest";
const _CFPRr = "CreateFargateProfileResponse";
const _CH = "ClusterHealth";
const _CI = "ClusterIssue";
const _CIL = "ClusterIssueList";
const _CN = "CreateNodegroup";
const _CNR = "CreateNodegroupRequest";
const _CNRr = "CreateNodegroupResponse";
const _CPIA = "CreatePodIdentityAssociation";
const _CPIAR = "CreatePodIdentityAssociationRequest";
const _CPIARr = "CreatePodIdentityAssociationResponse";
const _CPPR = "ControlPlanePlacementRequest";
const _CPPRo = "ControlPlanePlacementResponse";
const _CS = "ClientStat";
const _CSl = "ClientStats";
const _CVI = "ClusterVersionInformation";
const _CVL = "ClusterVersionList";
const _Cl = "Cluster";
const _Co = "Compatibility";
const _Com = "Compatibilities";
const _DA = "DeleteAddon";
const _DAC = "DescribeAddonConfiguration";
const _DACR = "DescribeAddonConfigurationRequest";
const _DACRe = "DescribeAddonConfigurationResponse";
const _DAE = "DeleteAccessEntry";
const _DAER = "DeleteAccessEntryRequest";
const _DAERe = "DeleteAccessEntryResponse";
const _DAERes = "DescribeAccessEntryRequest";
const _DAEResc = "DescribeAccessEntryResponse";
const _DAEe = "DescribeAccessEntry";
const _DAP = "DisassociateAccessPolicy";
const _DAPR = "DisassociateAccessPolicyRequest";
const _DAPRi = "DisassociateAccessPolicyResponse";
const _DAR = "DeleteAddonRequest";
const _DARe = "DeleteAddonResponse";
const _DARes = "DescribeAddonRequest";
const _DAResc = "DescribeAddonResponse";
const _DAV = "DescribeAddonVersions";
const _DAVR = "DescribeAddonVersionsRequest";
const _DAVRe = "DescribeAddonVersionsResponse";
const _DAe = "DescribeAddon";
const _DC = "DeleteCluster";
const _DCR = "DeleteClusterRequest";
const _DCRe = "DeleteClusterResponse";
const _DCRer = "DeregisterClusterRequest";
const _DCRere = "DeregisterClusterResponse";
const _DCRes = "DescribeClusterRequest";
const _DCResc = "DescribeClusterResponse";
const _DCV = "DescribeClusterVersions";
const _DCVR = "DescribeClusterVersionsRequest";
const _DCVRe = "DescribeClusterVersionsResponse";
const _DCe = "DeregisterCluster";
const _DCes = "DescribeCluster";
const _DD = "DeprecationDetail";
const _DDe = "DeprecationDetails";
const _DEAS = "DeleteEksAnywhereSubscription";
const _DEASR = "DeleteEksAnywhereSubscriptionRequest";
const _DEASRe = "DeleteEksAnywhereSubscriptionResponse";
const _DEASRes = "DescribeEksAnywhereSubscriptionRequest";
const _DEASResc = "DescribeEksAnywhereSubscriptionResponse";
const _DEASe = "DescribeEksAnywhereSubscription";
const _DFP = "DeleteFargateProfile";
const _DFPR = "DeleteFargateProfileRequest";
const _DFPRe = "DeleteFargateProfileResponse";
const _DFPRes = "DescribeFargateProfileRequest";
const _DFPResc = "DescribeFargateProfileResponse";
const _DFPe = "DescribeFargateProfile";
const _DI = "DescribeInsight";
const _DIPC = "DescribeIdentityProviderConfig";
const _DIPCR = "DescribeIdentityProviderConfigRequest";
const _DIPCRe = "DescribeIdentityProviderConfigResponse";
const _DIPCRi = "DisassociateIdentityProviderConfigRequest";
const _DIPCRis = "DisassociateIdentityProviderConfigResponse";
const _DIPCi = "DisassociateIdentityProviderConfig";
const _DIR = "DescribeInsightRequest";
const _DIRR = "DescribeInsightsRefreshRequest";
const _DIRRe = "DescribeInsightsRefreshResponse";
const _DIRe = "DescribeInsightResponse";
const _DIRes = "DescribeInsightsRefresh";
const _DN = "DeleteNodegroup";
const _DNR = "DeleteNodegroupRequest";
const _DNRe = "DeleteNodegroupResponse";
const _DNRes = "DescribeNodegroupRequest";
const _DNResc = "DescribeNodegroupResponse";
const _DNe = "DescribeNodegroup";
const _DPIA = "DeletePodIdentityAssociation";
const _DPIAR = "DeletePodIdentityAssociationRequest";
const _DPIARe = "DeletePodIdentityAssociationResponse";
const _DPIARes = "DescribePodIdentityAssociationRequest";
const _DPIAResc = "DescribePodIdentityAssociationResponse";
const _DPIAe = "DescribePodIdentityAssociation";
const _DU = "DescribeUpdate";
const _DUR = "DescribeUpdateRequest";
const _DURe = "DescribeUpdateResponse";
const _EAS = "EksAnywhereSubscription";
const _EASL = "EksAnywhereSubscriptionList";
const _EAST = "EksAnywhereSubscriptionTerm";
const _EC = "EncryptionConfig";
const _ECL = "EncryptionConfigList";
const _ED = "ErrorDetail";
const _EDr = "ErrorDetails";
const _ELB = "ElasticLoadBalancing";
const _FP = "FargateProfile";
const _FPH = "FargateProfileHealth";
const _FPI = "FargateProfileIssue";
const _FPIL = "FargateProfileIssueList";
const _FPS = "FargateProfileSelector";
const _FPSa = "FargateProfileSelectors";
const _I = "Identity";
const _ICSS = "InsightCategorySpecificSummary";
const _IF = "InsightsFilter";
const _IL = "IssueList";
const _IPC = "IdentityProviderConfig";
const _IPCR = "IdentityProviderConfigResponse";
const _IPCd = "IdentityProviderConfigs";
const _IPE = "InvalidParameterException";
const _IRD = "InsightResourceDetail";
const _IRDn = "InsightResourceDetails";
const _IRE = "InvalidRequestException";
const _IS = "InsightStatus";
const _ISE = "InvalidStateException";
const _ISn = "InsightSummary";
const _ISns = "InsightSummaries";
const _In = "Insight";
const _Is = "Issue";
const _KNCR = "KubernetesNetworkConfigRequest";
const _KNCRu = "KubernetesNetworkConfigResponse";
const _L = "License";
const _LA = "ListAddons";
const _LAAP = "ListAssociatedAccessPolicies";
const _LAAPR = "ListAssociatedAccessPoliciesRequest";
const _LAAPRi = "ListAssociatedAccessPoliciesResponse";
const _LAE = "ListAccessEntries";
const _LAER = "ListAccessEntriesRequest";
const _LAERi = "ListAccessEntriesResponse";
const _LAP = "ListAccessPolicies";
const _LAPR = "ListAccessPoliciesRequest";
const _LAPRi = "ListAccessPoliciesResponse";
const _LAR = "ListAddonsRequest";
const _LARi = "ListAddonsResponse";
const _LC = "ListClusters";
const _LCR = "ListClustersRequest";
const _LCRi = "ListClustersResponse";
const _LEAS = "ListEksAnywhereSubscriptions";
const _LEASR = "ListEksAnywhereSubscriptionsRequest";
const _LEASRi = "ListEksAnywhereSubscriptionsResponse";
const _LFP = "ListFargateProfiles";
const _LFPR = "ListFargateProfilesRequest";
const _LFPRi = "ListFargateProfilesResponse";
const _LI = "ListInsights";
const _LIPC = "ListIdentityProviderConfigs";
const _LIPCR = "ListIdentityProviderConfigsRequest";
const _LIPCRi = "ListIdentityProviderConfigsResponse";
const _LIR = "ListInsightsRequest";
const _LIRi = "ListInsightsResponse";
const _LL = "LicenseList";
const _LN = "ListNodegroups";
const _LNR = "ListNodegroupsRequest";
const _LNRi = "ListNodegroupsResponse";
const _LPIA = "ListPodIdentityAssociations";
const _LPIAR = "ListPodIdentityAssociationsRequest";
const _LPIARi = "ListPodIdentityAssociationsResponse";
const _LS = "LogSetup";
const _LSo = "LogSetups";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _LTS = "LaunchTemplateSpecification";
const _LU = "ListUpdates";
const _LUR = "ListUpdatesRequest";
const _LURi = "ListUpdatesResponse";
const _Lo = "Logging";
const _MI = "MarketplaceInformation";
const _N = "Nodegroup";
const _NFE = "NotFoundException";
const _NH = "NodegroupHealth";
const _NR = "NodegroupResources";
const _NRC = "NodeRepairConfig";
const _NRCO = "NodeRepairConfigOverrides";
const _NRCOL = "NodeRepairConfigOverridesList";
const _NSC = "NodegroupScalingConfig";
const _NUC = "NodegroupUpdateConfig";
const _OCR = "OutpostConfigRequest";
const _OCRu = "OutpostConfigResponse";
const _OIDC = "OIDC";
const _OIPC = "OidcIdentityProviderConfig";
const _OIPCR = "OidcIdentityProviderConfigRequest";
const _P = "Provider";
const _PIA = "PodIdentityAssociation";
const _PIAS = "PodIdentityAssociationSummary";
const _PIASo = "PodIdentityAssociationSummaries";
const _RAC = "RemoteAccessConfig";
const _RC = "RegisterCluster";
const _RCR = "RegisterClusterRequest";
const _RCRe = "RegisterClusterResponse";
const _RIUE = "ResourceInUseException";
const _RLEE = "ResourceLimitExceededException";
const _RNCR = "RemoteNetworkConfigRequest";
const _RNCRe = "RemoteNetworkConfigResponse";
const _RNFE = "ResourceNotFoundException";
const _RNN = "RemoteNodeNetwork";
const _RNNL = "RemoteNodeNetworkList";
const _RPDE = "ResourcePropagationDelayException";
const _RPN = "RemotePodNetwork";
const _RPNL = "RemotePodNetworkList";
const _SCR = "StorageConfigRequest";
const _SCRt = "StorageConfigResponse";
const _SE = "ServerException";
const _SIR = "StartInsightsRefresh";
const _SIRR = "StartInsightsRefreshRequest";
const _SIRRt = "StartInsightsRefreshResponse";
const _SUE = "ServiceUnavailableException";
const _T = "Taint";
const _TE = "ThrottlingException";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _U = "Update";
const _UA = "UpdateAddon";
const _UACR = "UpdateAccessConfigRequest";
const _UAE = "UpdateAccessEntry";
const _UAER = "UpdateAccessEntryRequest";
const _UAERp = "UpdateAccessEntryResponse";
const _UAR = "UpdateAddonRequest";
const _UARp = "UpdateAddonResponse";
const _UAZE = "UnsupportedAvailabilityZoneException";
const _UCC = "UpdateClusterConfig";
const _UCCR = "UpdateClusterConfigRequest";
const _UCCRp = "UpdateClusterConfigResponse";
const _UCV = "UpdateClusterVersion";
const _UCVR = "UpdateClusterVersionRequest";
const _UCVRp = "UpdateClusterVersionResponse";
const _UEAS = "UpdateEksAnywhereSubscription";
const _UEASR = "UpdateEksAnywhereSubscriptionRequest";
const _UEASRp = "UpdateEksAnywhereSubscriptionResponse";
const _ULP = "UpdateLabelsPayload";
const _UNC = "UpdateNodegroupConfig";
const _UNCR = "UpdateNodegroupConfigRequest";
const _UNCRp = "UpdateNodegroupConfigResponse";
const _UNV = "UpdateNodegroupVersion";
const _UNVR = "UpdateNodegroupVersionRequest";
const _UNVRp = "UpdateNodegroupVersionResponse";
const _UP = "UpdateParam";
const _UPIA = "UpdatePodIdentityAssociation";
const _UPIAR = "UpdatePodIdentityAssociationRequest";
const _UPIARp = "UpdatePodIdentityAssociationResponse";
const _UPR = "UpgradePolicyRequest";
const _UPRp = "UpgradePolicyResponse";
const _UPp = "UpdateParams";
const _UR = "UntagResource";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _UTP = "UpdateTaintsPayload";
const _VCR = "VpcConfigRequest";
const _VCRp = "VpcConfigResponse";
const _ZSCR = "ZonalShiftConfigRequest";
const _ZSCRo = "ZonalShiftConfigResponse";
const _a = "arn";
const _aA = "addonArn";
const _aAP = "associatedAccessPolicy";
const _aAPs = "associatedAccessPolicies";
const _aAs = "associatedAt";
const _aAss = "associationArn";
const _aC = "accessConfig";
const _aCD = "addonCompatibilityDetails";
const _aCc = "activationCode";
const _aE = "activationExpiry";
const _aEA = "accessEntryArn";
const _aEc = "accessEntry";
const _aEcc = "accessEntries";
const _aI = "activationId";
const _aId = "additionalInfo";
const _aIs = "associationId";
const _aM = "authenticationMode";
const _aN = "addonName";
const _aOUL = "addOrUpdateLabels";
const _aOUT = "addOrUpdateTaints";
const _aP = "accessPolicies";
const _aPA = "associatedPolicyArn";
const _aR = "autoRenew";
const _aS = "accessScope";
const _aSG = "autoScalingGroups";
const _aT = "amiType";
const _aV = "addonVersion";
const _aVd = "addonVersions";
const _ad = "addon";
const _add = "addons";
const _ar = "architecture";
const _as = "association";
const _ass = "associations";
const _bCCAP = "bootstrapClusterCreatorAdminPermissions";
const _bS = "blockStorage";
const _bSMA = "bootstrapSelfManagedAddons";
const _c = "client";
const _cA = "createdAt";
const _cAe = "certificateAuthority";
const _cC = "connectorConfig";
const _cCo = "computeConfig";
const _cI = "clientId";
const _cL = "clusterLogging";
const _cN = "clusterName";
const _cPIT = "controlPlaneInstanceType";
const _cPP = "controlPlanePlacement";
const _cRT = "clientRequestToken";
const _cS = "clientStats";
const _cSGI = "clusterSecurityGroupId";
const _cSS = "categorySpecificSummary";
const _cSo = "configurationSchema";
const _cT = "computeTypes";
const _cTa = "capacityType";
const _cTl = "clusterType";
const _cV = "configurationValues";
const _cVl = "clusterVersion";
const _cVlu = "clusterVersions";
const _cVo = "compatibleVersions";
const _ca = "category";
const _cat = "categories";
const _ci = "cidrs";
const _cl = "cluster";
const _clu = "clusters";
const _co = "code";
const _com = "compatibilities";
const _d = "data";
const _dD = "deprecationDetails";
const _dN = "defaultNamespace";
const _dO = "defaultOnly";
const _dP = "deletionProtection";
const _dPV = "defaultPlatformVersion";
const _dS = "diskSize";
const _dST = "disableSessionTags";
const _dSe = "desiredSize";
const _dV = "defaultVersion";
const _de = "description";
const _du = "duration";
const _e = "error";
const _eA = "endedAt";
const _eC = "encryptionConfig";
const _eCr = "errorCode";
const _eD = "effectiveDate";
const _eDx = "expirationDate";
const _eI = "externalId";
const _eLB = "elasticLoadBalancing";
const _eM = "errorMessage";
const _eOESD = "endOfExtendedSupportDate";
const _eOSSD = "endOfStandardSupportDate";
const _ePA = "endpointPublicAccess";
const _ePAn = "endpointPrivateAccess";
const _eSK = "ec2SshKey";
const _ef = "effect";
const _en = "enabled";
const _end = "endpoint";
const _er = "errors";
const _f = "filter";
const _fP = "fargateProfile";
const _fPA = "fargateProfileArn";
const _fPN = "fargateProfileName";
const _fPNa = "fargateProfileNames";
const _fo = "force";
const _gC = "groupsClaim";
const _gN = "groupName";
const _gP = "groupsPrefix";
const _h = "health";
const _hE = "httpError";
const _hQ = "httpQuery";
const _ht = "http";
const _i = "issues";
const _iA = "includeAll";
const _iF = "ipFamily";
const _iPC = "identityProviderConfig";
const _iPCA = "identityProviderConfigArn";
const _iPCN = "identityProviderConfigName";
const _iPCd = "identityProviderConfigs";
const _iS = "insightStatus";
const _iSn = "includeStatus";
const _iT = "instanceTypes";
const _iU = "issuerUrl";
const _id = "identity";
const _id_ = "id";
const _in = "insight";
const _inc = "include";
const _ins = "insights";
const _is = "issuer";
const _k = "key";
const _kA = "keyArn";
const _kG = "kubernetesGroups";
const _kNC = "kubernetesNetworkConfig";
const _kPV = "kubernetesPatchVersion";
const _kRU = "kubernetesResourceUri";
const _kV = "kubernetesVersion";
const _kVu = "kubernetesVersions";
const _l = "logging";
const _lA = "licenseArns";
const _lQ = "licenseQuantity";
const _lRT = "lastRequestTime";
const _lRTa = "lastRefreshTime";
const _lT = "licenseType";
const _lTT = "lastTransitionTime";
const _lTa = "launchTemplate";
const _la = "labels";
const _li = "licenses";
const _m = "message";
const _mA = "modifiedAt";
const _mI = "marketplaceInformation";
const _mPNRC = "maxParallelNodesRepairedCount";
const _mPNRP = "maxParallelNodesRepairedPercentage";
const _mR = "maxResults";
const _mRWTM = "minRepairWaitTimeMins";
const _mS = "minSize";
const _mSa = "maxSize";
const _mU = "maxUnavailable";
const _mUNTC = "maxUnhealthyNodeThresholdCount";
const _mUNTP = "maxUnhealthyNodeThresholdPercentage";
const _mUP = "maxUnavailablePercentage";
const _n = "name";
const _nA = "nodegroupArn";
const _nC = "namespaceConfig";
const _nMC = "nodeMonitoringCondition";
const _nN = "nodegroupName";
const _nORLD = "numberOfRequestsLast30Days";
const _nP = "nodePools";
const _nR = "nodeRole";
const _nRA = "nodeRoleArn";
const _nRC = "nodeRepairConfig";
const _nRCO = "nodeRepairConfigOverrides";
const _nT = "nextToken";
const _nUR = "nodeUnhealthyReason";
const _na = "namespaces";
const _nam = "namespace";
const _no = "nodegroup";
const _nod = "nodegroups";
const _o = "owner";
const _oA = "outpostArns";
const _oAw = "ownerArn";
const _oC = "outpostConfig";
const _oi = "oidc";
const _ow = "owners";
const _p = "publisher";
const _pA = "principalArn";
const _pAC = "publicAccessCidrs";
const _pAo = "policyArn";
const _pERA = "podExecutionRoleArn";
const _pI = "productId";
const _pIA = "podIdentityAssociations";
const _pIC = "podIdentityConfiguration";
const _pU = "productUrl";
const _pV = "platformVersion";
const _pVl = "platformVersions";
const _pa = "params";
const _pr = "provider";
const _pre = "preserve";
const _pu = "publishers";
const _r = "resources";
const _rA = "roleArn";
const _rASG = "remoteAccessSecurityGroup";
const _rAe = "remoteAccess";
const _rAep = "repairAction";
const _rAes = "resourceArn";
const _rC = "requiresConfiguration";
const _rCe = "resolveConflicts";
const _rCeq = "requiredClaims";
const _rD = "releaseDate";
const _rI = "resourceIds";
const _rIP = "requiresIamPermissions";
const _rL = "removeLabels";
const _rMP = "recommendedManagedPolicies";
const _rNC = "remoteNetworkConfig";
const _rNN = "remoteNodeNetworks";
const _rPN = "remotePodNetworks";
const _rT = "removeTaints";
const _rV = "releaseVersion";
const _rVC = "resourcesVpcConfig";
const _rW = "replacedWith";
const _re = "recommendation";
const _rea = "reason";
const _s = "status";
const _sA = "serviceAccount";
const _sARA = "serviceAccountRoleArn";
const _sAt = "startedAt";
const _sC = "storageConfig";
const _sCc = "scalingConfig";
const _sGI = "securityGroupIds";
const _sI = "subscriptionId";
const _sIC = "serviceIpv4Cidr";
const _sICe = "serviceIpv6Cidr";
const _sIu = "subnetIds";
const _sSG = "sourceSecurityGroups";
const _sSRV = "startServingReplacementVersion";
const _sSV = "stopServingVersion";
const _sT = "supportType";
const _se = "selectors";
const _ser = "server";
const _sm = "smithy.ts.sdk.synthetic.com.amazonaws.eks";
const _st = "statuses";
const _su = "subscription";
const _sub = "subnets";
const _subs = "subscriptions";
const _t = "tags";
const _tK = "tagKeys";
const _tL = "taintsList";
const _tRA = "targetRoleArn";
const _ta = "taints";
const _te = "term";
const _to = "token";
const _ty = "type";
const _typ = "types";
const _u = "username";
const _uA = "userAgent";
const _uC = "updateConfig";
const _uCs = "usernameClaim";
const _uI = "updateId";
const _uIp = "updateIds";
const _uP = "upgradePolicy";
const _uPs = "usernamePrefix";
const _uS = "updateStrategy";
const _un = "unit";
const _up = "update";
const _us = "usage";
const _v = "version";
const _vI = "vpcId";
const _vS = "versionStatus";
const _vZ = "validZones";
const _va = "value";
const _zSC = "zonalShiftConfig";
const n0 = "com.amazonaws.eks";
import { TypeRegistry } from "@smithy/core/schema";
import { EKSServiceException as __EKSServiceException } from "../models/EKSServiceException";
import { AccessDeniedException as __AccessDeniedException, BadRequestException as __BadRequestException, ClientException as __ClientException, InvalidParameterException as __InvalidParameterException, InvalidRequestException as __InvalidRequestException, InvalidStateException as __InvalidStateException, NotFoundException as __NotFoundException, ResourceInUseException as __ResourceInUseException, ResourceLimitExceededException as __ResourceLimitExceededException, ResourceNotFoundException as __ResourceNotFoundException, ResourcePropagationDelayException as __ResourcePropagationDelayException, ServerException as __ServerException, ServiceUnavailableException as __ServiceUnavailableException, ThrottlingException as __ThrottlingException, UnsupportedAvailabilityZoneException as __UnsupportedAvailabilityZoneException, } from "../models/errors";
export var AccessConfigResponse = [3, n0, _ACR, 0, [_bCCAP, _aM], [2, 0]];
export var AccessDeniedException = [
    -3,
    n0,
    _ADE,
    {
        [_e]: _c,
        [_hE]: 403,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(AccessDeniedException, __AccessDeniedException);
export var AccessEntry = [
    3,
    n0,
    _AE,
    0,
    [_cN, _pA, _kG, _aEA, _cA, _mA, _t, _u, _ty],
    [0, 0, 64 | 0, 0, 4, 4, 128 | 0, 0, 0],
];
export var AccessPolicy = [3, n0, _AP, 0, [_n, _a], [0, 0]];
export var AccessScope = [3, n0, _AS, 0, [_ty, _na], [0, 64 | 0]];
export var Addon = [
    3,
    n0,
    _A,
    0,
    [_aN, _cN, _s, _aV, _h, _aA, _cA, _mA, _sARA, _t, _p, _o, _mI, _cV, _pIA, _nC],
    [
        0,
        0,
        0,
        0,
        () => AddonHealth,
        0,
        4,
        4,
        0,
        128 | 0,
        0,
        0,
        () => MarketplaceInformation,
        0,
        64 | 0,
        () => AddonNamespaceConfigResponse,
    ],
];
export var AddonCompatibilityDetail = [3, n0, _ACD, 0, [_n, _cVo], [0, 64 | 0]];
export var AddonHealth = [3, n0, _AH, 0, [_i], [() => AddonIssueList]];
export var AddonInfo = [
    3,
    n0,
    _AI,
    0,
    [_aN, _ty, _aVd, _p, _o, _mI, _dN],
    [0, 0, () => AddonVersionInfoList, 0, 0, () => MarketplaceInformation, 0],
];
export var AddonIssue = [3, n0, _AId, 0, [_co, _m, _rI], [0, 0, 64 | 0]];
export var AddonNamespaceConfigRequest = [3, n0, _ANCR, 0, [_nam], [0]];
export var AddonNamespaceConfigResponse = [3, n0, _ANCRd, 0, [_nam], [0]];
export var AddonPodIdentityAssociations = [3, n0, _APIA, 0, [_sA, _rA], [0, 0]];
export var AddonPodIdentityConfiguration = [3, n0, _APIC, 0, [_sA, _rMP], [0, 64 | 0]];
export var AddonVersionInfo = [
    3,
    n0,
    _AVI,
    0,
    [_aV, _ar, _cT, _com, _rC, _rIP],
    [0, 64 | 0, 64 | 0, () => Compatibilities, 2, 2],
];
export var AssociateAccessPolicyRequest = [
    3,
    n0,
    _AAPR,
    0,
    [_cN, _pA, _pAo, _aS],
    [[0, 1], [0, 1], 0, () => AccessScope],
];
export var AssociateAccessPolicyResponse = [
    3,
    n0,
    _AAPRs,
    0,
    [_cN, _pA, _aAP],
    [0, 0, () => AssociatedAccessPolicy],
];
export var AssociatedAccessPolicy = [
    3,
    n0,
    _AAP,
    0,
    [_pAo, _aS, _aAs, _mA],
    [0, () => AccessScope, 4, 4],
];
export var AssociateEncryptionConfigRequest = [
    3,
    n0,
    _AECR,
    0,
    [_cN, _eC, _cRT],
    [[0, 1], () => EncryptionConfigList, [0, 4]],
];
export var AssociateEncryptionConfigResponse = [3, n0, _AECRs, 0, [_up], [() => Update]];
export var AssociateIdentityProviderConfigRequest = [
    3,
    n0,
    _AIPCR,
    0,
    [_cN, _oi, _t, _cRT],
    [[0, 1], () => OidcIdentityProviderConfigRequest, 128 | 0, [0, 4]],
];
export var AssociateIdentityProviderConfigResponse = [
    3,
    n0,
    _AIPCRs,
    0,
    [_up, _t],
    [() => Update, 128 | 0],
];
export var AutoScalingGroup = [3, n0, _ASG, 0, [_n], [0]];
export var BadRequestException = [
    -3,
    n0,
    _BRE,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(BadRequestException, __BadRequestException);
export var BlockStorage = [3, n0, _BS, 0, [_en], [2]];
export var Certificate = [3, n0, _C, 0, [_d], [0]];
export var ClientException = [
    -3,
    n0,
    _CE,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_cN, _nN, _aN, _sI, _m],
    [0, 0, 0, 0, 0],
];
TypeRegistry.for(n0).registerError(ClientException, __ClientException);
export var ClientStat = [3, n0, _CS, 0, [_uA, _nORLD, _lRT], [0, 1, 4]];
export var Cluster = [
    3,
    n0,
    _Cl,
    0,
    [
        _n,
        _a,
        _cA,
        _v,
        _end,
        _rA,
        _rVC,
        _kNC,
        _l,
        _id,
        _s,
        _cAe,
        _cRT,
        _pV,
        _t,
        _eC,
        _cC,
        _id_,
        _h,
        _oC,
        _aC,
        _uP,
        _zSC,
        _rNC,
        _cCo,
        _sC,
        _dP,
    ],
    [
        0,
        0,
        4,
        0,
        0,
        0,
        () => VpcConfigResponse,
        () => KubernetesNetworkConfigResponse,
        () => Logging,
        () => Identity,
        0,
        () => Certificate,
        0,
        0,
        128 | 0,
        () => EncryptionConfigList,
        () => ConnectorConfigResponse,
        0,
        () => ClusterHealth,
        () => OutpostConfigResponse,
        () => AccessConfigResponse,
        () => UpgradePolicyResponse,
        () => ZonalShiftConfigResponse,
        () => RemoteNetworkConfigResponse,
        () => ComputeConfigResponse,
        () => StorageConfigResponse,
        2,
    ],
];
export var ClusterHealth = [3, n0, _CH, 0, [_i], [() => ClusterIssueList]];
export var ClusterIssue = [3, n0, _CI, 0, [_co, _m, _rI], [0, 0, 64 | 0]];
export var ClusterVersionInformation = [
    3,
    n0,
    _CVI,
    0,
    [_cVl, _cTl, _dPV, _dV, _rD, _eOSSD, _eOESD, _s, _vS, _kPV],
    [0, 0, 0, 2, 4, 4, 4, 0, 0, 0],
];
export var Compatibility = [3, n0, _Co, 0, [_cVl, _pVl, _dV], [0, 64 | 0, 2]];
export var ComputeConfigRequest = [3, n0, _CCR, 0, [_en, _nP, _nRA], [2, 64 | 0, 0]];
export var ComputeConfigResponse = [3, n0, _CCRo, 0, [_en, _nP, _nRA], [2, 64 | 0, 0]];
export var ConnectorConfigRequest = [3, n0, _CCRon, 0, [_rA, _pr], [0, 0]];
export var ConnectorConfigResponse = [
    3,
    n0,
    _CCRonn,
    0,
    [_aI, _aCc, _aE, _pr, _rA],
    [0, 0, 4, 0, 0],
];
export var ControlPlanePlacementRequest = [3, n0, _CPPR, 0, [_gN], [0]];
export var ControlPlanePlacementResponse = [3, n0, _CPPRo, 0, [_gN], [0]];
export var CreateAccessConfigRequest = [3, n0, _CACR, 0, [_bCCAP, _aM], [2, 0]];
export var CreateAccessEntryRequest = [
    3,
    n0,
    _CAER,
    0,
    [_cN, _pA, _kG, _t, _cRT, _u, _ty],
    [[0, 1], 0, 64 | 0, 128 | 0, [0, 4], 0, 0],
];
export var CreateAccessEntryResponse = [3, n0, _CAERr, 0, [_aEc], [() => AccessEntry]];
export var CreateAddonRequest = [
    3,
    n0,
    _CAR,
    0,
    [_cN, _aN, _aV, _sARA, _rCe, _cRT, _t, _cV, _pIA, _nC],
    [[0, 1], 0, 0, 0, 0, [0, 4], 128 | 0, 0, () => AddonPodIdentityAssociationsList, () => AddonNamespaceConfigRequest],
];
export var CreateAddonResponse = [3, n0, _CARr, 0, [_ad], [() => Addon]];
export var CreateClusterRequest = [
    3,
    n0,
    _CCRr,
    0,
    [_n, _v, _rA, _rVC, _kNC, _l, _cRT, _t, _eC, _oC, _aC, _bSMA, _uP, _zSC, _rNC, _cCo, _sC, _dP],
    [
        0,
        0,
        0,
        () => VpcConfigRequest,
        () => KubernetesNetworkConfigRequest,
        () => Logging,
        [0, 4],
        128 | 0,
        () => EncryptionConfigList,
        () => OutpostConfigRequest,
        () => CreateAccessConfigRequest,
        2,
        () => UpgradePolicyRequest,
        () => ZonalShiftConfigRequest,
        () => RemoteNetworkConfigRequest,
        () => ComputeConfigRequest,
        () => StorageConfigRequest,
        2,
    ],
];
export var CreateClusterResponse = [3, n0, _CCRre, 0, [_cl], [() => Cluster]];
export var CreateEksAnywhereSubscriptionRequest = [
    3,
    n0,
    _CEASR,
    0,
    [_n, _te, _lQ, _lT, _aR, _cRT, _t],
    [0, () => EksAnywhereSubscriptionTerm, 1, 0, 2, [0, 4], 128 | 0],
];
export var CreateEksAnywhereSubscriptionResponse = [
    3,
    n0,
    _CEASRr,
    0,
    [_su],
    [() => EksAnywhereSubscription],
];
export var CreateFargateProfileRequest = [
    3,
    n0,
    _CFPR,
    0,
    [_fPN, _cN, _pERA, _sub, _se, _cRT, _t],
    [0, [0, 1], 0, 64 | 0, () => FargateProfileSelectors, [0, 4], 128 | 0],
];
export var CreateFargateProfileResponse = [3, n0, _CFPRr, 0, [_fP], [() => FargateProfile]];
export var CreateNodegroupRequest = [
    3,
    n0,
    _CNR,
    0,
    [_cN, _nN, _sCc, _dS, _sub, _iT, _aT, _rAe, _nR, _la, _ta, _t, _cRT, _lTa, _uC, _nRC, _cTa, _v, _rV],
    [
        [0, 1],
        0,
        () => NodegroupScalingConfig,
        1,
        64 | 0,
        64 | 0,
        0,
        () => RemoteAccessConfig,
        0,
        128 | 0,
        () => taintsList,
        128 | 0,
        [0, 4],
        () => LaunchTemplateSpecification,
        () => NodegroupUpdateConfig,
        () => NodeRepairConfig,
        0,
        0,
        0,
    ],
];
export var CreateNodegroupResponse = [3, n0, _CNRr, 0, [_no], [() => Nodegroup]];
export var CreatePodIdentityAssociationRequest = [
    3,
    n0,
    _CPIAR,
    0,
    [_cN, _nam, _sA, _rA, _cRT, _t, _dST, _tRA],
    [[0, 1], 0, 0, 0, [0, 4], 128 | 0, 2, 0],
];
export var CreatePodIdentityAssociationResponse = [
    3,
    n0,
    _CPIARr,
    0,
    [_as],
    [() => PodIdentityAssociation],
];
export var DeleteAccessEntryRequest = [
    3,
    n0,
    _DAER,
    0,
    [_cN, _pA],
    [
        [0, 1],
        [0, 1],
    ],
];
export var DeleteAccessEntryResponse = [3, n0, _DAERe, 0, [], []];
export var DeleteAddonRequest = [
    3,
    n0,
    _DAR,
    0,
    [_cN, _aN, _pre],
    [
        [0, 1],
        [0, 1],
        [
            2,
            {
                [_hQ]: _pre,
            },
        ],
    ],
];
export var DeleteAddonResponse = [3, n0, _DARe, 0, [_ad], [() => Addon]];
export var DeleteClusterRequest = [3, n0, _DCR, 0, [_n], [[0, 1]]];
export var DeleteClusterResponse = [3, n0, _DCRe, 0, [_cl], [() => Cluster]];
export var DeleteEksAnywhereSubscriptionRequest = [3, n0, _DEASR, 0, [_id_], [[0, 1]]];
export var DeleteEksAnywhereSubscriptionResponse = [
    3,
    n0,
    _DEASRe,
    0,
    [_su],
    [() => EksAnywhereSubscription],
];
export var DeleteFargateProfileRequest = [
    3,
    n0,
    _DFPR,
    0,
    [_cN, _fPN],
    [
        [0, 1],
        [0, 1],
    ],
];
export var DeleteFargateProfileResponse = [3, n0, _DFPRe, 0, [_fP], [() => FargateProfile]];
export var DeleteNodegroupRequest = [
    3,
    n0,
    _DNR,
    0,
    [_cN, _nN],
    [
        [0, 1],
        [0, 1],
    ],
];
export var DeleteNodegroupResponse = [3, n0, _DNRe, 0, [_no], [() => Nodegroup]];
export var DeletePodIdentityAssociationRequest = [
    3,
    n0,
    _DPIAR,
    0,
    [_cN, _aIs],
    [
        [0, 1],
        [0, 1],
    ],
];
export var DeletePodIdentityAssociationResponse = [
    3,
    n0,
    _DPIARe,
    0,
    [_as],
    [() => PodIdentityAssociation],
];
export var DeprecationDetail = [
    3,
    n0,
    _DD,
    0,
    [_us, _rW, _sSV, _sSRV, _cS],
    [0, 0, 0, 0, () => ClientStats],
];
export var DeregisterClusterRequest = [3, n0, _DCRer, 0, [_n], [[0, 1]]];
export var DeregisterClusterResponse = [3, n0, _DCRere, 0, [_cl], [() => Cluster]];
export var DescribeAccessEntryRequest = [
    3,
    n0,
    _DAERes,
    0,
    [_cN, _pA],
    [
        [0, 1],
        [0, 1],
    ],
];
export var DescribeAccessEntryResponse = [3, n0, _DAEResc, 0, [_aEc], [() => AccessEntry]];
export var DescribeAddonConfigurationRequest = [
    3,
    n0,
    _DACR,
    0,
    [_aN, _aV],
    [
        [
            0,
            {
                [_hQ]: _aN,
            },
        ],
        [
            0,
            {
                [_hQ]: _aV,
            },
        ],
    ],
];
export var DescribeAddonConfigurationResponse = [
    3,
    n0,
    _DACRe,
    0,
    [_aN, _aV, _cSo, _pIC],
    [0, 0, 0, () => AddonPodIdentityConfigurationList],
];
export var DescribeAddonRequest = [
    3,
    n0,
    _DARes,
    0,
    [_cN, _aN],
    [
        [0, 1],
        [0, 1],
    ],
];
export var DescribeAddonResponse = [3, n0, _DAResc, 0, [_ad], [() => Addon]];
export var DescribeAddonVersionsRequest = [
    3,
    n0,
    _DAVR,
    0,
    [_kV, _mR, _nT, _aN, _typ, _pu, _ow],
    [
        [
            0,
            {
                [_hQ]: _kV,
            },
        ],
        [
            1,
            {
                [_hQ]: _mR,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            0,
            {
                [_hQ]: _aN,
            },
        ],
        [
            64 | 0,
            {
                [_hQ]: _typ,
            },
        ],
        [
            64 | 0,
            {
                [_hQ]: _pu,
            },
        ],
        [
            64 | 0,
            {
                [_hQ]: _ow,
            },
        ],
    ],
];
export var DescribeAddonVersionsResponse = [3, n0, _DAVRe, 0, [_add, _nT], [() => Addons, 0]];
export var DescribeClusterRequest = [3, n0, _DCRes, 0, [_n], [[0, 1]]];
export var DescribeClusterResponse = [3, n0, _DCResc, 0, [_cl], [() => Cluster]];
export var DescribeClusterVersionsRequest = [
    3,
    n0,
    _DCVR,
    0,
    [_cTl, _mR, _nT, _dO, _iA, _cVlu, _s, _vS],
    [
        [
            0,
            {
                [_hQ]: _cTl,
            },
        ],
        [
            1,
            {
                [_hQ]: _mR,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            2,
            {
                [_hQ]: _dO,
            },
        ],
        [
            2,
            {
                [_hQ]: _iA,
            },
        ],
        [
            64 | 0,
            {
                [_hQ]: _cVlu,
            },
        ],
        [
            0,
            {
                [_hQ]: _s,
            },
        ],
        [
            0,
            {
                [_hQ]: _vS,
            },
        ],
    ],
];
export var DescribeClusterVersionsResponse = [
    3,
    n0,
    _DCVRe,
    0,
    [_nT, _cVlu],
    [0, () => ClusterVersionList],
];
export var DescribeEksAnywhereSubscriptionRequest = [3, n0, _DEASRes, 0, [_id_], [[0, 1]]];
export var DescribeEksAnywhereSubscriptionResponse = [
    3,
    n0,
    _DEASResc,
    0,
    [_su],
    [() => EksAnywhereSubscription],
];
export var DescribeFargateProfileRequest = [
    3,
    n0,
    _DFPRes,
    0,
    [_cN, _fPN],
    [
        [0, 1],
        [0, 1],
    ],
];
export var DescribeFargateProfileResponse = [3, n0, _DFPResc, 0, [_fP], [() => FargateProfile]];
export var DescribeIdentityProviderConfigRequest = [
    3,
    n0,
    _DIPCR,
    0,
    [_cN, _iPC],
    [[0, 1], () => IdentityProviderConfig],
];
export var DescribeIdentityProviderConfigResponse = [
    3,
    n0,
    _DIPCRe,
    0,
    [_iPC],
    [() => IdentityProviderConfigResponse],
];
export var DescribeInsightRequest = [
    3,
    n0,
    _DIR,
    0,
    [_cN, _id_],
    [
        [0, 1],
        [0, 1],
    ],
];
export var DescribeInsightResponse = [3, n0, _DIRe, 0, [_in], [() => Insight]];
export var DescribeInsightsRefreshRequest = [3, n0, _DIRR, 0, [_cN], [[0, 1]]];
export var DescribeInsightsRefreshResponse = [
    3,
    n0,
    _DIRRe,
    0,
    [_m, _s, _sAt, _eA],
    [0, 0, 4, 4],
];
export var DescribeNodegroupRequest = [
    3,
    n0,
    _DNRes,
    0,
    [_cN, _nN],
    [
        [0, 1],
        [0, 1],
    ],
];
export var DescribeNodegroupResponse = [3, n0, _DNResc, 0, [_no], [() => Nodegroup]];
export var DescribePodIdentityAssociationRequest = [
    3,
    n0,
    _DPIARes,
    0,
    [_cN, _aIs],
    [
        [0, 1],
        [0, 1],
    ],
];
export var DescribePodIdentityAssociationResponse = [
    3,
    n0,
    _DPIAResc,
    0,
    [_as],
    [() => PodIdentityAssociation],
];
export var DescribeUpdateRequest = [
    3,
    n0,
    _DUR,
    0,
    [_n, _uI, _nN, _aN],
    [
        [0, 1],
        [0, 1],
        [
            0,
            {
                [_hQ]: _nN,
            },
        ],
        [
            0,
            {
                [_hQ]: _aN,
            },
        ],
    ],
];
export var DescribeUpdateResponse = [3, n0, _DURe, 0, [_up], [() => Update]];
export var DisassociateAccessPolicyRequest = [
    3,
    n0,
    _DAPR,
    0,
    [_cN, _pA, _pAo],
    [
        [0, 1],
        [0, 1],
        [0, 1],
    ],
];
export var DisassociateAccessPolicyResponse = [3, n0, _DAPRi, 0, [], []];
export var DisassociateIdentityProviderConfigRequest = [
    3,
    n0,
    _DIPCRi,
    0,
    [_cN, _iPC, _cRT],
    [[0, 1], () => IdentityProviderConfig, [0, 4]],
];
export var DisassociateIdentityProviderConfigResponse = [
    3,
    n0,
    _DIPCRis,
    0,
    [_up],
    [() => Update],
];
export var EksAnywhereSubscription = [
    3,
    n0,
    _EAS,
    0,
    [_id_, _a, _cA, _eD, _eDx, _lQ, _lT, _te, _s, _aR, _lA, _li, _t],
    [0, 0, 4, 4, 4, 1, 0, () => EksAnywhereSubscriptionTerm, 0, 2, 64 | 0, () => LicenseList, 128 | 0],
];
export var EksAnywhereSubscriptionTerm = [3, n0, _EAST, 0, [_du, _un], [1, 0]];
export var ElasticLoadBalancing = [3, n0, _ELB, 0, [_en], [2]];
export var EncryptionConfig = [3, n0, _EC, 0, [_r, _pr], [64 | 0, () => Provider]];
export var ErrorDetail = [3, n0, _ED, 0, [_eCr, _eM, _rI], [0, 0, 64 | 0]];
export var FargateProfile = [
    3,
    n0,
    _FP,
    0,
    [_fPN, _fPA, _cN, _cA, _pERA, _sub, _se, _s, _t, _h],
    [0, 0, 0, 4, 0, 64 | 0, () => FargateProfileSelectors, 0, 128 | 0, () => FargateProfileHealth],
];
export var FargateProfileHealth = [3, n0, _FPH, 0, [_i], [() => FargateProfileIssueList]];
export var FargateProfileIssue = [3, n0, _FPI, 0, [_co, _m, _rI], [0, 0, 64 | 0]];
export var FargateProfileSelector = [3, n0, _FPS, 0, [_nam, _la], [0, 128 | 0]];
export var Identity = [3, n0, _I, 0, [_oi], [() => OIDC]];
export var IdentityProviderConfig = [3, n0, _IPC, 0, [_ty, _n], [0, 0]];
export var IdentityProviderConfigResponse = [
    3,
    n0,
    _IPCR,
    0,
    [_oi],
    [() => OidcIdentityProviderConfig],
];
export var Insight = [
    3,
    n0,
    _In,
    0,
    [_id_, _n, _ca, _kV, _lRTa, _lTT, _de, _iS, _re, _aId, _r, _cSS],
    [
        0,
        0,
        0,
        0,
        4,
        4,
        0,
        () => InsightStatus,
        0,
        128 | 0,
        () => InsightResourceDetails,
        () => InsightCategorySpecificSummary,
    ],
];
export var InsightCategorySpecificSummary = [
    3,
    n0,
    _ICSS,
    0,
    [_dD, _aCD],
    [() => DeprecationDetails, () => AddonCompatibilityDetails],
];
export var InsightResourceDetail = [
    3,
    n0,
    _IRD,
    0,
    [_iS, _kRU, _a],
    [() => InsightStatus, 0, 0],
];
export var InsightsFilter = [3, n0, _IF, 0, [_cat, _kVu, _st], [64 | 0, 64 | 0, 64 | 0]];
export var InsightStatus = [3, n0, _IS, 0, [_s, _rea], [0, 0]];
export var InsightSummary = [
    3,
    n0,
    _ISn,
    0,
    [_id_, _n, _ca, _kV, _lRTa, _lTT, _de, _iS],
    [0, 0, 0, 0, 4, 4, 0, () => InsightStatus],
];
export var InvalidParameterException = [
    -3,
    n0,
    _IPE,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_cN, _nN, _fPN, _aN, _sI, _m],
    [0, 0, 0, 0, 0, 0],
];
TypeRegistry.for(n0).registerError(InvalidParameterException, __InvalidParameterException);
export var InvalidRequestException = [
    -3,
    n0,
    _IRE,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_cN, _nN, _aN, _sI, _m],
    [0, 0, 0, 0, 0],
];
TypeRegistry.for(n0).registerError(InvalidRequestException, __InvalidRequestException);
export var InvalidStateException = [
    -3,
    n0,
    _ISE,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_cN, _m],
    [0, 0],
];
TypeRegistry.for(n0).registerError(InvalidStateException, __InvalidStateException);
export var Issue = [3, n0, _Is, 0, [_co, _m, _rI], [0, 0, 64 | 0]];
export var KubernetesNetworkConfigRequest = [
    3,
    n0,
    _KNCR,
    0,
    [_sIC, _iF, _eLB],
    [0, 0, () => ElasticLoadBalancing],
];
export var KubernetesNetworkConfigResponse = [
    3,
    n0,
    _KNCRu,
    0,
    [_sIC, _sICe, _iF, _eLB],
    [0, 0, 0, () => ElasticLoadBalancing],
];
export var LaunchTemplateSpecification = [3, n0, _LTS, 0, [_n, _v, _id_], [0, 0, 0]];
export var License = [3, n0, _L, 0, [_id_, _to], [0, 0]];
export var ListAccessEntriesRequest = [
    3,
    n0,
    _LAER,
    0,
    [_cN, _aPA, _mR, _nT],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _aPA,
            },
        ],
        [
            1,
            {
                [_hQ]: _mR,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
    ],
];
export var ListAccessEntriesResponse = [3, n0, _LAERi, 0, [_aEcc, _nT], [64 | 0, 0]];
export var ListAccessPoliciesRequest = [
    3,
    n0,
    _LAPR,
    0,
    [_mR, _nT],
    [
        [
            1,
            {
                [_hQ]: _mR,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
    ],
];
export var ListAccessPoliciesResponse = [
    3,
    n0,
    _LAPRi,
    0,
    [_aP, _nT],
    [() => AccessPoliciesList, 0],
];
export var ListAddonsRequest = [
    3,
    n0,
    _LAR,
    0,
    [_cN, _mR, _nT],
    [
        [0, 1],
        [
            1,
            {
                [_hQ]: _mR,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
    ],
];
export var ListAddonsResponse = [3, n0, _LARi, 0, [_add, _nT], [64 | 0, 0]];
export var ListAssociatedAccessPoliciesRequest = [
    3,
    n0,
    _LAAPR,
    0,
    [_cN, _pA, _mR, _nT],
    [
        [0, 1],
        [0, 1],
        [
            1,
            {
                [_hQ]: _mR,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
    ],
];
export var ListAssociatedAccessPoliciesResponse = [
    3,
    n0,
    _LAAPRi,
    0,
    [_cN, _pA, _nT, _aAPs],
    [0, 0, 0, () => AssociatedAccessPoliciesList],
];
export var ListClustersRequest = [
    3,
    n0,
    _LCR,
    0,
    [_mR, _nT, _inc],
    [
        [
            1,
            {
                [_hQ]: _mR,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            64 | 0,
            {
                [_hQ]: _inc,
            },
        ],
    ],
];
export var ListClustersResponse = [3, n0, _LCRi, 0, [_clu, _nT], [64 | 0, 0]];
export var ListEksAnywhereSubscriptionsRequest = [
    3,
    n0,
    _LEASR,
    0,
    [_mR, _nT, _iSn],
    [
        [
            1,
            {
                [_hQ]: _mR,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            64 | 0,
            {
                [_hQ]: _iSn,
            },
        ],
    ],
];
export var ListEksAnywhereSubscriptionsResponse = [
    3,
    n0,
    _LEASRi,
    0,
    [_subs, _nT],
    [() => EksAnywhereSubscriptionList, 0],
];
export var ListFargateProfilesRequest = [
    3,
    n0,
    _LFPR,
    0,
    [_cN, _mR, _nT],
    [
        [0, 1],
        [
            1,
            {
                [_hQ]: _mR,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
    ],
];
export var ListFargateProfilesResponse = [3, n0, _LFPRi, 0, [_fPNa, _nT], [64 | 0, 0]];
export var ListIdentityProviderConfigsRequest = [
    3,
    n0,
    _LIPCR,
    0,
    [_cN, _mR, _nT],
    [
        [0, 1],
        [
            1,
            {
                [_hQ]: _mR,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
    ],
];
export var ListIdentityProviderConfigsResponse = [
    3,
    n0,
    _LIPCRi,
    0,
    [_iPCd, _nT],
    [() => IdentityProviderConfigs, 0],
];
export var ListInsightsRequest = [
    3,
    n0,
    _LIR,
    0,
    [_cN, _f, _mR, _nT],
    [[0, 1], () => InsightsFilter, 1, 0],
];
export var ListInsightsResponse = [3, n0, _LIRi, 0, [_ins, _nT], [() => InsightSummaries, 0]];
export var ListNodegroupsRequest = [
    3,
    n0,
    _LNR,
    0,
    [_cN, _mR, _nT],
    [
        [0, 1],
        [
            1,
            {
                [_hQ]: _mR,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
    ],
];
export var ListNodegroupsResponse = [3, n0, _LNRi, 0, [_nod, _nT], [64 | 0, 0]];
export var ListPodIdentityAssociationsRequest = [
    3,
    n0,
    _LPIAR,
    0,
    [_cN, _nam, _sA, _mR, _nT],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _nam,
            },
        ],
        [
            0,
            {
                [_hQ]: _sA,
            },
        ],
        [
            1,
            {
                [_hQ]: _mR,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
    ],
];
export var ListPodIdentityAssociationsResponse = [
    3,
    n0,
    _LPIARi,
    0,
    [_ass, _nT],
    [() => PodIdentityAssociationSummaries, 0],
];
export var ListTagsForResourceRequest = [3, n0, _LTFRR, 0, [_rAes], [[0, 1]]];
export var ListTagsForResourceResponse = [3, n0, _LTFRRi, 0, [_t], [128 | 0]];
export var ListUpdatesRequest = [
    3,
    n0,
    _LUR,
    0,
    [_n, _nN, _aN, _nT, _mR],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _nN,
            },
        ],
        [
            0,
            {
                [_hQ]: _aN,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            1,
            {
                [_hQ]: _mR,
            },
        ],
    ],
];
export var ListUpdatesResponse = [3, n0, _LURi, 0, [_uIp, _nT], [64 | 0, 0]];
export var Logging = [3, n0, _Lo, 0, [_cL], [() => LogSetups]];
export var LogSetup = [3, n0, _LS, 0, [_typ, _en], [64 | 0, 2]];
export var MarketplaceInformation = [3, n0, _MI, 0, [_pI, _pU], [0, 0]];
export var Nodegroup = [
    3,
    n0,
    _N,
    0,
    [
        _nN,
        _nA,
        _cN,
        _v,
        _rV,
        _cA,
        _mA,
        _s,
        _cTa,
        _sCc,
        _iT,
        _sub,
        _rAe,
        _aT,
        _nR,
        _la,
        _ta,
        _r,
        _dS,
        _h,
        _uC,
        _nRC,
        _lTa,
        _t,
    ],
    [
        0,
        0,
        0,
        0,
        0,
        4,
        4,
        0,
        0,
        () => NodegroupScalingConfig,
        64 | 0,
        64 | 0,
        () => RemoteAccessConfig,
        0,
        0,
        128 | 0,
        () => taintsList,
        () => NodegroupResources,
        1,
        () => NodegroupHealth,
        () => NodegroupUpdateConfig,
        () => NodeRepairConfig,
        () => LaunchTemplateSpecification,
        128 | 0,
    ],
];
export var NodegroupHealth = [3, n0, _NH, 0, [_i], [() => IssueList]];
export var NodegroupResources = [3, n0, _NR, 0, [_aSG, _rASG], [() => AutoScalingGroupList, 0]];
export var NodegroupScalingConfig = [3, n0, _NSC, 0, [_mS, _mSa, _dSe], [1, 1, 1]];
export var NodegroupUpdateConfig = [3, n0, _NUC, 0, [_mU, _mUP, _uS], [1, 1, 0]];
export var NodeRepairConfig = [
    3,
    n0,
    _NRC,
    0,
    [_en, _mUNTC, _mUNTP, _mPNRC, _mPNRP, _nRCO],
    [2, 1, 1, 1, 1, () => NodeRepairConfigOverridesList],
];
export var NodeRepairConfigOverrides = [
    3,
    n0,
    _NRCO,
    0,
    [_nMC, _nUR, _mRWTM, _rAep],
    [0, 0, 1, 0],
];
export var NotFoundException = [
    -3,
    n0,
    _NFE,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(NotFoundException, __NotFoundException);
export var OIDC = [3, n0, _OIDC, 0, [_is], [0]];
export var OidcIdentityProviderConfig = [
    3,
    n0,
    _OIPC,
    0,
    [_iPCN, _iPCA, _cN, _iU, _cI, _uCs, _uPs, _gC, _gP, _rCeq, _t, _s],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 128 | 0, 128 | 0, 0],
];
export var OidcIdentityProviderConfigRequest = [
    3,
    n0,
    _OIPCR,
    0,
    [_iPCN, _iU, _cI, _uCs, _uPs, _gC, _gP, _rCeq],
    [0, 0, 0, 0, 0, 0, 0, 128 | 0],
];
export var OutpostConfigRequest = [
    3,
    n0,
    _OCR,
    0,
    [_oA, _cPIT, _cPP],
    [64 | 0, 0, () => ControlPlanePlacementRequest],
];
export var OutpostConfigResponse = [
    3,
    n0,
    _OCRu,
    0,
    [_oA, _cPIT, _cPP],
    [64 | 0, 0, () => ControlPlanePlacementResponse],
];
export var PodIdentityAssociation = [
    3,
    n0,
    _PIA,
    0,
    [_cN, _nam, _sA, _rA, _aAss, _aIs, _t, _cA, _mA, _oAw, _dST, _tRA, _eI],
    [0, 0, 0, 0, 0, 0, 128 | 0, 4, 4, 0, 2, 0, 0],
];
export var PodIdentityAssociationSummary = [
    3,
    n0,
    _PIAS,
    0,
    [_cN, _nam, _sA, _aAss, _aIs, _oAw],
    [0, 0, 0, 0, 0, 0],
];
export var Provider = [3, n0, _P, 0, [_kA], [0]];
export var RegisterClusterRequest = [
    3,
    n0,
    _RCR,
    0,
    [_n, _cC, _cRT, _t],
    [0, () => ConnectorConfigRequest, [0, 4], 128 | 0],
];
export var RegisterClusterResponse = [3, n0, _RCRe, 0, [_cl], [() => Cluster]];
export var RemoteAccessConfig = [3, n0, _RAC, 0, [_eSK, _sSG], [0, 64 | 0]];
export var RemoteNetworkConfigRequest = [
    3,
    n0,
    _RNCR,
    0,
    [_rNN, _rPN],
    [() => RemoteNodeNetworkList, () => RemotePodNetworkList],
];
export var RemoteNetworkConfigResponse = [
    3,
    n0,
    _RNCRe,
    0,
    [_rNN, _rPN],
    [() => RemoteNodeNetworkList, () => RemotePodNetworkList],
];
export var RemoteNodeNetwork = [3, n0, _RNN, 0, [_ci], [64 | 0]];
export var RemotePodNetwork = [3, n0, _RPN, 0, [_ci], [64 | 0]];
export var ResourceInUseException = [
    -3,
    n0,
    _RIUE,
    {
        [_e]: _c,
        [_hE]: 409,
    },
    [_cN, _nN, _aN, _m],
    [0, 0, 0, 0],
];
TypeRegistry.for(n0).registerError(ResourceInUseException, __ResourceInUseException);
export var ResourceLimitExceededException = [
    -3,
    n0,
    _RLEE,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_cN, _nN, _sI, _m],
    [0, 0, 0, 0],
];
TypeRegistry.for(n0).registerError(ResourceLimitExceededException, __ResourceLimitExceededException);
export var ResourceNotFoundException = [
    -3,
    n0,
    _RNFE,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_cN, _nN, _fPN, _aN, _sI, _m],
    [0, 0, 0, 0, 0, 0],
];
TypeRegistry.for(n0).registerError(ResourceNotFoundException, __ResourceNotFoundException);
export var ResourcePropagationDelayException = [
    -3,
    n0,
    _RPDE,
    {
        [_e]: _c,
        [_hE]: 428,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ResourcePropagationDelayException, __ResourcePropagationDelayException);
export var ServerException = [
    -3,
    n0,
    _SE,
    {
        [_e]: _ser,
        [_hE]: 500,
    },
    [_cN, _nN, _aN, _sI, _m],
    [0, 0, 0, 0, 0],
];
TypeRegistry.for(n0).registerError(ServerException, __ServerException);
export var ServiceUnavailableException = [
    -3,
    n0,
    _SUE,
    {
        [_e]: _ser,
        [_hE]: 503,
    },
    [_m],
    [0],
];
TypeRegistry.for(n0).registerError(ServiceUnavailableException, __ServiceUnavailableException);
export var StartInsightsRefreshRequest = [3, n0, _SIRR, 0, [_cN], [[0, 1]]];
export var StartInsightsRefreshResponse = [3, n0, _SIRRt, 0, [_m, _s], [0, 0]];
export var StorageConfigRequest = [3, n0, _SCR, 0, [_bS], [() => BlockStorage]];
export var StorageConfigResponse = [3, n0, _SCRt, 0, [_bS], [() => BlockStorage]];
export var TagResourceRequest = [3, n0, _TRR, 0, [_rAes, _t], [[0, 1], 128 | 0]];
export var TagResourceResponse = [3, n0, _TRRa, 0, [], []];
export var Taint = [3, n0, _T, 0, [_k, _va, _ef], [0, 0, 0]];
export var ThrottlingException = [
    -3,
    n0,
    _TE,
    {
        [_e]: _c,
        [_hE]: 429,
    },
    [_cN, _m],
    [0, 0],
];
TypeRegistry.for(n0).registerError(ThrottlingException, __ThrottlingException);
export var UnsupportedAvailabilityZoneException = [
    -3,
    n0,
    _UAZE,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m, _cN, _nN, _vZ],
    [0, 0, 0, 64 | 0],
];
TypeRegistry.for(n0).registerError(UnsupportedAvailabilityZoneException, __UnsupportedAvailabilityZoneException);
export var UntagResourceRequest = [
    3,
    n0,
    _URR,
    0,
    [_rAes, _tK],
    [
        [0, 1],
        [
            64 | 0,
            {
                [_hQ]: _tK,
            },
        ],
    ],
];
export var UntagResourceResponse = [3, n0, _URRn, 0, [], []];
export var Update = [
    3,
    n0,
    _U,
    0,
    [_id_, _s, _ty, _pa, _cA, _er],
    [0, 0, 0, () => UpdateParams, 4, () => ErrorDetails],
];
export var UpdateAccessConfigRequest = [3, n0, _UACR, 0, [_aM], [0]];
export var UpdateAccessEntryRequest = [
    3,
    n0,
    _UAER,
    0,
    [_cN, _pA, _kG, _cRT, _u],
    [[0, 1], [0, 1], 64 | 0, [0, 4], 0],
];
export var UpdateAccessEntryResponse = [3, n0, _UAERp, 0, [_aEc], [() => AccessEntry]];
export var UpdateAddonRequest = [
    3,
    n0,
    _UAR,
    0,
    [_cN, _aN, _aV, _sARA, _rCe, _cRT, _cV, _pIA],
    [[0, 1], [0, 1], 0, 0, 0, [0, 4], 0, () => AddonPodIdentityAssociationsList],
];
export var UpdateAddonResponse = [3, n0, _UARp, 0, [_up], [() => Update]];
export var UpdateClusterConfigRequest = [
    3,
    n0,
    _UCCR,
    0,
    [_n, _rVC, _l, _cRT, _aC, _uP, _zSC, _cCo, _kNC, _sC, _rNC, _dP],
    [
        [0, 1],
        () => VpcConfigRequest,
        () => Logging,
        [0, 4],
        () => UpdateAccessConfigRequest,
        () => UpgradePolicyRequest,
        () => ZonalShiftConfigRequest,
        () => ComputeConfigRequest,
        () => KubernetesNetworkConfigRequest,
        () => StorageConfigRequest,
        () => RemoteNetworkConfigRequest,
        2,
    ],
];
export var UpdateClusterConfigResponse = [3, n0, _UCCRp, 0, [_up], [() => Update]];
export var UpdateClusterVersionRequest = [
    3,
    n0,
    _UCVR,
    0,
    [_n, _v, _cRT, _fo],
    [[0, 1], 0, [0, 4], 2],
];
export var UpdateClusterVersionResponse = [3, n0, _UCVRp, 0, [_up], [() => Update]];
export var UpdateEksAnywhereSubscriptionRequest = [
    3,
    n0,
    _UEASR,
    0,
    [_id_, _aR, _cRT],
    [[0, 1], 2, [0, 4]],
];
export var UpdateEksAnywhereSubscriptionResponse = [
    3,
    n0,
    _UEASRp,
    0,
    [_su],
    [() => EksAnywhereSubscription],
];
export var UpdateLabelsPayload = [3, n0, _ULP, 0, [_aOUL, _rL], [128 | 0, 64 | 0]];
export var UpdateNodegroupConfigRequest = [
    3,
    n0,
    _UNCR,
    0,
    [_cN, _nN, _la, _ta, _sCc, _uC, _nRC, _cRT],
    [
        [0, 1],
        [0, 1],
        () => UpdateLabelsPayload,
        () => UpdateTaintsPayload,
        () => NodegroupScalingConfig,
        () => NodegroupUpdateConfig,
        () => NodeRepairConfig,
        [0, 4],
    ],
];
export var UpdateNodegroupConfigResponse = [3, n0, _UNCRp, 0, [_up], [() => Update]];
export var UpdateNodegroupVersionRequest = [
    3,
    n0,
    _UNVR,
    0,
    [_cN, _nN, _v, _rV, _lTa, _fo, _cRT],
    [[0, 1], [0, 1], 0, 0, () => LaunchTemplateSpecification, 2, [0, 4]],
];
export var UpdateNodegroupVersionResponse = [3, n0, _UNVRp, 0, [_up], [() => Update]];
export var UpdateParam = [3, n0, _UP, 0, [_ty, _va], [0, 0]];
export var UpdatePodIdentityAssociationRequest = [
    3,
    n0,
    _UPIAR,
    0,
    [_cN, _aIs, _rA, _cRT, _dST, _tRA],
    [[0, 1], [0, 1], 0, [0, 4], 2, 0],
];
export var UpdatePodIdentityAssociationResponse = [
    3,
    n0,
    _UPIARp,
    0,
    [_as],
    [() => PodIdentityAssociation],
];
export var UpdateTaintsPayload = [
    3,
    n0,
    _UTP,
    0,
    [_aOUT, _rT],
    [() => taintsList, () => taintsList],
];
export var UpgradePolicyRequest = [3, n0, _UPR, 0, [_sT], [0]];
export var UpgradePolicyResponse = [3, n0, _UPRp, 0, [_sT], [0]];
export var VpcConfigRequest = [
    3,
    n0,
    _VCR,
    0,
    [_sIu, _sGI, _ePA, _ePAn, _pAC],
    [64 | 0, 64 | 0, 2, 2, 64 | 0],
];
export var VpcConfigResponse = [
    3,
    n0,
    _VCRp,
    0,
    [_sIu, _sGI, _cSGI, _vI, _ePA, _ePAn, _pAC],
    [64 | 0, 64 | 0, 0, 0, 2, 2, 64 | 0],
];
export var ZonalShiftConfigRequest = [3, n0, _ZSCR, 0, [_en], [2]];
export var ZonalShiftConfigResponse = [3, n0, _ZSCRo, 0, [_en], [2]];
export var __Unit = "unit";
export var EKSServiceException = [-3, _sm, "EKSServiceException", 0, [], []];
TypeRegistry.for(_sm).registerError(EKSServiceException, __EKSServiceException);
export var AccessPoliciesList = [1, n0, _APL, 0, () => AccessPolicy];
export var AddonCompatibilityDetails = [1, n0, _ACDd, 0, () => AddonCompatibilityDetail];
export var AddonIssueList = [1, n0, _AIL, 0, () => AddonIssue];
export var AddonPodIdentityAssociationsList = [1, n0, _APIAL, 0, () => AddonPodIdentityAssociations];
export var AddonPodIdentityConfigurationList = [
    1,
    n0,
    _APICL,
    0,
    () => AddonPodIdentityConfiguration,
];
export var Addons = [1, n0, _Ad, 0, () => AddonInfo];
export var AddonVersionInfoList = [1, n0, _AVIL, 0, () => AddonVersionInfo];
export var AssociatedAccessPoliciesList = [1, n0, _AAPL, 0, () => AssociatedAccessPolicy];
export var AutoScalingGroupList = [1, n0, _ASGL, 0, () => AutoScalingGroup];
export var CategoryList = 64 | 0;
export var ClientStats = [1, n0, _CSl, 0, () => ClientStat];
export var ClusterIssueList = [1, n0, _CIL, 0, () => ClusterIssue];
export var ClusterVersionList = [1, n0, _CVL, 0, () => ClusterVersionInformation];
export var Compatibilities = [1, n0, _Com, 0, () => Compatibility];
export var DeprecationDetails = [1, n0, _DDe, 0, () => DeprecationDetail];
export var EksAnywhereSubscriptionList = [1, n0, _EASL, 0, () => EksAnywhereSubscription];
export var EksAnywhereSubscriptionStatusValues = 64 | 0;
export var EncryptionConfigList = [1, n0, _ECL, 0, () => EncryptionConfig];
export var ErrorDetails = [1, n0, _EDr, 0, () => ErrorDetail];
export var FargateProfileIssueList = [1, n0, _FPIL, 0, () => FargateProfileIssue];
export var FargateProfileSelectors = [1, n0, _FPSa, 0, () => FargateProfileSelector];
export var IdentityProviderConfigs = [1, n0, _IPCd, 0, () => IdentityProviderConfig];
export var IncludeClustersList = 64 | 0;
export var InsightResourceDetails = [1, n0, _IRDn, 0, () => InsightResourceDetail];
export var InsightStatusValueList = 64 | 0;
export var InsightSummaries = [1, n0, _ISns, 0, () => InsightSummary];
export var IssueList = [1, n0, _IL, 0, () => Issue];
export var labelsKeyList = 64 | 0;
export var LicenseList = [1, n0, _LL, 0, () => License];
export var LogSetups = [1, n0, _LSo, 0, () => LogSetup];
export var LogTypes = 64 | 0;
export var NodeRepairConfigOverridesList = [1, n0, _NRCOL, 0, () => NodeRepairConfigOverrides];
export var PodIdentityAssociationSummaries = [1, n0, _PIASo, 0, () => PodIdentityAssociationSummary];
export var RemoteNodeNetworkList = [1, n0, _RNNL, 0, () => RemoteNodeNetwork];
export var RemotePodNetworkList = [1, n0, _RPNL, 0, () => RemotePodNetwork];
export var StringList = 64 | 0;
export var TagKeyList = 64 | 0;
export var taintsList = [1, n0, _tL, 0, () => Taint];
export var UpdateParams = [1, n0, _UPp, 0, () => UpdateParam];
export var AdditionalInfoMap = 128 | 0;
export var FargateProfileLabel = 128 | 0;
export var labelsMap = 128 | 0;
export var requiredClaimsMap = 128 | 0;
export var TagMap = 128 | 0;
export var AssociateAccessPolicy = [
    9,
    n0,
    _AAPs,
    {
        [_ht]: ["POST", "/clusters/{clusterName}/access-entries/{principalArn}/access-policies", 200],
    },
    () => AssociateAccessPolicyRequest,
    () => AssociateAccessPolicyResponse,
];
export var AssociateEncryptionConfig = [
    9,
    n0,
    _AEC,
    {
        [_ht]: ["POST", "/clusters/{clusterName}/encryption-config/associate", 200],
    },
    () => AssociateEncryptionConfigRequest,
    () => AssociateEncryptionConfigResponse,
];
export var AssociateIdentityProviderConfig = [
    9,
    n0,
    _AIPC,
    {
        [_ht]: ["POST", "/clusters/{clusterName}/identity-provider-configs/associate", 200],
    },
    () => AssociateIdentityProviderConfigRequest,
    () => AssociateIdentityProviderConfigResponse,
];
export var CreateAccessEntry = [
    9,
    n0,
    _CAE,
    {
        [_ht]: ["POST", "/clusters/{clusterName}/access-entries", 200],
    },
    () => CreateAccessEntryRequest,
    () => CreateAccessEntryResponse,
];
export var CreateAddon = [
    9,
    n0,
    _CA,
    {
        [_ht]: ["POST", "/clusters/{clusterName}/addons", 200],
    },
    () => CreateAddonRequest,
    () => CreateAddonResponse,
];
export var CreateCluster = [
    9,
    n0,
    _CC,
    {
        [_ht]: ["POST", "/clusters", 200],
    },
    () => CreateClusterRequest,
    () => CreateClusterResponse,
];
export var CreateEksAnywhereSubscription = [
    9,
    n0,
    _CEAS,
    {
        [_ht]: ["POST", "/eks-anywhere-subscriptions", 200],
    },
    () => CreateEksAnywhereSubscriptionRequest,
    () => CreateEksAnywhereSubscriptionResponse,
];
export var CreateFargateProfile = [
    9,
    n0,
    _CFP,
    {
        [_ht]: ["POST", "/clusters/{clusterName}/fargate-profiles", 200],
    },
    () => CreateFargateProfileRequest,
    () => CreateFargateProfileResponse,
];
export var CreateNodegroup = [
    9,
    n0,
    _CN,
    {
        [_ht]: ["POST", "/clusters/{clusterName}/node-groups", 200],
    },
    () => CreateNodegroupRequest,
    () => CreateNodegroupResponse,
];
export var CreatePodIdentityAssociation = [
    9,
    n0,
    _CPIA,
    {
        [_ht]: ["POST", "/clusters/{clusterName}/pod-identity-associations", 200],
    },
    () => CreatePodIdentityAssociationRequest,
    () => CreatePodIdentityAssociationResponse,
];
export var DeleteAccessEntry = [
    9,
    n0,
    _DAE,
    {
        [_ht]: ["DELETE", "/clusters/{clusterName}/access-entries/{principalArn}", 200],
    },
    () => DeleteAccessEntryRequest,
    () => DeleteAccessEntryResponse,
];
export var DeleteAddon = [
    9,
    n0,
    _DA,
    {
        [_ht]: ["DELETE", "/clusters/{clusterName}/addons/{addonName}", 200],
    },
    () => DeleteAddonRequest,
    () => DeleteAddonResponse,
];
export var DeleteCluster = [
    9,
    n0,
    _DC,
    {
        [_ht]: ["DELETE", "/clusters/{name}", 200],
    },
    () => DeleteClusterRequest,
    () => DeleteClusterResponse,
];
export var DeleteEksAnywhereSubscription = [
    9,
    n0,
    _DEAS,
    {
        [_ht]: ["DELETE", "/eks-anywhere-subscriptions/{id}", 200],
    },
    () => DeleteEksAnywhereSubscriptionRequest,
    () => DeleteEksAnywhereSubscriptionResponse,
];
export var DeleteFargateProfile = [
    9,
    n0,
    _DFP,
    {
        [_ht]: ["DELETE", "/clusters/{clusterName}/fargate-profiles/{fargateProfileName}", 200],
    },
    () => DeleteFargateProfileRequest,
    () => DeleteFargateProfileResponse,
];
export var DeleteNodegroup = [
    9,
    n0,
    _DN,
    {
        [_ht]: ["DELETE", "/clusters/{clusterName}/node-groups/{nodegroupName}", 200],
    },
    () => DeleteNodegroupRequest,
    () => DeleteNodegroupResponse,
];
export var DeletePodIdentityAssociation = [
    9,
    n0,
    _DPIA,
    {
        [_ht]: ["DELETE", "/clusters/{clusterName}/pod-identity-associations/{associationId}", 200],
    },
    () => DeletePodIdentityAssociationRequest,
    () => DeletePodIdentityAssociationResponse,
];
export var DeregisterCluster = [
    9,
    n0,
    _DCe,
    {
        [_ht]: ["DELETE", "/cluster-registrations/{name}", 200],
    },
    () => DeregisterClusterRequest,
    () => DeregisterClusterResponse,
];
export var DescribeAccessEntry = [
    9,
    n0,
    _DAEe,
    {
        [_ht]: ["GET", "/clusters/{clusterName}/access-entries/{principalArn}", 200],
    },
    () => DescribeAccessEntryRequest,
    () => DescribeAccessEntryResponse,
];
export var DescribeAddon = [
    9,
    n0,
    _DAe,
    {
        [_ht]: ["GET", "/clusters/{clusterName}/addons/{addonName}", 200],
    },
    () => DescribeAddonRequest,
    () => DescribeAddonResponse,
];
export var DescribeAddonConfiguration = [
    9,
    n0,
    _DAC,
    {
        [_ht]: ["GET", "/addons/configuration-schemas", 200],
    },
    () => DescribeAddonConfigurationRequest,
    () => DescribeAddonConfigurationResponse,
];
export var DescribeAddonVersions = [
    9,
    n0,
    _DAV,
    {
        [_ht]: ["GET", "/addons/supported-versions", 200],
    },
    () => DescribeAddonVersionsRequest,
    () => DescribeAddonVersionsResponse,
];
export var DescribeCluster = [
    9,
    n0,
    _DCes,
    {
        [_ht]: ["GET", "/clusters/{name}", 200],
    },
    () => DescribeClusterRequest,
    () => DescribeClusterResponse,
];
export var DescribeClusterVersions = [
    9,
    n0,
    _DCV,
    {
        [_ht]: ["GET", "/cluster-versions", 200],
    },
    () => DescribeClusterVersionsRequest,
    () => DescribeClusterVersionsResponse,
];
export var DescribeEksAnywhereSubscription = [
    9,
    n0,
    _DEASe,
    {
        [_ht]: ["GET", "/eks-anywhere-subscriptions/{id}", 200],
    },
    () => DescribeEksAnywhereSubscriptionRequest,
    () => DescribeEksAnywhereSubscriptionResponse,
];
export var DescribeFargateProfile = [
    9,
    n0,
    _DFPe,
    {
        [_ht]: ["GET", "/clusters/{clusterName}/fargate-profiles/{fargateProfileName}", 200],
    },
    () => DescribeFargateProfileRequest,
    () => DescribeFargateProfileResponse,
];
export var DescribeIdentityProviderConfig = [
    9,
    n0,
    _DIPC,
    {
        [_ht]: ["POST", "/clusters/{clusterName}/identity-provider-configs/describe", 200],
    },
    () => DescribeIdentityProviderConfigRequest,
    () => DescribeIdentityProviderConfigResponse,
];
export var DescribeInsight = [
    9,
    n0,
    _DI,
    {
        [_ht]: ["GET", "/clusters/{clusterName}/insights/{id}", 200],
    },
    () => DescribeInsightRequest,
    () => DescribeInsightResponse,
];
export var DescribeInsightsRefresh = [
    9,
    n0,
    _DIRes,
    {
        [_ht]: ["GET", "/clusters/{clusterName}/insights-refresh", 200],
    },
    () => DescribeInsightsRefreshRequest,
    () => DescribeInsightsRefreshResponse,
];
export var DescribeNodegroup = [
    9,
    n0,
    _DNe,
    {
        [_ht]: ["GET", "/clusters/{clusterName}/node-groups/{nodegroupName}", 200],
    },
    () => DescribeNodegroupRequest,
    () => DescribeNodegroupResponse,
];
export var DescribePodIdentityAssociation = [
    9,
    n0,
    _DPIAe,
    {
        [_ht]: ["GET", "/clusters/{clusterName}/pod-identity-associations/{associationId}", 200],
    },
    () => DescribePodIdentityAssociationRequest,
    () => DescribePodIdentityAssociationResponse,
];
export var DescribeUpdate = [
    9,
    n0,
    _DU,
    {
        [_ht]: ["GET", "/clusters/{name}/updates/{updateId}", 200],
    },
    () => DescribeUpdateRequest,
    () => DescribeUpdateResponse,
];
export var DisassociateAccessPolicy = [
    9,
    n0,
    _DAP,
    {
        [_ht]: ["DELETE", "/clusters/{clusterName}/access-entries/{principalArn}/access-policies/{policyArn}", 200],
    },
    () => DisassociateAccessPolicyRequest,
    () => DisassociateAccessPolicyResponse,
];
export var DisassociateIdentityProviderConfig = [
    9,
    n0,
    _DIPCi,
    {
        [_ht]: ["POST", "/clusters/{clusterName}/identity-provider-configs/disassociate", 200],
    },
    () => DisassociateIdentityProviderConfigRequest,
    () => DisassociateIdentityProviderConfigResponse,
];
export var ListAccessEntries = [
    9,
    n0,
    _LAE,
    {
        [_ht]: ["GET", "/clusters/{clusterName}/access-entries", 200],
    },
    () => ListAccessEntriesRequest,
    () => ListAccessEntriesResponse,
];
export var ListAccessPolicies = [
    9,
    n0,
    _LAP,
    {
        [_ht]: ["GET", "/access-policies", 200],
    },
    () => ListAccessPoliciesRequest,
    () => ListAccessPoliciesResponse,
];
export var ListAddons = [
    9,
    n0,
    _LA,
    {
        [_ht]: ["GET", "/clusters/{clusterName}/addons", 200],
    },
    () => ListAddonsRequest,
    () => ListAddonsResponse,
];
export var ListAssociatedAccessPolicies = [
    9,
    n0,
    _LAAP,
    {
        [_ht]: ["GET", "/clusters/{clusterName}/access-entries/{principalArn}/access-policies", 200],
    },
    () => ListAssociatedAccessPoliciesRequest,
    () => ListAssociatedAccessPoliciesResponse,
];
export var ListClusters = [
    9,
    n0,
    _LC,
    {
        [_ht]: ["GET", "/clusters", 200],
    },
    () => ListClustersRequest,
    () => ListClustersResponse,
];
export var ListEksAnywhereSubscriptions = [
    9,
    n0,
    _LEAS,
    {
        [_ht]: ["GET", "/eks-anywhere-subscriptions", 200],
    },
    () => ListEksAnywhereSubscriptionsRequest,
    () => ListEksAnywhereSubscriptionsResponse,
];
export var ListFargateProfiles = [
    9,
    n0,
    _LFP,
    {
        [_ht]: ["GET", "/clusters/{clusterName}/fargate-profiles", 200],
    },
    () => ListFargateProfilesRequest,
    () => ListFargateProfilesResponse,
];
export var ListIdentityProviderConfigs = [
    9,
    n0,
    _LIPC,
    {
        [_ht]: ["GET", "/clusters/{clusterName}/identity-provider-configs", 200],
    },
    () => ListIdentityProviderConfigsRequest,
    () => ListIdentityProviderConfigsResponse,
];
export var ListInsights = [
    9,
    n0,
    _LI,
    {
        [_ht]: ["POST", "/clusters/{clusterName}/insights", 200],
    },
    () => ListInsightsRequest,
    () => ListInsightsResponse,
];
export var ListNodegroups = [
    9,
    n0,
    _LN,
    {
        [_ht]: ["GET", "/clusters/{clusterName}/node-groups", 200],
    },
    () => ListNodegroupsRequest,
    () => ListNodegroupsResponse,
];
export var ListPodIdentityAssociations = [
    9,
    n0,
    _LPIA,
    {
        [_ht]: ["GET", "/clusters/{clusterName}/pod-identity-associations", 200],
    },
    () => ListPodIdentityAssociationsRequest,
    () => ListPodIdentityAssociationsResponse,
];
export var ListTagsForResource = [
    9,
    n0,
    _LTFR,
    {
        [_ht]: ["GET", "/tags/{resourceArn}", 200],
    },
    () => ListTagsForResourceRequest,
    () => ListTagsForResourceResponse,
];
export var ListUpdates = [
    9,
    n0,
    _LU,
    {
        [_ht]: ["GET", "/clusters/{name}/updates", 200],
    },
    () => ListUpdatesRequest,
    () => ListUpdatesResponse,
];
export var RegisterCluster = [
    9,
    n0,
    _RC,
    {
        [_ht]: ["POST", "/cluster-registrations", 200],
    },
    () => RegisterClusterRequest,
    () => RegisterClusterResponse,
];
export var StartInsightsRefresh = [
    9,
    n0,
    _SIR,
    {
        [_ht]: ["POST", "/clusters/{clusterName}/insights-refresh", 200],
    },
    () => StartInsightsRefreshRequest,
    () => StartInsightsRefreshResponse,
];
export var TagResource = [
    9,
    n0,
    _TR,
    {
        [_ht]: ["POST", "/tags/{resourceArn}", 200],
    },
    () => TagResourceRequest,
    () => TagResourceResponse,
];
export var UntagResource = [
    9,
    n0,
    _UR,
    {
        [_ht]: ["DELETE", "/tags/{resourceArn}", 200],
    },
    () => UntagResourceRequest,
    () => UntagResourceResponse,
];
export var UpdateAccessEntry = [
    9,
    n0,
    _UAE,
    {
        [_ht]: ["POST", "/clusters/{clusterName}/access-entries/{principalArn}", 200],
    },
    () => UpdateAccessEntryRequest,
    () => UpdateAccessEntryResponse,
];
export var UpdateAddon = [
    9,
    n0,
    _UA,
    {
        [_ht]: ["POST", "/clusters/{clusterName}/addons/{addonName}/update", 200],
    },
    () => UpdateAddonRequest,
    () => UpdateAddonResponse,
];
export var UpdateClusterConfig = [
    9,
    n0,
    _UCC,
    {
        [_ht]: ["POST", "/clusters/{name}/update-config", 200],
    },
    () => UpdateClusterConfigRequest,
    () => UpdateClusterConfigResponse,
];
export var UpdateClusterVersion = [
    9,
    n0,
    _UCV,
    {
        [_ht]: ["POST", "/clusters/{name}/updates", 200],
    },
    () => UpdateClusterVersionRequest,
    () => UpdateClusterVersionResponse,
];
export var UpdateEksAnywhereSubscription = [
    9,
    n0,
    _UEAS,
    {
        [_ht]: ["POST", "/eks-anywhere-subscriptions/{id}", 200],
    },
    () => UpdateEksAnywhereSubscriptionRequest,
    () => UpdateEksAnywhereSubscriptionResponse,
];
export var UpdateNodegroupConfig = [
    9,
    n0,
    _UNC,
    {
        [_ht]: ["POST", "/clusters/{clusterName}/node-groups/{nodegroupName}/update-config", 200],
    },
    () => UpdateNodegroupConfigRequest,
    () => UpdateNodegroupConfigResponse,
];
export var UpdateNodegroupVersion = [
    9,
    n0,
    _UNV,
    {
        [_ht]: ["POST", "/clusters/{clusterName}/node-groups/{nodegroupName}/update-version", 200],
    },
    () => UpdateNodegroupVersionRequest,
    () => UpdateNodegroupVersionResponse,
];
export var UpdatePodIdentityAssociation = [
    9,
    n0,
    _UPIA,
    {
        [_ht]: ["POST", "/clusters/{clusterName}/pod-identity-associations/{associationId}", 200],
    },
    () => UpdatePodIdentityAssociationRequest,
    () => UpdatePodIdentityAssociationResponse,
];
