// @aws-sdk/client-elastic-load-balancing-v2/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');
var utilWaiter = require('@smithy/util-waiter');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "elasticloadbalancing",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class ElasticLoadBalancingV2Client extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultElasticLoadBalancingV2HttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

let ElasticLoadBalancingV2ServiceException$1 = class ElasticLoadBalancingV2ServiceException extends smithyClient.ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, ElasticLoadBalancingV2ServiceException.prototype);
    }
};

let CertificateNotFoundException$1 = class CertificateNotFoundException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "CertificateNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CertificateNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CertificateNotFoundException.prototype);
        this.Message = opts.Message;
    }
};
let ListenerNotFoundException$1 = class ListenerNotFoundException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "ListenerNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ListenerNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ListenerNotFoundException.prototype);
        this.Message = opts.Message;
    }
};
let TooManyCertificatesException$1 = class TooManyCertificatesException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "TooManyCertificatesException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyCertificatesException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyCertificatesException.prototype);
        this.Message = opts.Message;
    }
};
let DuplicateTagKeysException$1 = class DuplicateTagKeysException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "DuplicateTagKeysException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "DuplicateTagKeysException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DuplicateTagKeysException.prototype);
        this.Message = opts.Message;
    }
};
let LoadBalancerNotFoundException$1 = class LoadBalancerNotFoundException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "LoadBalancerNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "LoadBalancerNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LoadBalancerNotFoundException.prototype);
        this.Message = opts.Message;
    }
};
let RuleNotFoundException$1 = class RuleNotFoundException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "RuleNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "RuleNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, RuleNotFoundException.prototype);
        this.Message = opts.Message;
    }
};
let TargetGroupNotFoundException$1 = class TargetGroupNotFoundException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "TargetGroupNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TargetGroupNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TargetGroupNotFoundException.prototype);
        this.Message = opts.Message;
    }
};
let TooManyTagsException$1 = class TooManyTagsException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "TooManyTagsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyTagsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyTagsException.prototype);
        this.Message = opts.Message;
    }
};
let TrustStoreNotFoundException$1 = class TrustStoreNotFoundException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "TrustStoreNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TrustStoreNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TrustStoreNotFoundException.prototype);
        this.Message = opts.Message;
    }
};
let InvalidRevocationContentException$1 = class InvalidRevocationContentException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "InvalidRevocationContentException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidRevocationContentException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidRevocationContentException.prototype);
        this.Message = opts.Message;
    }
};
let RevocationContentNotFoundException$1 = class RevocationContentNotFoundException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "RevocationContentNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "RevocationContentNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, RevocationContentNotFoundException.prototype);
        this.Message = opts.Message;
    }
};
let TooManyTrustStoreRevocationEntriesException$1 = class TooManyTrustStoreRevocationEntriesException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "TooManyTrustStoreRevocationEntriesException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyTrustStoreRevocationEntriesException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyTrustStoreRevocationEntriesException.prototype);
        this.Message = opts.Message;
    }
};
let AllocationIdNotFoundException$1 = class AllocationIdNotFoundException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "AllocationIdNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "AllocationIdNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AllocationIdNotFoundException.prototype);
        this.Message = opts.Message;
    }
};
let ALPNPolicyNotSupportedException$1 = class ALPNPolicyNotSupportedException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "ALPNPolicyNotSupportedException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ALPNPolicyNotSupportedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ALPNPolicyNotSupportedException.prototype);
        this.Message = opts.Message;
    }
};
let AvailabilityZoneNotSupportedException$1 = class AvailabilityZoneNotSupportedException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "AvailabilityZoneNotSupportedException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "AvailabilityZoneNotSupportedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AvailabilityZoneNotSupportedException.prototype);
        this.Message = opts.Message;
    }
};
let CaCertificatesBundleNotFoundException$1 = class CaCertificatesBundleNotFoundException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "CaCertificatesBundleNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CaCertificatesBundleNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CaCertificatesBundleNotFoundException.prototype);
        this.Message = opts.Message;
    }
};
let CapacityDecreaseRequestsLimitExceededException$1 = class CapacityDecreaseRequestsLimitExceededException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "CapacityDecreaseRequestsLimitExceededException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CapacityDecreaseRequestsLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CapacityDecreaseRequestsLimitExceededException.prototype);
        this.Message = opts.Message;
    }
};
let CapacityReservationPendingException$1 = class CapacityReservationPendingException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "CapacityReservationPendingException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CapacityReservationPendingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CapacityReservationPendingException.prototype);
        this.Message = opts.Message;
    }
};
let CapacityUnitsLimitExceededException$1 = class CapacityUnitsLimitExceededException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "CapacityUnitsLimitExceededException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CapacityUnitsLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CapacityUnitsLimitExceededException.prototype);
        this.Message = opts.Message;
    }
};
let DuplicateListenerException$1 = class DuplicateListenerException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "DuplicateListenerException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "DuplicateListenerException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DuplicateListenerException.prototype);
        this.Message = opts.Message;
    }
};
let IncompatibleProtocolsException$1 = class IncompatibleProtocolsException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "IncompatibleProtocolsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "IncompatibleProtocolsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IncompatibleProtocolsException.prototype);
        this.Message = opts.Message;
    }
};
let InvalidConfigurationRequestException$1 = class InvalidConfigurationRequestException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "InvalidConfigurationRequestException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidConfigurationRequestException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidConfigurationRequestException.prototype);
        this.Message = opts.Message;
    }
};
let InvalidLoadBalancerActionException$1 = class InvalidLoadBalancerActionException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "InvalidLoadBalancerActionException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidLoadBalancerActionException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidLoadBalancerActionException.prototype);
        this.Message = opts.Message;
    }
};
let SSLPolicyNotFoundException$1 = class SSLPolicyNotFoundException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "SSLPolicyNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "SSLPolicyNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, SSLPolicyNotFoundException.prototype);
        this.Message = opts.Message;
    }
};
let TargetGroupAssociationLimitException$1 = class TargetGroupAssociationLimitException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "TargetGroupAssociationLimitException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TargetGroupAssociationLimitException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TargetGroupAssociationLimitException.prototype);
        this.Message = opts.Message;
    }
};
let TooManyActionsException$1 = class TooManyActionsException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "TooManyActionsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyActionsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyActionsException.prototype);
        this.Message = opts.Message;
    }
};
let TooManyListenersException$1 = class TooManyListenersException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "TooManyListenersException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyListenersException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyListenersException.prototype);
        this.Message = opts.Message;
    }
};
let TooManyRegistrationsForTargetIdException$1 = class TooManyRegistrationsForTargetIdException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "TooManyRegistrationsForTargetIdException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyRegistrationsForTargetIdException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyRegistrationsForTargetIdException.prototype);
        this.Message = opts.Message;
    }
};
let TooManyTargetsException$1 = class TooManyTargetsException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "TooManyTargetsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyTargetsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyTargetsException.prototype);
        this.Message = opts.Message;
    }
};
let TooManyUniqueTargetGroupsPerLoadBalancerException$1 = class TooManyUniqueTargetGroupsPerLoadBalancerException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "TooManyUniqueTargetGroupsPerLoadBalancerException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyUniqueTargetGroupsPerLoadBalancerException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyUniqueTargetGroupsPerLoadBalancerException.prototype);
        this.Message = opts.Message;
    }
};
let TrustStoreNotReadyException$1 = class TrustStoreNotReadyException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "TrustStoreNotReadyException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TrustStoreNotReadyException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TrustStoreNotReadyException.prototype);
        this.Message = opts.Message;
    }
};
let UnsupportedProtocolException$1 = class UnsupportedProtocolException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "UnsupportedProtocolException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "UnsupportedProtocolException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, UnsupportedProtocolException.prototype);
        this.Message = opts.Message;
    }
};
let DuplicateLoadBalancerNameException$1 = class DuplicateLoadBalancerNameException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "DuplicateLoadBalancerNameException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "DuplicateLoadBalancerNameException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DuplicateLoadBalancerNameException.prototype);
        this.Message = opts.Message;
    }
};
let InvalidSchemeException$1 = class InvalidSchemeException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "InvalidSchemeException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidSchemeException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidSchemeException.prototype);
        this.Message = opts.Message;
    }
};
let InvalidSecurityGroupException$1 = class InvalidSecurityGroupException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "InvalidSecurityGroupException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidSecurityGroupException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidSecurityGroupException.prototype);
        this.Message = opts.Message;
    }
};
let InvalidSubnetException$1 = class InvalidSubnetException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "InvalidSubnetException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidSubnetException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidSubnetException.prototype);
        this.Message = opts.Message;
    }
};
let OperationNotPermittedException$1 = class OperationNotPermittedException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "OperationNotPermittedException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "OperationNotPermittedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, OperationNotPermittedException.prototype);
        this.Message = opts.Message;
    }
};
let ResourceInUseException$1 = class ResourceInUseException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "ResourceInUseException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ResourceInUseException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceInUseException.prototype);
        this.Message = opts.Message;
    }
};
let SubnetNotFoundException$1 = class SubnetNotFoundException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "SubnetNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "SubnetNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, SubnetNotFoundException.prototype);
        this.Message = opts.Message;
    }
};
let TooManyLoadBalancersException$1 = class TooManyLoadBalancersException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "TooManyLoadBalancersException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyLoadBalancersException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyLoadBalancersException.prototype);
        this.Message = opts.Message;
    }
};
let PriorityInUseException$1 = class PriorityInUseException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "PriorityInUseException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "PriorityInUseException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, PriorityInUseException.prototype);
        this.Message = opts.Message;
    }
};
let TooManyRulesException$1 = class TooManyRulesException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "TooManyRulesException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyRulesException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyRulesException.prototype);
        this.Message = opts.Message;
    }
};
let TooManyTargetGroupsException$1 = class TooManyTargetGroupsException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "TooManyTargetGroupsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyTargetGroupsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyTargetGroupsException.prototype);
        this.Message = opts.Message;
    }
};
let DuplicateTargetGroupNameException$1 = class DuplicateTargetGroupNameException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "DuplicateTargetGroupNameException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "DuplicateTargetGroupNameException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DuplicateTargetGroupNameException.prototype);
        this.Message = opts.Message;
    }
};
let DuplicateTrustStoreNameException$1 = class DuplicateTrustStoreNameException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "DuplicateTrustStoreNameException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "DuplicateTrustStoreNameException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DuplicateTrustStoreNameException.prototype);
        this.Message = opts.Message;
    }
};
let InvalidCaCertificatesBundleException$1 = class InvalidCaCertificatesBundleException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "InvalidCaCertificatesBundleException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidCaCertificatesBundleException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidCaCertificatesBundleException.prototype);
        this.Message = opts.Message;
    }
};
let TooManyTrustStoresException$1 = class TooManyTrustStoresException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "TooManyTrustStoresException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyTrustStoresException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyTrustStoresException.prototype);
        this.Message = opts.Message;
    }
};
let DeleteAssociationSameAccountException$1 = class DeleteAssociationSameAccountException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "DeleteAssociationSameAccountException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "DeleteAssociationSameAccountException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeleteAssociationSameAccountException.prototype);
        this.Message = opts.Message;
    }
};
let TrustStoreAssociationNotFoundException$1 = class TrustStoreAssociationNotFoundException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "TrustStoreAssociationNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TrustStoreAssociationNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TrustStoreAssociationNotFoundException.prototype);
        this.Message = opts.Message;
    }
};
let TrustStoreInUseException$1 = class TrustStoreInUseException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "TrustStoreInUseException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TrustStoreInUseException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TrustStoreInUseException.prototype);
        this.Message = opts.Message;
    }
};
let InvalidTargetException$1 = class InvalidTargetException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "InvalidTargetException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidTargetException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidTargetException.prototype);
        this.Message = opts.Message;
    }
};
let HealthUnavailableException$1 = class HealthUnavailableException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "HealthUnavailableException";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "HealthUnavailableException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, HealthUnavailableException.prototype);
        this.Message = opts.Message;
    }
};
let RevocationIdNotFoundException$1 = class RevocationIdNotFoundException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "RevocationIdNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "RevocationIdNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, RevocationIdNotFoundException.prototype);
        this.Message = opts.Message;
    }
};
let ResourceNotFoundException$1 = class ResourceNotFoundException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "ResourceNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.Message = opts.Message;
    }
};
let InsufficientCapacityException$1 = class InsufficientCapacityException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "InsufficientCapacityException";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "InsufficientCapacityException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InsufficientCapacityException.prototype);
        this.Message = opts.Message;
    }
};
let PriorRequestNotCompleteException$1 = class PriorRequestNotCompleteException extends ElasticLoadBalancingV2ServiceException$1 {
    name = "PriorRequestNotCompleteException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "PriorRequestNotCompleteException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, PriorRequestNotCompleteException.prototype);
        this.Message = opts.Message;
    }
};

const _A = "Action";
const _AC = "AdditionalClaims";
const _ACAC = "AuthenticateCognitoActionConfig";
const _ACC = "AuthenticateCognitoConfig";
const _AD = "AnomalyDetection";
const _AE = "AuthorizationEndpoint";
const _AI = "AllocationId";
const _AINFE = "AllocationIdNotFoundException";
const _ALC = "AddListenerCertificates";
const _ALCI = "AddListenerCertificatesInput";
const _ALCO = "AddListenerCertificatesOutput";
const _ALPNPNSE = "ALPNPolicyNotSupportedException";
const _AO = "AdministrativeOverride";
const _AOAC = "AuthenticateOidcActionConfig";
const _AOC = "AuthenticateOidcConfig";
const _AP = "AlpnPolicy";
const _AREP = "AuthenticationRequestExtraParams";
const _AT = "AddTags";
const _ATI = "AddTagsInput";
const _ATO = "AddTagsOutput";
const _ATSCN = "AdvertiseTrustStoreCaNames";
const _ATSR = "AddTrustStoreRevocations";
const _ATSRI = "AddTrustStoreRevocationsInput";
const _ATSRO = "AddTrustStoreRevocationsOutput";
const _AZ = "AvailabilityZone";
const _AZNSE = "AvailabilityZoneNotSupportedException";
const _AZv = "AvailabilityZones";
const _Ac = "Actions";
const _At = "Attributes";
const _C = "Certificates";
const _CA = "CertificateArn";
const _CCBNFE = "CaCertificatesBundleNotFoundException";
const _CCBSB = "CaCertificatesBundleS3Bucket";
const _CCBSK = "CaCertificatesBundleS3Key";
const _CCBSOV = "CaCertificatesBundleS3ObjectVersion";
const _CDRLEE = "CapacityDecreaseRequestsLimitExceededException";
const _CHZI = "CanonicalHostedZoneId";
const _CI = "ClientId";
const _CL = "CertificateList";
const _CLB = "CreateLoadBalancer";
const _CLBI = "CreateLoadBalancerInput";
const _CLBO = "CreateLoadBalancerOutput";
const _CLI = "CreateListenerInput";
const _CLO = "CreateListenerOutput";
const _CLr = "CreateListener";
const _CNFE = "CertificateNotFoundException";
const _COIP = "CustomerOwnedIpv4Pool";
const _CR = "CreateRule";
const _CRI = "CreateRuleInput";
const _CRO = "CreateRuleOutput";
const _CRPE = "CapacityReservationPendingException";
const _CRS = "CapacityReservationStatus";
const _CRSa = "CapacityReservationState";
const _CS = "ClientSecret";
const _CT = "ContentType";
const _CTG = "CreateTargetGroup";
const _CTGI = "CreateTargetGroupInput";
const _CTGO = "CreateTargetGroupOutput";
const _CTS = "CreateTrustStore";
const _CTSI = "CreateTrustStoreInput";
const _CTSO = "CreateTrustStoreOutput";
const _CTr = "CreatedTime";
const _CU = "CapacityUnits";
const _CULEE = "CapacityUnitsLimitExceededException";
const _Ce = "Certificate";
const _Ci = "Cipher";
const _Cip = "Ciphers";
const _Co = "Code";
const _Con = "Conditions";
const _D = "Description";
const _DA = "DefaultActions";
const _DAL = "DescribeAccountLimits";
const _DALI = "DescribeAccountLimitsInput";
const _DALO = "DescribeAccountLimitsOutput";
const _DASAE = "DeleteAssociationSameAccountException";
const _DCR = "DescribeCapacityReservation";
const _DCRI = "DescribeCapacityReservationInput";
const _DCRO = "DescribeCapacityReservationOutput";
const _DL = "DeleteListener";
const _DLA = "DescribeListenerAttributes";
const _DLAI = "DescribeListenerAttributesInput";
const _DLAO = "DescribeListenerAttributesOutput";
const _DLB = "DeleteLoadBalancer";
const _DLBA = "DescribeLoadBalancerAttributes";
const _DLBAI = "DescribeLoadBalancerAttributesInput";
const _DLBAO = "DescribeLoadBalancerAttributesOutput";
const _DLBI = "DeleteLoadBalancerInput";
const _DLBIe = "DescribeLoadBalancersInput";
const _DLBNE = "DuplicateLoadBalancerNameException";
const _DLBO = "DeleteLoadBalancerOutput";
const _DLBOe = "DescribeLoadBalancersOutput";
const _DLBe = "DescribeLoadBalancers";
const _DLC = "DescribeListenerCertificates";
const _DLCI = "DescribeListenerCertificatesInput";
const _DLCO = "DescribeListenerCertificatesOutput";
const _DLE = "DuplicateListenerException";
const _DLI = "DeleteListenerInput";
const _DLIe = "DescribeListenersInput";
const _DLO = "DeleteListenerOutput";
const _DLOe = "DescribeListenersOutput";
const _DLe = "DescribeListeners";
const _DNSN = "DNSName";
const _DR = "DeleteRule";
const _DRI = "DeleteRuleInput";
const _DRIe = "DescribeRulesInput";
const _DRO = "DeleteRuleOutput";
const _DROe = "DescribeRulesOutput";
const _DRR = "DecreaseRequestsRemaining";
const _DRe = "DescribeRules";
const _DS = "DurationSeconds";
const _DSSLP = "DescribeSSLPolicies";
const _DSSLPI = "DescribeSSLPoliciesInput";
const _DSSLPO = "DescribeSSLPoliciesOutput";
const _DSTSA = "DeleteSharedTrustStoreAssociation";
const _DSTSAI = "DeleteSharedTrustStoreAssociationInput";
const _DSTSAO = "DeleteSharedTrustStoreAssociationOutput";
const _DT = "DeregisterTargets";
const _DTG = "DeleteTargetGroup";
const _DTGA = "DescribeTargetGroupAttributes";
const _DTGAI = "DescribeTargetGroupAttributesInput";
const _DTGAO = "DescribeTargetGroupAttributesOutput";
const _DTGI = "DeleteTargetGroupInput";
const _DTGIe = "DescribeTargetGroupsInput";
const _DTGNE = "DuplicateTargetGroupNameException";
const _DTGO = "DeleteTargetGroupOutput";
const _DTGOe = "DescribeTargetGroupsOutput";
const _DTGe = "DescribeTargetGroups";
const _DTH = "DescribeTargetHealth";
const _DTHI = "DescribeTargetHealthInput";
const _DTHO = "DescribeTargetHealthOutput";
const _DTI = "DeregisterTargetsInput";
const _DTIe = "DescribeTagsInput";
const _DTKE = "DuplicateTagKeysException";
const _DTO = "DeregisterTargetsOutput";
const _DTOe = "DescribeTagsOutput";
const _DTS = "DeleteTrustStore";
const _DTSA = "DescribeTrustStoreAssociations";
const _DTSAI = "DescribeTrustStoreAssociationsInput";
const _DTSAO = "DescribeTrustStoreAssociationsOutput";
const _DTSI = "DeleteTrustStoreInput";
const _DTSIe = "DescribeTrustStoresInput";
const _DTSNE = "DuplicateTrustStoreNameException";
const _DTSO = "DeleteTrustStoreOutput";
const _DTSOe = "DescribeTrustStoresOutput";
const _DTSR = "DescribeTrustStoreRevocation";
const _DTSRI = "DescribeTrustStoreRevocationsInput";
const _DTSRO = "DescribeTrustStoreRevocationsOutput";
const _DTSRR = "DescribeTrustStoreRevocationResponse";
const _DTSRe = "DescribeTrustStoreRevocations";
const _DTSe = "DescribeTrustStores";
const _DTe = "DescribeTags";
const _E = "Enabled";
const _ECU = "EffectiveCapacityUnits";
const _EPFISN = "EnablePrefixForIpv6SourceNat";
const _ESGIROPLT = "EnforceSecurityGroupInboundRulesOnPrivateLinkTraffic";
const _F = "Format";
const _FAC = "ForwardActionConfig";
const _FC = "ForwardConfig";
const _FRAC = "FixedResponseActionConfig";
const _FRC = "FixedResponseConfig";
const _Fi = "Field";
const _GC = "GrpcCode";
const _GRP = "GetResourcePolicy";
const _GRPI = "GetResourcePolicyInput";
const _GRPO = "GetResourcePolicyOutput";
const _GTSCCB = "GetTrustStoreCaCertificatesBundle";
const _GTSCCBI = "GetTrustStoreCaCertificatesBundleInput";
const _GTSCCBO = "GetTrustStoreCaCertificatesBundleOutput";
const _GTSRC = "GetTrustStoreRevocationContent";
const _GTSRCI = "GetTrustStoreRevocationContentInput";
const _GTSRCO = "GetTrustStoreRevocationContentOutput";
const _H = "Host";
const _HC = "HttpCode";
const _HCE = "HealthCheckEnabled";
const _HCIS = "HealthCheckIntervalSeconds";
const _HCP = "HealthCheckProtocol";
const _HCPe = "HealthCheckPort";
const _HCPea = "HealthCheckPath";
const _HCTS = "HealthCheckTimeoutSeconds";
const _HHC = "HostHeaderConfig";
const _HHCC = "HostHeaderConditionConfig";
const _HHCCt = "HttpHeaderConditionConfig";
const _HHCt = "HttpHeaderConfig";
const _HHN = "HttpHeaderName";
const _HHRC = "HostHeaderRewriteConfig";
const _HRMC = "HttpRequestMethodConfig";
const _HRMCC = "HttpRequestMethodConditionConfig";
const _HTC = "HealthyThresholdCount";
const _HUE = "HealthUnavailableException";
const _I = "Issuer";
const _IA = "IpAddress";
const _IAT = "IpAddressType";
const _ICCBE = "InvalidCaCertificatesBundleException";
const _ICCE = "IgnoreClientCertificateExpiry";
const _ICE = "InsufficientCapacityException";
const _ICRE = "InvalidConfigurationRequestException";
const _ID = "IsDefault";
const _IIPI = "Ipv4IpamPoolId";
const _ILBAE = "InvalidLoadBalancerActionException";
const _IP = "IpamPools";
const _IPA = "IPv6Address";
const _IPE = "IncompatibleProtocolsException";
const _IRCE = "InvalidRevocationContentException";
const _ISE = "InvalidSchemeException";
const _ISEn = "InvalidSubnetException";
const _ISGE = "InvalidSecurityGroupException";
const _ITE = "InvalidTargetException";
const _Id = "Id";
const _In = "Include";
const _JE = "JwksEndpoint";
const _JVAAC = "JwtValidationActionAdditionalClaim";
const _JVAACw = "JwtValidationActionAdditionalClaims";
const _JVAC = "JwtValidationActionConfig";
const _JVC = "JwtValidationConfig";
const _K = "Key";
const _L = "Listeners";
const _LA = "ListenerArn";
const _LAi = "ListenerArns";
const _LAis = "ListenerAttribute";
const _LAist = "ListenerAttributes";
const _LB = "LoadBalancers";
const _LBA = "LoadBalancerAddresses";
const _LBAo = "LoadBalancerArn";
const _LBAoa = "LoadBalancerArns";
const _LBAoad = "LoadBalancerAddress";
const _LBAoada = "LoadBalancerAttribute";
const _LBAoadal = "LoadBalancerAttributes";
const _LBN = "LoadBalancerName";
const _LBNFE = "LoadBalancerNotFoundException";
const _LBS = "LoadBalancerState";
const _LBT = "LoadBalancerType";
const _LBo = "LoadBalancer";
const _LMT = "LastModifiedTime";
const _LNFE = "ListenerNotFoundException";
const _Li = "Limits";
const _Lim = "Limit";
const _Lis = "Listener";
const _Lo = "Location";
const _M = "Message";
const _MA = "MutualAuthentication";
const _MAA = "MutualAuthenticationAttributes";
const _MB = "MessageBody";
const _MCR = "ModifyCapacityReservation";
const _MCRI = "ModifyCapacityReservationInput";
const _MCRO = "ModifyCapacityReservationOutput";
const _MIE = "MitigationInEffect";
const _MIP = "ModifyIpPools";
const _MIPI = "ModifyIpPoolsInput";
const _MIPO = "ModifyIpPoolsOutput";
const _ML = "ModifyListener";
const _MLA = "ModifyListenerAttributes";
const _MLAI = "ModifyListenerAttributesInput";
const _MLAO = "ModifyListenerAttributesOutput";
const _MLBA = "ModifyLoadBalancerAttributes";
const _MLBAI = "ModifyLoadBalancerAttributesInput";
const _MLBAO = "ModifyLoadBalancerAttributesOutput";
const _MLBC = "MinimumLoadBalancerCapacity";
const _MLI = "ModifyListenerInput";
const _MLO = "ModifyListenerOutput";
const _MR = "ModifyRule";
const _MRI = "ModifyRuleInput";
const _MRO = "ModifyRuleOutput";
const _MTG = "ModifyTargetGroup";
const _MTGA = "ModifyTargetGroupAttributes";
const _MTGAI = "ModifyTargetGroupAttributesInput";
const _MTGAO = "ModifyTargetGroupAttributesOutput";
const _MTGI = "ModifyTargetGroupInput";
const _MTGO = "ModifyTargetGroupOutput";
const _MTS = "ModifyTrustStore";
const _MTSI = "ModifyTrustStoreInput";
const _MTSO = "ModifyTrustStoreOutput";
const _Ma = "Matcher";
const _Mar = "Marker";
const _Max = "Max";
const _Mo = "Mode";
const _N = "Name";
const _NM = "NextMarker";
const _NOCC = "NumberOfCaCertificates";
const _NORE = "NumberOfRevokedEntries";
const _Na = "Names";
const _O = "Order";
const _OI = "OutpostId";
const _ONPE = "OperationNotPermittedException";
const _OUR = "OnUnauthenticatedRequest";
const _P = "Priority";
const _PIPA = "PrivateIPv4Address";
const _PIUE = "PriorityInUseException";
const _PPC = "PathPatternConfig";
const _PPCC = "PathPatternConditionConfig";
const _PRNCE = "PriorRequestNotCompleteException";
const _PS = "PageSize";
const _PV = "ProtocolVersion";
const _Pa = "Path";
const _Po = "Port";
const _Pol = "Policy";
const _Pr = "Protocol";
const _Q = "Query";
const _QSC = "QueryStringConfig";
const _QSCC = "QueryStringConditionConfig";
const _QSI = "QuicServerId";
const _QSKVP = "QueryStringKeyValuePair";
const _QSKVPL = "QueryStringKeyValuePairList";
const _R = "Reason";
const _RA = "ResourceArns";
const _RAC = "RedirectActionConfig";
const _RAe = "ResourceArn";
const _RAu = "RuleArn";
const _RAul = "RuleArns";
const _RC = "RedirectConfig";
const _RCL = "RewriteConfigList";
const _RCLu = "RuleConditionList";
const _RCNFE = "RevocationContentNotFoundException";
const _RCR = "ResetCapacityReservation";
const _RCe = "RevocationContents";
const _RCev = "RevocationContent";
const _RCew = "RewriteConfig";
const _RCu = "RuleCondition";
const _RI = "RevocationId";
const _RINFE = "RevocationIdNotFoundException";
const _RIP = "RemoveIpamPools";
const _RIUE = "ResourceInUseException";
const _RIe = "RevocationIds";
const _RLC = "RemoveListenerCertificates";
const _RLCI = "RemoveListenerCertificatesInput";
const _RLCO = "RemoveListenerCertificatesOutput";
const _RNFE = "ResourceNotFoundException";
const _RNFEu = "RuleNotFoundException";
const _RP = "RulePriorities";
const _RPL = "RulePriorityList";
const _RPP = "RulePriorityPair";
const _RT = "RevocationType";
const _RTI = "RegisterTargetsInput";
const _RTIe = "RemoveTagsInput";
const _RTL = "RuleTransformList";
const _RTO = "RegisterTargetsOutput";
const _RTOe = "RemoveTagsOutput";
const _RTSR = "RemoveTrustStoreRevocations";
const _RTSRI = "RemoveTrustStoreRevocationsInput";
const _RTSRO = "RemoveTrustStoreRevocationsOutput";
const _RTe = "ResetTransforms";
const _RTeg = "RegisterTargets";
const _RTem = "RemoveTags";
const _RTu = "RuleTransform";
const _RV = "RegexValues";
const _Re = "Result";
const _Reg = "Regex";
const _Rep = "Replace";
const _Rew = "Rewrites";
const _Ru = "Rules";
const _Rul = "Rule";
const _S = "State";
const _SB = "S3Bucket";
const _SC = "StatusCode";
const _SCN = "SessionCookieName";
const _SG = "SecurityGroups";
const _SGI = "SecurityGroupIds";
const _SI = "SubnetId";
const _SIAT = "SetIpAddressType";
const _SIATI = "SetIpAddressTypeInput";
const _SIATO = "SetIpAddressTypeOutput";
const _SIC = "SourceIpConfig";
const _SICC = "SourceIpConditionConfig";
const _SK = "S3Key";
const _SLBT = "SupportedLoadBalancerTypes";
const _SM = "SubnetMappings";
const _SMu = "SubnetMapping";
const _SNFE = "SubnetNotFoundException";
const _SNIP = "SourceNatIpv6Prefixes";
const _SNIPo = "SourceNatIpv6Prefix";
const _SOV = "S3ObjectVersion";
const _SP = "SslPolicy";
const _SPs = "SslPolicies";
const _SPsl = "SslProtocols";
const _SRP = "SetRulePriorities";
const _SRPI = "SetRulePrioritiesInput";
const _SRPO = "SetRulePrioritiesOutput";
const _SS = "SetSubnets";
const _SSG = "SetSecurityGroups";
const _SSGI = "SetSecurityGroupsInput";
const _SSGO = "SetSecurityGroupsOutput";
const _SSI = "SetSubnetsInput";
const _SSLPNFE = "SSLPolicyNotFoundException";
const _SSO = "SetSubnetsOutput";
const _ST = "SessionTimeout";
const _Sc = "Scope";
const _Sch = "Scheme";
const _St = "Status";
const _Su = "Subnets";
const _T = "Type";
const _TCP = "TargetControlPort";
const _TD = "TagDescriptions";
const _TDa = "TagDescription";
const _TDar = "TargetDescription";
const _TDarg = "TargetDescriptions";
const _TE = "TokenEndpoint";
const _TG = "TargetGroups";
const _TGA = "TargetGroupArn";
const _TGALE = "TargetGroupAssociationLimitException";
const _TGAa = "TargetGroupArns";
const _TGAar = "TargetGroupAttribute";
const _TGAarg = "TargetGroupAttributes";
const _TGL = "TargetGroupList";
const _TGN = "TargetGroupName";
const _TGNFE = "TargetGroupNotFoundException";
const _TGSC = "TargetGroupStickinessConfig";
const _TGT = "TargetGroupTuple";
const _TGa = "TargetGroup";
const _TH = "TargetHealth";
const _THD = "TargetHealthDescriptions";
const _THDa = "TargetHealthDescription";
const _TK = "TagKeys";
const _TL = "TagList";
const _TMAE = "TooManyActionsException";
const _TMCE = "TooManyCertificatesException";
const _TMLBE = "TooManyLoadBalancersException";
const _TMLE = "TooManyListenersException";
const _TMRE = "TooManyRulesException";
const _TMRFTIE = "TooManyRegistrationsForTargetIdException";
const _TMTE = "TooManyTagsException";
const _TMTEo = "TooManyTargetsException";
const _TMTGE = "TooManyTargetGroupsException";
const _TMTSE = "TooManyTrustStoresException";
const _TMTSREE = "TooManyTrustStoreRevocationEntriesException";
const _TMUTGPLBE = "TooManyUniqueTargetGroupsPerLoadBalancerException";
const _TRE = "TotalRevokedEntries";
const _TS = "TrustStores";
const _TSA = "TrustStoreArn";
const _TSANFE = "TrustStoreAssociationNotFoundException";
const _TSAS = "TrustStoreAssociationStatus";
const _TSAr = "TrustStoreAssociations";
const _TSAru = "TrustStoreArns";
const _TSArus = "TrustStoreAssociation";
const _TSIUE = "TrustStoreInUseException";
const _TSNFE = "TrustStoreNotFoundException";
const _TSNRE = "TrustStoreNotReadyException";
const _TSR = "TrustStoreRevocations";
const _TSRr = "TrustStoreRevocation";
const _TSr = "TrustStore";
const _TT = "TargetType";
const _Ta = "Tags";
const _Tag = "Tag";
const _Tar = "Targets";
const _Targ = "Target";
const _Tr = "Transforms";
const _UECS = "UseExistingClientSecret";
const _UIE = "UserInfoEndpoint";
const _UPA = "UserPoolArn";
const _UPCI = "UserPoolClientId";
const _UPD = "UserPoolDomain";
const _UPE = "UnsupportedProtocolException";
const _URC = "UrlRewriteConfig";
const _UTC = "UnhealthyThresholdCount";
const _V = "Values";
const _VI = "VpcId";
const _Va = "Value";
const _W = "Weight";
const _ZCRS = "ZonalCapacityReservationState";
const _ZCRSo = "ZonalCapacityReservationStates";
const _ZN = "ZoneName";
const _aQE = "awsQueryError";
const _c = "client";
const _e = "error";
const _hE = "httpError";
const _s = "server";
const _sm = "smithy.ts.sdk.synthetic.com.amazonaws.elasticloadbalancingv2";
const n0 = "com.amazonaws.elasticloadbalancingv2";
var Action = [
    3,
    n0,
    _A,
    0,
    [_T, _TGA, _AOC, _ACC, _O, _RC, _FRC, _FC, _JVC],
    [
        0,
        0,
        () => AuthenticateOidcActionConfig,
        () => AuthenticateCognitoActionConfig,
        1,
        () => RedirectActionConfig,
        () => FixedResponseActionConfig,
        () => ForwardActionConfig,
        () => JwtValidationActionConfig,
    ],
];
var AddListenerCertificatesInput = [
    3,
    n0,
    _ALCI,
    0,
    [_LA, _C],
    [0, () => CertificateList],
];
var AddListenerCertificatesOutput = [3, n0, _ALCO, 0, [_C], [() => CertificateList]];
var AddTagsInput = [3, n0, _ATI, 0, [_RA, _Ta], [64 | 0, () => TagList]];
var AddTagsOutput = [3, n0, _ATO, 0, [], []];
var AddTrustStoreRevocationsInput = [
    3,
    n0,
    _ATSRI,
    0,
    [_TSA, _RCe],
    [0, () => RevocationContents],
];
var AddTrustStoreRevocationsOutput = [
    3,
    n0,
    _ATSRO,
    0,
    [_TSR],
    [() => TrustStoreRevocations],
];
var AdministrativeOverride = [3, n0, _AO, 0, [_S, _R, _D], [0, 0, 0]];
var AllocationIdNotFoundException = [
    -3,
    n0,
    _AINFE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`AllocationIdNotFound`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(AllocationIdNotFoundException, AllocationIdNotFoundException$1);
var ALPNPolicyNotSupportedException = [
    -3,
    n0,
    _ALPNPNSE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`ALPNPolicyNotFound`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ALPNPolicyNotSupportedException, ALPNPolicyNotSupportedException$1);
var AnomalyDetection = [3, n0, _AD, 0, [_Re, _MIE], [0, 0]];
var AuthenticateCognitoActionConfig = [
    3,
    n0,
    _ACAC,
    0,
    [_UPA, _UPCI, _UPD, _SCN, _Sc, _ST, _AREP, _OUR],
    [0, 0, 0, 0, 0, 1, 128 | 0, 0],
];
var AuthenticateOidcActionConfig = [
    3,
    n0,
    _AOAC,
    0,
    [_I, _AE, _TE, _UIE, _CI, _CS, _SCN, _Sc, _ST, _AREP, _OUR, _UECS],
    [0, 0, 0, 0, 0, 0, 0, 0, 1, 128 | 0, 0, 2],
];
var AvailabilityZone = [
    3,
    n0,
    _AZ,
    0,
    [_ZN, _SI, _OI, _LBA, _SNIP],
    [0, 0, 0, () => LoadBalancerAddresses, 64 | 0],
];
var AvailabilityZoneNotSupportedException = [
    -3,
    n0,
    _AZNSE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`AvailabilityZoneNotSupported`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(AvailabilityZoneNotSupportedException, AvailabilityZoneNotSupportedException$1);
var CaCertificatesBundleNotFoundException = [
    -3,
    n0,
    _CCBNFE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`CaCertificatesBundleNotFound`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(CaCertificatesBundleNotFoundException, CaCertificatesBundleNotFoundException$1);
var CapacityDecreaseRequestsLimitExceededException = [
    -3,
    n0,
    _CDRLEE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`CapacityDecreaseRequestLimitExceeded`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(CapacityDecreaseRequestsLimitExceededException, CapacityDecreaseRequestsLimitExceededException$1);
var CapacityReservationPendingException = [
    -3,
    n0,
    _CRPE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`CapacityReservationPending`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(CapacityReservationPendingException, CapacityReservationPendingException$1);
var CapacityReservationStatus = [3, n0, _CRS, 0, [_Co, _R], [0, 0]];
var CapacityUnitsLimitExceededException = [
    -3,
    n0,
    _CULEE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`CapacityUnitsLimitExceeded`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(CapacityUnitsLimitExceededException, CapacityUnitsLimitExceededException$1);
var Certificate = [3, n0, _Ce, 0, [_CA, _ID], [0, 2]];
var CertificateNotFoundException = [
    -3,
    n0,
    _CNFE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`CertificateNotFound`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(CertificateNotFoundException, CertificateNotFoundException$1);
var Cipher = [3, n0, _Ci, 0, [_N, _P], [0, 1]];
var CreateListenerInput = [
    3,
    n0,
    _CLI,
    0,
    [_LBAo, _Pr, _Po, _SP, _C, _DA, _AP, _Ta, _MA],
    [0, 0, 1, 0, () => CertificateList, () => Actions, 64 | 0, () => TagList, () => MutualAuthenticationAttributes],
];
var CreateListenerOutput = [3, n0, _CLO, 0, [_L], [() => Listeners]];
var CreateLoadBalancerInput = [
    3,
    n0,
    _CLBI,
    0,
    [_N, _Su, _SM, _SG, _Sch, _Ta, _T, _IAT, _COIP, _EPFISN, _IP],
    [0, 64 | 0, () => SubnetMappings, 64 | 0, 0, () => TagList, 0, 0, 0, 0, () => IpamPools],
];
var CreateLoadBalancerOutput = [3, n0, _CLBO, 0, [_LB], [() => LoadBalancers]];
var CreateRuleInput = [
    3,
    n0,
    _CRI,
    0,
    [_LA, _Con, _P, _Ac, _Ta, _Tr],
    [0, () => RuleConditionList, 1, () => Actions, () => TagList, () => RuleTransformList],
];
var CreateRuleOutput = [3, n0, _CRO, 0, [_Ru], [() => Rules]];
var CreateTargetGroupInput = [
    3,
    n0,
    _CTGI,
    0,
    [_N, _Pr, _PV, _Po, _VI, _HCP, _HCPe, _HCE, _HCPea, _HCIS, _HCTS, _HTC, _UTC, _Ma, _TT, _Ta, _IAT, _TCP],
    [0, 0, 0, 1, 0, 0, 0, 2, 0, 1, 1, 1, 1, () => Matcher, 0, () => TagList, 0, 1],
];
var CreateTargetGroupOutput = [3, n0, _CTGO, 0, [_TG], [() => TargetGroups]];
var CreateTrustStoreInput = [
    3,
    n0,
    _CTSI,
    0,
    [_N, _CCBSB, _CCBSK, _CCBSOV, _Ta],
    [0, 0, 0, 0, () => TagList],
];
var CreateTrustStoreOutput = [3, n0, _CTSO, 0, [_TS], [() => TrustStores]];
var DeleteAssociationSameAccountException = [
    -3,
    n0,
    _DASAE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`DeleteAssociationSameAccount`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DeleteAssociationSameAccountException, DeleteAssociationSameAccountException$1);
var DeleteListenerInput = [3, n0, _DLI, 0, [_LA], [0]];
var DeleteListenerOutput = [3, n0, _DLO, 0, [], []];
var DeleteLoadBalancerInput = [3, n0, _DLBI, 0, [_LBAo], [0]];
var DeleteLoadBalancerOutput = [3, n0, _DLBO, 0, [], []];
var DeleteRuleInput = [3, n0, _DRI, 0, [_RAu], [0]];
var DeleteRuleOutput = [3, n0, _DRO, 0, [], []];
var DeleteSharedTrustStoreAssociationInput = [3, n0, _DSTSAI, 0, [_TSA, _RAe], [0, 0]];
var DeleteSharedTrustStoreAssociationOutput = [3, n0, _DSTSAO, 0, [], []];
var DeleteTargetGroupInput = [3, n0, _DTGI, 0, [_TGA], [0]];
var DeleteTargetGroupOutput = [3, n0, _DTGO, 0, [], []];
var DeleteTrustStoreInput = [3, n0, _DTSI, 0, [_TSA], [0]];
var DeleteTrustStoreOutput = [3, n0, _DTSO, 0, [], []];
var DeregisterTargetsInput = [
    3,
    n0,
    _DTI,
    0,
    [_TGA, _Tar],
    [0, () => TargetDescriptions],
];
var DeregisterTargetsOutput = [3, n0, _DTO, 0, [], []];
var DescribeAccountLimitsInput = [3, n0, _DALI, 0, [_Mar, _PS], [0, 1]];
var DescribeAccountLimitsOutput = [3, n0, _DALO, 0, [_Li, _NM], [() => Limits, 0]];
var DescribeCapacityReservationInput = [3, n0, _DCRI, 0, [_LBAo], [0]];
var DescribeCapacityReservationOutput = [
    3,
    n0,
    _DCRO,
    0,
    [_LMT, _DRR, _MLBC, _CRSa],
    [4, 1, () => MinimumLoadBalancerCapacity, () => ZonalCapacityReservationStates],
];
var DescribeListenerAttributesInput = [3, n0, _DLAI, 0, [_LA], [0]];
var DescribeListenerAttributesOutput = [
    3,
    n0,
    _DLAO,
    0,
    [_At],
    [() => ListenerAttributes],
];
var DescribeListenerCertificatesInput = [3, n0, _DLCI, 0, [_LA, _Mar, _PS], [0, 0, 1]];
var DescribeListenerCertificatesOutput = [
    3,
    n0,
    _DLCO,
    0,
    [_C, _NM],
    [() => CertificateList, 0],
];
var DescribeListenersInput = [
    3,
    n0,
    _DLIe,
    0,
    [_LBAo, _LAi, _Mar, _PS],
    [0, 64 | 0, 0, 1],
];
var DescribeListenersOutput = [3, n0, _DLOe, 0, [_L, _NM], [() => Listeners, 0]];
var DescribeLoadBalancerAttributesInput = [3, n0, _DLBAI, 0, [_LBAo], [0]];
var DescribeLoadBalancerAttributesOutput = [
    3,
    n0,
    _DLBAO,
    0,
    [_At],
    [() => LoadBalancerAttributes],
];
var DescribeLoadBalancersInput = [
    3,
    n0,
    _DLBIe,
    0,
    [_LBAoa, _Na, _Mar, _PS],
    [64 | 0, 64 | 0, 0, 1],
];
var DescribeLoadBalancersOutput = [
    3,
    n0,
    _DLBOe,
    0,
    [_LB, _NM],
    [() => LoadBalancers, 0],
];
var DescribeRulesInput = [3, n0, _DRIe, 0, [_LA, _RAul, _Mar, _PS], [0, 64 | 0, 0, 1]];
var DescribeRulesOutput = [3, n0, _DROe, 0, [_Ru, _NM], [() => Rules, 0]];
var DescribeSSLPoliciesInput = [
    3,
    n0,
    _DSSLPI,
    0,
    [_Na, _Mar, _PS, _LBT],
    [64 | 0, 0, 1, 0],
];
var DescribeSSLPoliciesOutput = [3, n0, _DSSLPO, 0, [_SPs, _NM], [() => SslPolicies, 0]];
var DescribeTagsInput = [3, n0, _DTIe, 0, [_RA], [64 | 0]];
var DescribeTagsOutput = [3, n0, _DTOe, 0, [_TD], [() => TagDescriptions]];
var DescribeTargetGroupAttributesInput = [3, n0, _DTGAI, 0, [_TGA], [0]];
var DescribeTargetGroupAttributesOutput = [
    3,
    n0,
    _DTGAO,
    0,
    [_At],
    [() => TargetGroupAttributes],
];
var DescribeTargetGroupsInput = [
    3,
    n0,
    _DTGIe,
    0,
    [_LBAo, _TGAa, _Na, _Mar, _PS],
    [0, 64 | 0, 64 | 0, 0, 1],
];
var DescribeTargetGroupsOutput = [3, n0, _DTGOe, 0, [_TG, _NM], [() => TargetGroups, 0]];
var DescribeTargetHealthInput = [
    3,
    n0,
    _DTHI,
    0,
    [_TGA, _Tar, _In],
    [0, () => TargetDescriptions, 64 | 0],
];
var DescribeTargetHealthOutput = [
    3,
    n0,
    _DTHO,
    0,
    [_THD],
    [() => TargetHealthDescriptions],
];
var DescribeTrustStoreAssociationsInput = [
    3,
    n0,
    _DTSAI,
    0,
    [_TSA, _Mar, _PS],
    [0, 0, 1],
];
var DescribeTrustStoreAssociationsOutput = [
    3,
    n0,
    _DTSAO,
    0,
    [_TSAr, _NM],
    [() => TrustStoreAssociations, 0],
];
var DescribeTrustStoreRevocation = [
    3,
    n0,
    _DTSR,
    0,
    [_TSA, _RI, _RT, _NORE],
    [0, 1, 0, 1],
];
var DescribeTrustStoreRevocationsInput = [
    3,
    n0,
    _DTSRI,
    0,
    [_TSA, _RIe, _Mar, _PS],
    [0, 64 | 1, 0, 1],
];
var DescribeTrustStoreRevocationsOutput = [
    3,
    n0,
    _DTSRO,
    0,
    [_TSR, _NM],
    [() => DescribeTrustStoreRevocationResponse, 0],
];
var DescribeTrustStoresInput = [
    3,
    n0,
    _DTSIe,
    0,
    [_TSAru, _Na, _Mar, _PS],
    [64 | 0, 64 | 0, 0, 1],
];
var DescribeTrustStoresOutput = [3, n0, _DTSOe, 0, [_TS, _NM], [() => TrustStores, 0]];
var DuplicateListenerException = [
    -3,
    n0,
    _DLE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`DuplicateListener`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DuplicateListenerException, DuplicateListenerException$1);
var DuplicateLoadBalancerNameException = [
    -3,
    n0,
    _DLBNE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`DuplicateLoadBalancerName`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DuplicateLoadBalancerNameException, DuplicateLoadBalancerNameException$1);
var DuplicateTagKeysException = [
    -3,
    n0,
    _DTKE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`DuplicateTagKeys`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DuplicateTagKeysException, DuplicateTagKeysException$1);
var DuplicateTargetGroupNameException = [
    -3,
    n0,
    _DTGNE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`DuplicateTargetGroupName`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DuplicateTargetGroupNameException, DuplicateTargetGroupNameException$1);
var DuplicateTrustStoreNameException = [
    -3,
    n0,
    _DTSNE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`DuplicateTrustStoreName`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(DuplicateTrustStoreNameException, DuplicateTrustStoreNameException$1);
var FixedResponseActionConfig = [3, n0, _FRAC, 0, [_MB, _SC, _CT], [0, 0, 0]];
var ForwardActionConfig = [
    3,
    n0,
    _FAC,
    0,
    [_TG, _TGSC],
    [() => TargetGroupList, () => TargetGroupStickinessConfig],
];
var GetResourcePolicyInput = [3, n0, _GRPI, 0, [_RAe], [0]];
var GetResourcePolicyOutput = [3, n0, _GRPO, 0, [_Pol], [0]];
var GetTrustStoreCaCertificatesBundleInput = [3, n0, _GTSCCBI, 0, [_TSA], [0]];
var GetTrustStoreCaCertificatesBundleOutput = [3, n0, _GTSCCBO, 0, [_Lo], [0]];
var GetTrustStoreRevocationContentInput = [3, n0, _GTSRCI, 0, [_TSA, _RI], [0, 1]];
var GetTrustStoreRevocationContentOutput = [3, n0, _GTSRCO, 0, [_Lo], [0]];
var HealthUnavailableException = [
    -3,
    n0,
    _HUE,
    {
        [_e]: _s,
        [_hE]: 500,
        [_aQE]: [`HealthUnavailable`, 500],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(HealthUnavailableException, HealthUnavailableException$1);
var HostHeaderConditionConfig = [3, n0, _HHCC, 0, [_V, _RV], [64 | 0, 64 | 0]];
var HostHeaderRewriteConfig = [3, n0, _HHRC, 0, [_Rew], [() => RewriteConfigList]];
var HttpHeaderConditionConfig = [3, n0, _HHCCt, 0, [_HHN, _V, _RV], [0, 64 | 0, 64 | 0]];
var HttpRequestMethodConditionConfig = [3, n0, _HRMCC, 0, [_V], [64 | 0]];
var IncompatibleProtocolsException = [
    -3,
    n0,
    _IPE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`IncompatibleProtocols`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(IncompatibleProtocolsException, IncompatibleProtocolsException$1);
var InsufficientCapacityException = [
    -3,
    n0,
    _ICE,
    {
        [_e]: _s,
        [_hE]: 500,
        [_aQE]: [`InsufficientCapacity`, 500],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InsufficientCapacityException, InsufficientCapacityException$1);
var InvalidCaCertificatesBundleException = [
    -3,
    n0,
    _ICCBE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`InvalidCaCertificatesBundle`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidCaCertificatesBundleException, InvalidCaCertificatesBundleException$1);
var InvalidConfigurationRequestException = [
    -3,
    n0,
    _ICRE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`InvalidConfigurationRequest`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidConfigurationRequestException, InvalidConfigurationRequestException$1);
var InvalidLoadBalancerActionException = [
    -3,
    n0,
    _ILBAE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`InvalidLoadBalancerAction`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidLoadBalancerActionException, InvalidLoadBalancerActionException$1);
var InvalidRevocationContentException = [
    -3,
    n0,
    _IRCE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`InvalidRevocationContent`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidRevocationContentException, InvalidRevocationContentException$1);
var InvalidSchemeException = [
    -3,
    n0,
    _ISE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`InvalidScheme`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidSchemeException, InvalidSchemeException$1);
var InvalidSecurityGroupException = [
    -3,
    n0,
    _ISGE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`InvalidSecurityGroup`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidSecurityGroupException, InvalidSecurityGroupException$1);
var InvalidSubnetException = [
    -3,
    n0,
    _ISEn,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`InvalidSubnet`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidSubnetException, InvalidSubnetException$1);
var InvalidTargetException = [
    -3,
    n0,
    _ITE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`InvalidTarget`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InvalidTargetException, InvalidTargetException$1);
var IpamPools = [3, n0, _IP, 0, [_IIPI], [0]];
var JwtValidationActionAdditionalClaim = [3, n0, _JVAAC, 0, [_F, _N, _V], [0, 0, 64 | 0]];
var JwtValidationActionConfig = [
    3,
    n0,
    _JVAC,
    0,
    [_JE, _I, _AC],
    [0, 0, () => JwtValidationActionAdditionalClaims],
];
var Limit = [3, n0, _Lim, 0, [_N, _Max], [0, 0]];
var Listener = [
    3,
    n0,
    _Lis,
    0,
    [_LA, _LBAo, _Po, _Pr, _C, _SP, _DA, _AP, _MA],
    [0, 0, 1, 0, () => CertificateList, 0, () => Actions, 64 | 0, () => MutualAuthenticationAttributes],
];
var ListenerAttribute = [3, n0, _LAis, 0, [_K, _Va], [0, 0]];
var ListenerNotFoundException = [
    -3,
    n0,
    _LNFE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`ListenerNotFound`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ListenerNotFoundException, ListenerNotFoundException$1);
var LoadBalancer = [
    3,
    n0,
    _LBo,
    0,
    [_LBAo, _DNSN, _CHZI, _CTr, _LBN, _Sch, _VI, _S, _T, _AZv, _SG, _IAT, _COIP, _ESGIROPLT, _EPFISN, _IP],
    [0, 0, 0, 4, 0, 0, 0, () => LoadBalancerState, 0, () => AvailabilityZones, 64 | 0, 0, 0, 0, 0, () => IpamPools],
];
var LoadBalancerAddress = [3, n0, _LBAoad, 0, [_IA, _AI, _PIPA, _IPA], [0, 0, 0, 0]];
var LoadBalancerAttribute = [3, n0, _LBAoada, 0, [_K, _Va], [0, 0]];
var LoadBalancerNotFoundException = [
    -3,
    n0,
    _LBNFE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`LoadBalancerNotFound`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(LoadBalancerNotFoundException, LoadBalancerNotFoundException$1);
var LoadBalancerState = [3, n0, _LBS, 0, [_Co, _R], [0, 0]];
var Matcher = [3, n0, _Ma, 0, [_HC, _GC], [0, 0]];
var MinimumLoadBalancerCapacity = [3, n0, _MLBC, 0, [_CU], [1]];
var ModifyCapacityReservationInput = [
    3,
    n0,
    _MCRI,
    0,
    [_LBAo, _MLBC, _RCR],
    [0, () => MinimumLoadBalancerCapacity, 2],
];
var ModifyCapacityReservationOutput = [
    3,
    n0,
    _MCRO,
    0,
    [_LMT, _DRR, _MLBC, _CRSa],
    [4, 1, () => MinimumLoadBalancerCapacity, () => ZonalCapacityReservationStates],
];
var ModifyIpPoolsInput = [
    3,
    n0,
    _MIPI,
    0,
    [_LBAo, _IP, _RIP],
    [0, () => IpamPools, 64 | 0],
];
var ModifyIpPoolsOutput = [3, n0, _MIPO, 0, [_IP], [() => IpamPools]];
var ModifyListenerAttributesInput = [
    3,
    n0,
    _MLAI,
    0,
    [_LA, _At],
    [0, () => ListenerAttributes],
];
var ModifyListenerAttributesOutput = [3, n0, _MLAO, 0, [_At], [() => ListenerAttributes]];
var ModifyListenerInput = [
    3,
    n0,
    _MLI,
    0,
    [_LA, _Po, _Pr, _SP, _C, _DA, _AP, _MA],
    [0, 1, 0, 0, () => CertificateList, () => Actions, 64 | 0, () => MutualAuthenticationAttributes],
];
var ModifyListenerOutput = [3, n0, _MLO, 0, [_L], [() => Listeners]];
var ModifyLoadBalancerAttributesInput = [
    3,
    n0,
    _MLBAI,
    0,
    [_LBAo, _At],
    [0, () => LoadBalancerAttributes],
];
var ModifyLoadBalancerAttributesOutput = [
    3,
    n0,
    _MLBAO,
    0,
    [_At],
    [() => LoadBalancerAttributes],
];
var ModifyRuleInput = [
    3,
    n0,
    _MRI,
    0,
    [_RAu, _Con, _Ac, _Tr, _RTe],
    [0, () => RuleConditionList, () => Actions, () => RuleTransformList, 2],
];
var ModifyRuleOutput = [3, n0, _MRO, 0, [_Ru], [() => Rules]];
var ModifyTargetGroupAttributesInput = [
    3,
    n0,
    _MTGAI,
    0,
    [_TGA, _At],
    [0, () => TargetGroupAttributes],
];
var ModifyTargetGroupAttributesOutput = [
    3,
    n0,
    _MTGAO,
    0,
    [_At],
    [() => TargetGroupAttributes],
];
var ModifyTargetGroupInput = [
    3,
    n0,
    _MTGI,
    0,
    [_TGA, _HCP, _HCPe, _HCPea, _HCE, _HCIS, _HCTS, _HTC, _UTC, _Ma],
    [0, 0, 0, 0, 2, 1, 1, 1, 1, () => Matcher],
];
var ModifyTargetGroupOutput = [3, n0, _MTGO, 0, [_TG], [() => TargetGroups]];
var ModifyTrustStoreInput = [
    3,
    n0,
    _MTSI,
    0,
    [_TSA, _CCBSB, _CCBSK, _CCBSOV],
    [0, 0, 0, 0],
];
var ModifyTrustStoreOutput = [3, n0, _MTSO, 0, [_TS], [() => TrustStores]];
var MutualAuthenticationAttributes = [
    3,
    n0,
    _MAA,
    0,
    [_Mo, _TSA, _ICCE, _TSAS, _ATSCN],
    [0, 0, 2, 0, 0],
];
var OperationNotPermittedException = [
    -3,
    n0,
    _ONPE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`OperationNotPermitted`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(OperationNotPermittedException, OperationNotPermittedException$1);
var PathPatternConditionConfig = [3, n0, _PPCC, 0, [_V, _RV], [64 | 0, 64 | 0]];
var PriorityInUseException = [
    -3,
    n0,
    _PIUE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`PriorityInUse`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(PriorityInUseException, PriorityInUseException$1);
var PriorRequestNotCompleteException = [
    -3,
    n0,
    _PRNCE,
    {
        [_e]: _c,
        [_hE]: 429,
        [_aQE]: [`PriorRequestNotComplete`, 429],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(PriorRequestNotCompleteException, PriorRequestNotCompleteException$1);
var QueryStringConditionConfig = [
    3,
    n0,
    _QSCC,
    0,
    [_V],
    [() => QueryStringKeyValuePairList],
];
var QueryStringKeyValuePair = [3, n0, _QSKVP, 0, [_K, _Va], [0, 0]];
var RedirectActionConfig = [
    3,
    n0,
    _RAC,
    0,
    [_Pr, _Po, _H, _Pa, _Q, _SC],
    [0, 0, 0, 0, 0, 0],
];
var RegisterTargetsInput = [3, n0, _RTI, 0, [_TGA, _Tar], [0, () => TargetDescriptions]];
var RegisterTargetsOutput = [3, n0, _RTO, 0, [], []];
var RemoveListenerCertificatesInput = [
    3,
    n0,
    _RLCI,
    0,
    [_LA, _C],
    [0, () => CertificateList],
];
var RemoveListenerCertificatesOutput = [3, n0, _RLCO, 0, [], []];
var RemoveTagsInput = [3, n0, _RTIe, 0, [_RA, _TK], [64 | 0, 64 | 0]];
var RemoveTagsOutput = [3, n0, _RTOe, 0, [], []];
var RemoveTrustStoreRevocationsInput = [3, n0, _RTSRI, 0, [_TSA, _RIe], [0, 64 | 1]];
var RemoveTrustStoreRevocationsOutput = [3, n0, _RTSRO, 0, [], []];
var ResourceInUseException = [
    -3,
    n0,
    _RIUE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`ResourceInUse`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ResourceInUseException, ResourceInUseException$1);
var ResourceNotFoundException = [
    -3,
    n0,
    _RNFE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`ResourceNotFound`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ResourceNotFoundException, ResourceNotFoundException$1);
var RevocationContent = [3, n0, _RCev, 0, [_SB, _SK, _SOV, _RT], [0, 0, 0, 0]];
var RevocationContentNotFoundException = [
    -3,
    n0,
    _RCNFE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`RevocationContentNotFound`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(RevocationContentNotFoundException, RevocationContentNotFoundException$1);
var RevocationIdNotFoundException = [
    -3,
    n0,
    _RINFE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`RevocationIdNotFound`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(RevocationIdNotFoundException, RevocationIdNotFoundException$1);
var RewriteConfig = [3, n0, _RCew, 0, [_Reg, _Rep], [0, 0]];
var Rule = [
    3,
    n0,
    _Rul,
    0,
    [_RAu, _P, _Con, _Ac, _ID, _Tr],
    [0, 0, () => RuleConditionList, () => Actions, 2, () => RuleTransformList],
];
var RuleCondition = [
    3,
    n0,
    _RCu,
    0,
    [_Fi, _V, _HHC, _PPC, _HHCt, _QSC, _HRMC, _SIC, _RV],
    [
        0,
        64 | 0,
        () => HostHeaderConditionConfig,
        () => PathPatternConditionConfig,
        () => HttpHeaderConditionConfig,
        () => QueryStringConditionConfig,
        () => HttpRequestMethodConditionConfig,
        () => SourceIpConditionConfig,
        64 | 0,
    ],
];
var RuleNotFoundException = [
    -3,
    n0,
    _RNFEu,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`RuleNotFound`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(RuleNotFoundException, RuleNotFoundException$1);
var RulePriorityPair = [3, n0, _RPP, 0, [_RAu, _P], [0, 1]];
var RuleTransform = [
    3,
    n0,
    _RTu,
    0,
    [_T, _HHRC, _URC],
    [0, () => HostHeaderRewriteConfig, () => UrlRewriteConfig],
];
var SetIpAddressTypeInput = [3, n0, _SIATI, 0, [_LBAo, _IAT], [0, 0]];
var SetIpAddressTypeOutput = [3, n0, _SIATO, 0, [_IAT], [0]];
var SetRulePrioritiesInput = [3, n0, _SRPI, 0, [_RP], [() => RulePriorityList]];
var SetRulePrioritiesOutput = [3, n0, _SRPO, 0, [_Ru], [() => Rules]];
var SetSecurityGroupsInput = [3, n0, _SSGI, 0, [_LBAo, _SG, _ESGIROPLT], [0, 64 | 0, 0]];
var SetSecurityGroupsOutput = [3, n0, _SSGO, 0, [_SGI, _ESGIROPLT], [64 | 0, 0]];
var SetSubnetsInput = [
    3,
    n0,
    _SSI,
    0,
    [_LBAo, _Su, _SM, _IAT, _EPFISN],
    [0, 64 | 0, () => SubnetMappings, 0, 0],
];
var SetSubnetsOutput = [
    3,
    n0,
    _SSO,
    0,
    [_AZv, _IAT, _EPFISN],
    [() => AvailabilityZones, 0, 0],
];
var SourceIpConditionConfig = [3, n0, _SICC, 0, [_V], [64 | 0]];
var SslPolicy = [
    3,
    n0,
    _SP,
    0,
    [_SPsl, _Cip, _N, _SLBT],
    [64 | 0, () => Ciphers, 0, 64 | 0],
];
var SSLPolicyNotFoundException = [
    -3,
    n0,
    _SSLPNFE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`SSLPolicyNotFound`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(SSLPolicyNotFoundException, SSLPolicyNotFoundException$1);
var SubnetMapping = [3, n0, _SMu, 0, [_SI, _AI, _PIPA, _IPA, _SNIPo], [0, 0, 0, 0, 0]];
var SubnetNotFoundException = [
    -3,
    n0,
    _SNFE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`SubnetNotFound`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(SubnetNotFoundException, SubnetNotFoundException$1);
var Tag = [3, n0, _Tag, 0, [_K, _Va], [0, 0]];
var TagDescription = [3, n0, _TDa, 0, [_RAe, _Ta], [0, () => TagList]];
var TargetDescription = [3, n0, _TDar, 0, [_Id, _Po, _AZ, _QSI], [0, 1, 0, 0]];
var TargetGroup = [
    3,
    n0,
    _TGa,
    0,
    [_TGA, _TGN, _Pr, _Po, _VI, _HCP, _HCPe, _HCE, _HCIS, _HCTS, _HTC, _UTC, _HCPea, _Ma, _LBAoa, _TT, _PV, _IAT, _TCP],
    [0, 0, 0, 1, 0, 0, 0, 2, 1, 1, 1, 1, 0, () => Matcher, 64 | 0, 0, 0, 0, 1],
];
var TargetGroupAssociationLimitException = [
    -3,
    n0,
    _TGALE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`TargetGroupAssociationLimit`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TargetGroupAssociationLimitException, TargetGroupAssociationLimitException$1);
var TargetGroupAttribute = [3, n0, _TGAar, 0, [_K, _Va], [0, 0]];
var TargetGroupNotFoundException = [
    -3,
    n0,
    _TGNFE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`TargetGroupNotFound`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TargetGroupNotFoundException, TargetGroupNotFoundException$1);
var TargetGroupStickinessConfig = [3, n0, _TGSC, 0, [_E, _DS], [2, 1]];
var TargetGroupTuple = [3, n0, _TGT, 0, [_TGA, _W], [0, 1]];
var TargetHealth = [3, n0, _TH, 0, [_S, _R, _D], [0, 0, 0]];
var TargetHealthDescription = [
    3,
    n0,
    _THDa,
    0,
    [_Targ, _HCPe, _TH, _AD, _AO],
    [() => TargetDescription, 0, () => TargetHealth, () => AnomalyDetection, () => AdministrativeOverride],
];
var TooManyActionsException = [
    -3,
    n0,
    _TMAE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`TooManyActions`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TooManyActionsException, TooManyActionsException$1);
var TooManyCertificatesException = [
    -3,
    n0,
    _TMCE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`TooManyCertificates`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TooManyCertificatesException, TooManyCertificatesException$1);
var TooManyListenersException = [
    -3,
    n0,
    _TMLE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`TooManyListeners`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TooManyListenersException, TooManyListenersException$1);
var TooManyLoadBalancersException = [
    -3,
    n0,
    _TMLBE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`TooManyLoadBalancers`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TooManyLoadBalancersException, TooManyLoadBalancersException$1);
var TooManyRegistrationsForTargetIdException = [
    -3,
    n0,
    _TMRFTIE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`TooManyRegistrationsForTargetId`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TooManyRegistrationsForTargetIdException, TooManyRegistrationsForTargetIdException$1);
var TooManyRulesException = [
    -3,
    n0,
    _TMRE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`TooManyRules`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TooManyRulesException, TooManyRulesException$1);
var TooManyTagsException = [
    -3,
    n0,
    _TMTE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`TooManyTags`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TooManyTagsException, TooManyTagsException$1);
var TooManyTargetGroupsException = [
    -3,
    n0,
    _TMTGE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`TooManyTargetGroups`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TooManyTargetGroupsException, TooManyTargetGroupsException$1);
var TooManyTargetsException = [
    -3,
    n0,
    _TMTEo,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`TooManyTargets`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TooManyTargetsException, TooManyTargetsException$1);
var TooManyTrustStoreRevocationEntriesException = [
    -3,
    n0,
    _TMTSREE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`TooManyTrustStoreRevocationEntries`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TooManyTrustStoreRevocationEntriesException, TooManyTrustStoreRevocationEntriesException$1);
var TooManyTrustStoresException = [
    -3,
    n0,
    _TMTSE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`TooManyTrustStores`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TooManyTrustStoresException, TooManyTrustStoresException$1);
var TooManyUniqueTargetGroupsPerLoadBalancerException = [
    -3,
    n0,
    _TMUTGPLBE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`TooManyUniqueTargetGroupsPerLoadBalancer`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TooManyUniqueTargetGroupsPerLoadBalancerException, TooManyUniqueTargetGroupsPerLoadBalancerException$1);
var TrustStore = [3, n0, _TSr, 0, [_N, _TSA, _St, _NOCC, _TRE], [0, 0, 0, 1, 1]];
var TrustStoreAssociation = [3, n0, _TSArus, 0, [_RAe], [0]];
var TrustStoreAssociationNotFoundException = [
    -3,
    n0,
    _TSANFE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`AssociationNotFound`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TrustStoreAssociationNotFoundException, TrustStoreAssociationNotFoundException$1);
var TrustStoreInUseException = [
    -3,
    n0,
    _TSIUE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`TrustStoreInUse`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TrustStoreInUseException, TrustStoreInUseException$1);
var TrustStoreNotFoundException = [
    -3,
    n0,
    _TSNFE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`TrustStoreNotFound`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TrustStoreNotFoundException, TrustStoreNotFoundException$1);
var TrustStoreNotReadyException = [
    -3,
    n0,
    _TSNRE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`TrustStoreNotReady`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(TrustStoreNotReadyException, TrustStoreNotReadyException$1);
var TrustStoreRevocation = [3, n0, _TSRr, 0, [_TSA, _RI, _RT, _NORE], [0, 1, 0, 1]];
var UnsupportedProtocolException = [
    -3,
    n0,
    _UPE,
    {
        [_e]: _c,
        [_hE]: 400,
        [_aQE]: [`UnsupportedProtocol`, 400],
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(UnsupportedProtocolException, UnsupportedProtocolException$1);
var UrlRewriteConfig = [3, n0, _URC, 0, [_Rew], [() => RewriteConfigList]];
var ZonalCapacityReservationState = [
    3,
    n0,
    _ZCRS,
    0,
    [_S, _AZ, _ECU],
    [() => CapacityReservationStatus, 0, 1],
];
var ElasticLoadBalancingV2ServiceException = [
    -3,
    _sm,
    "ElasticLoadBalancingV2ServiceException",
    0,
    [],
    [],
];
schema.TypeRegistry.for(_sm).registerError(ElasticLoadBalancingV2ServiceException, ElasticLoadBalancingV2ServiceException$1);
var Actions = [1, n0, _Ac, 0, () => Action];
var AvailabilityZones = [1, n0, _AZv, 0, () => AvailabilityZone];
var CertificateList = [1, n0, _CL, 0, () => Certificate];
var Ciphers = [1, n0, _Cip, 0, () => Cipher];
var DescribeTrustStoreRevocationResponse = [
    1,
    n0,
    _DTSRR,
    0,
    () => DescribeTrustStoreRevocation,
];
var JwtValidationActionAdditionalClaims = [
    1,
    n0,
    _JVAACw,
    0,
    () => JwtValidationActionAdditionalClaim,
];
var Limits = [1, n0, _Li, 0, () => Limit];
var ListenerAttributes = [1, n0, _LAist, 0, () => ListenerAttribute];
var Listeners = [1, n0, _L, 0, () => Listener];
var LoadBalancerAddresses = [1, n0, _LBA, 0, () => LoadBalancerAddress];
var LoadBalancerAttributes = [1, n0, _LBAoadal, 0, () => LoadBalancerAttribute];
var LoadBalancers = [1, n0, _LB, 0, () => LoadBalancer];
var QueryStringKeyValuePairList = [1, n0, _QSKVPL, 0, () => QueryStringKeyValuePair];
var RevocationContents = [1, n0, _RCe, 0, () => RevocationContent];
var RewriteConfigList = [1, n0, _RCL, 0, () => RewriteConfig];
var RuleConditionList = [1, n0, _RCLu, 0, () => RuleCondition];
var RulePriorityList = [1, n0, _RPL, 0, () => RulePriorityPair];
var Rules = [1, n0, _Ru, 0, () => Rule];
var RuleTransformList = [1, n0, _RTL, 0, () => RuleTransform];
var SslPolicies = [1, n0, _SPs, 0, () => SslPolicy];
var SubnetMappings = [1, n0, _SM, 0, () => SubnetMapping];
var TagDescriptions = [1, n0, _TD, 0, () => TagDescription];
var TagList = [1, n0, _TL, 0, () => Tag];
var TargetDescriptions = [1, n0, _TDarg, 0, () => TargetDescription];
var TargetGroupAttributes = [1, n0, _TGAarg, 0, () => TargetGroupAttribute];
var TargetGroupList = [1, n0, _TGL, 0, () => TargetGroupTuple];
var TargetGroups = [1, n0, _TG, 0, () => TargetGroup];
var TargetHealthDescriptions = [1, n0, _THD, 0, () => TargetHealthDescription];
var TrustStoreAssociations = [1, n0, _TSAr, 0, () => TrustStoreAssociation];
var TrustStoreRevocations = [1, n0, _TSR, 0, () => TrustStoreRevocation];
var TrustStores = [1, n0, _TS, 0, () => TrustStore];
var ZonalCapacityReservationStates = [1, n0, _ZCRSo, 0, () => ZonalCapacityReservationState];
var AddListenerCertificates = [
    9,
    n0,
    _ALC,
    0,
    () => AddListenerCertificatesInput,
    () => AddListenerCertificatesOutput,
];
var AddTags = [9, n0, _AT, 0, () => AddTagsInput, () => AddTagsOutput];
var AddTrustStoreRevocations = [
    9,
    n0,
    _ATSR,
    0,
    () => AddTrustStoreRevocationsInput,
    () => AddTrustStoreRevocationsOutput,
];
var CreateListener = [
    9,
    n0,
    _CLr,
    0,
    () => CreateListenerInput,
    () => CreateListenerOutput,
];
var CreateLoadBalancer = [
    9,
    n0,
    _CLB,
    0,
    () => CreateLoadBalancerInput,
    () => CreateLoadBalancerOutput,
];
var CreateRule = [9, n0, _CR, 0, () => CreateRuleInput, () => CreateRuleOutput];
var CreateTargetGroup = [
    9,
    n0,
    _CTG,
    0,
    () => CreateTargetGroupInput,
    () => CreateTargetGroupOutput,
];
var CreateTrustStore = [
    9,
    n0,
    _CTS,
    0,
    () => CreateTrustStoreInput,
    () => CreateTrustStoreOutput,
];
var DeleteListener = [
    9,
    n0,
    _DL,
    0,
    () => DeleteListenerInput,
    () => DeleteListenerOutput,
];
var DeleteLoadBalancer = [
    9,
    n0,
    _DLB,
    0,
    () => DeleteLoadBalancerInput,
    () => DeleteLoadBalancerOutput,
];
var DeleteRule = [9, n0, _DR, 0, () => DeleteRuleInput, () => DeleteRuleOutput];
var DeleteSharedTrustStoreAssociation = [
    9,
    n0,
    _DSTSA,
    0,
    () => DeleteSharedTrustStoreAssociationInput,
    () => DeleteSharedTrustStoreAssociationOutput,
];
var DeleteTargetGroup = [
    9,
    n0,
    _DTG,
    0,
    () => DeleteTargetGroupInput,
    () => DeleteTargetGroupOutput,
];
var DeleteTrustStore = [
    9,
    n0,
    _DTS,
    0,
    () => DeleteTrustStoreInput,
    () => DeleteTrustStoreOutput,
];
var DeregisterTargets = [
    9,
    n0,
    _DT,
    0,
    () => DeregisterTargetsInput,
    () => DeregisterTargetsOutput,
];
var DescribeAccountLimits = [
    9,
    n0,
    _DAL,
    0,
    () => DescribeAccountLimitsInput,
    () => DescribeAccountLimitsOutput,
];
var DescribeCapacityReservation = [
    9,
    n0,
    _DCR,
    0,
    () => DescribeCapacityReservationInput,
    () => DescribeCapacityReservationOutput,
];
var DescribeListenerAttributes = [
    9,
    n0,
    _DLA,
    0,
    () => DescribeListenerAttributesInput,
    () => DescribeListenerAttributesOutput,
];
var DescribeListenerCertificates = [
    9,
    n0,
    _DLC,
    0,
    () => DescribeListenerCertificatesInput,
    () => DescribeListenerCertificatesOutput,
];
var DescribeListeners = [
    9,
    n0,
    _DLe,
    0,
    () => DescribeListenersInput,
    () => DescribeListenersOutput,
];
var DescribeLoadBalancerAttributes = [
    9,
    n0,
    _DLBA,
    0,
    () => DescribeLoadBalancerAttributesInput,
    () => DescribeLoadBalancerAttributesOutput,
];
var DescribeLoadBalancers = [
    9,
    n0,
    _DLBe,
    0,
    () => DescribeLoadBalancersInput,
    () => DescribeLoadBalancersOutput,
];
var DescribeRules = [9, n0, _DRe, 0, () => DescribeRulesInput, () => DescribeRulesOutput];
var DescribeSSLPolicies = [
    9,
    n0,
    _DSSLP,
    0,
    () => DescribeSSLPoliciesInput,
    () => DescribeSSLPoliciesOutput,
];
var DescribeTags = [9, n0, _DTe, 0, () => DescribeTagsInput, () => DescribeTagsOutput];
var DescribeTargetGroupAttributes = [
    9,
    n0,
    _DTGA,
    0,
    () => DescribeTargetGroupAttributesInput,
    () => DescribeTargetGroupAttributesOutput,
];
var DescribeTargetGroups = [
    9,
    n0,
    _DTGe,
    0,
    () => DescribeTargetGroupsInput,
    () => DescribeTargetGroupsOutput,
];
var DescribeTargetHealth = [
    9,
    n0,
    _DTH,
    0,
    () => DescribeTargetHealthInput,
    () => DescribeTargetHealthOutput,
];
var DescribeTrustStoreAssociations = [
    9,
    n0,
    _DTSA,
    0,
    () => DescribeTrustStoreAssociationsInput,
    () => DescribeTrustStoreAssociationsOutput,
];
var DescribeTrustStoreRevocations = [
    9,
    n0,
    _DTSRe,
    0,
    () => DescribeTrustStoreRevocationsInput,
    () => DescribeTrustStoreRevocationsOutput,
];
var DescribeTrustStores = [
    9,
    n0,
    _DTSe,
    0,
    () => DescribeTrustStoresInput,
    () => DescribeTrustStoresOutput,
];
var GetResourcePolicy = [
    9,
    n0,
    _GRP,
    0,
    () => GetResourcePolicyInput,
    () => GetResourcePolicyOutput,
];
var GetTrustStoreCaCertificatesBundle = [
    9,
    n0,
    _GTSCCB,
    0,
    () => GetTrustStoreCaCertificatesBundleInput,
    () => GetTrustStoreCaCertificatesBundleOutput,
];
var GetTrustStoreRevocationContent = [
    9,
    n0,
    _GTSRC,
    0,
    () => GetTrustStoreRevocationContentInput,
    () => GetTrustStoreRevocationContentOutput,
];
var ModifyCapacityReservation = [
    9,
    n0,
    _MCR,
    0,
    () => ModifyCapacityReservationInput,
    () => ModifyCapacityReservationOutput,
];
var ModifyIpPools = [9, n0, _MIP, 0, () => ModifyIpPoolsInput, () => ModifyIpPoolsOutput];
var ModifyListener = [
    9,
    n0,
    _ML,
    0,
    () => ModifyListenerInput,
    () => ModifyListenerOutput,
];
var ModifyListenerAttributes = [
    9,
    n0,
    _MLA,
    0,
    () => ModifyListenerAttributesInput,
    () => ModifyListenerAttributesOutput,
];
var ModifyLoadBalancerAttributes = [
    9,
    n0,
    _MLBA,
    0,
    () => ModifyLoadBalancerAttributesInput,
    () => ModifyLoadBalancerAttributesOutput,
];
var ModifyRule = [9, n0, _MR, 0, () => ModifyRuleInput, () => ModifyRuleOutput];
var ModifyTargetGroup = [
    9,
    n0,
    _MTG,
    0,
    () => ModifyTargetGroupInput,
    () => ModifyTargetGroupOutput,
];
var ModifyTargetGroupAttributes = [
    9,
    n0,
    _MTGA,
    0,
    () => ModifyTargetGroupAttributesInput,
    () => ModifyTargetGroupAttributesOutput,
];
var ModifyTrustStore = [
    9,
    n0,
    _MTS,
    0,
    () => ModifyTrustStoreInput,
    () => ModifyTrustStoreOutput,
];
var RegisterTargets = [
    9,
    n0,
    _RTeg,
    0,
    () => RegisterTargetsInput,
    () => RegisterTargetsOutput,
];
var RemoveListenerCertificates = [
    9,
    n0,
    _RLC,
    0,
    () => RemoveListenerCertificatesInput,
    () => RemoveListenerCertificatesOutput,
];
var RemoveTags = [9, n0, _RTem, 0, () => RemoveTagsInput, () => RemoveTagsOutput];
var RemoveTrustStoreRevocations = [
    9,
    n0,
    _RTSR,
    0,
    () => RemoveTrustStoreRevocationsInput,
    () => RemoveTrustStoreRevocationsOutput,
];
var SetIpAddressType = [
    9,
    n0,
    _SIAT,
    0,
    () => SetIpAddressTypeInput,
    () => SetIpAddressTypeOutput,
];
var SetRulePriorities = [
    9,
    n0,
    _SRP,
    0,
    () => SetRulePrioritiesInput,
    () => SetRulePrioritiesOutput,
];
var SetSecurityGroups = [
    9,
    n0,
    _SSG,
    0,
    () => SetSecurityGroupsInput,
    () => SetSecurityGroupsOutput,
];
var SetSubnets = [9, n0, _SS, 0, () => SetSubnetsInput, () => SetSubnetsOutput];

class AddListenerCertificatesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "AddListenerCertificates", {})
    .n("ElasticLoadBalancingV2Client", "AddListenerCertificatesCommand")
    .sc(AddListenerCertificates)
    .build() {
}

class AddTagsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "AddTags", {})
    .n("ElasticLoadBalancingV2Client", "AddTagsCommand")
    .sc(AddTags)
    .build() {
}

class AddTrustStoreRevocationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "AddTrustStoreRevocations", {})
    .n("ElasticLoadBalancingV2Client", "AddTrustStoreRevocationsCommand")
    .sc(AddTrustStoreRevocations)
    .build() {
}

class CreateListenerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "CreateListener", {})
    .n("ElasticLoadBalancingV2Client", "CreateListenerCommand")
    .sc(CreateListener)
    .build() {
}

class CreateLoadBalancerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "CreateLoadBalancer", {})
    .n("ElasticLoadBalancingV2Client", "CreateLoadBalancerCommand")
    .sc(CreateLoadBalancer)
    .build() {
}

class CreateRuleCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "CreateRule", {})
    .n("ElasticLoadBalancingV2Client", "CreateRuleCommand")
    .sc(CreateRule)
    .build() {
}

class CreateTargetGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "CreateTargetGroup", {})
    .n("ElasticLoadBalancingV2Client", "CreateTargetGroupCommand")
    .sc(CreateTargetGroup)
    .build() {
}

class CreateTrustStoreCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "CreateTrustStore", {})
    .n("ElasticLoadBalancingV2Client", "CreateTrustStoreCommand")
    .sc(CreateTrustStore)
    .build() {
}

class DeleteListenerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DeleteListener", {})
    .n("ElasticLoadBalancingV2Client", "DeleteListenerCommand")
    .sc(DeleteListener)
    .build() {
}

class DeleteLoadBalancerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DeleteLoadBalancer", {})
    .n("ElasticLoadBalancingV2Client", "DeleteLoadBalancerCommand")
    .sc(DeleteLoadBalancer)
    .build() {
}

class DeleteRuleCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DeleteRule", {})
    .n("ElasticLoadBalancingV2Client", "DeleteRuleCommand")
    .sc(DeleteRule)
    .build() {
}

class DeleteSharedTrustStoreAssociationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DeleteSharedTrustStoreAssociation", {})
    .n("ElasticLoadBalancingV2Client", "DeleteSharedTrustStoreAssociationCommand")
    .sc(DeleteSharedTrustStoreAssociation)
    .build() {
}

class DeleteTargetGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DeleteTargetGroup", {})
    .n("ElasticLoadBalancingV2Client", "DeleteTargetGroupCommand")
    .sc(DeleteTargetGroup)
    .build() {
}

class DeleteTrustStoreCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DeleteTrustStore", {})
    .n("ElasticLoadBalancingV2Client", "DeleteTrustStoreCommand")
    .sc(DeleteTrustStore)
    .build() {
}

class DeregisterTargetsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DeregisterTargets", {})
    .n("ElasticLoadBalancingV2Client", "DeregisterTargetsCommand")
    .sc(DeregisterTargets)
    .build() {
}

class DescribeAccountLimitsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeAccountLimits", {})
    .n("ElasticLoadBalancingV2Client", "DescribeAccountLimitsCommand")
    .sc(DescribeAccountLimits)
    .build() {
}

class DescribeCapacityReservationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeCapacityReservation", {})
    .n("ElasticLoadBalancingV2Client", "DescribeCapacityReservationCommand")
    .sc(DescribeCapacityReservation)
    .build() {
}

class DescribeListenerAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeListenerAttributes", {})
    .n("ElasticLoadBalancingV2Client", "DescribeListenerAttributesCommand")
    .sc(DescribeListenerAttributes)
    .build() {
}

class DescribeListenerCertificatesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeListenerCertificates", {})
    .n("ElasticLoadBalancingV2Client", "DescribeListenerCertificatesCommand")
    .sc(DescribeListenerCertificates)
    .build() {
}

class DescribeListenersCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeListeners", {})
    .n("ElasticLoadBalancingV2Client", "DescribeListenersCommand")
    .sc(DescribeListeners)
    .build() {
}

class DescribeLoadBalancerAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeLoadBalancerAttributes", {})
    .n("ElasticLoadBalancingV2Client", "DescribeLoadBalancerAttributesCommand")
    .sc(DescribeLoadBalancerAttributes)
    .build() {
}

class DescribeLoadBalancersCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeLoadBalancers", {})
    .n("ElasticLoadBalancingV2Client", "DescribeLoadBalancersCommand")
    .sc(DescribeLoadBalancers)
    .build() {
}

class DescribeRulesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeRules", {})
    .n("ElasticLoadBalancingV2Client", "DescribeRulesCommand")
    .sc(DescribeRules)
    .build() {
}

class DescribeSSLPoliciesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeSSLPolicies", {})
    .n("ElasticLoadBalancingV2Client", "DescribeSSLPoliciesCommand")
    .sc(DescribeSSLPolicies)
    .build() {
}

class DescribeTagsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeTags", {})
    .n("ElasticLoadBalancingV2Client", "DescribeTagsCommand")
    .sc(DescribeTags)
    .build() {
}

class DescribeTargetGroupAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeTargetGroupAttributes", {})
    .n("ElasticLoadBalancingV2Client", "DescribeTargetGroupAttributesCommand")
    .sc(DescribeTargetGroupAttributes)
    .build() {
}

class DescribeTargetGroupsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeTargetGroups", {})
    .n("ElasticLoadBalancingV2Client", "DescribeTargetGroupsCommand")
    .sc(DescribeTargetGroups)
    .build() {
}

class DescribeTargetHealthCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeTargetHealth", {})
    .n("ElasticLoadBalancingV2Client", "DescribeTargetHealthCommand")
    .sc(DescribeTargetHealth)
    .build() {
}

class DescribeTrustStoreAssociationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeTrustStoreAssociations", {})
    .n("ElasticLoadBalancingV2Client", "DescribeTrustStoreAssociationsCommand")
    .sc(DescribeTrustStoreAssociations)
    .build() {
}

class DescribeTrustStoreRevocationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeTrustStoreRevocations", {})
    .n("ElasticLoadBalancingV2Client", "DescribeTrustStoreRevocationsCommand")
    .sc(DescribeTrustStoreRevocations)
    .build() {
}

class DescribeTrustStoresCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeTrustStores", {})
    .n("ElasticLoadBalancingV2Client", "DescribeTrustStoresCommand")
    .sc(DescribeTrustStores)
    .build() {
}

class GetResourcePolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "GetResourcePolicy", {})
    .n("ElasticLoadBalancingV2Client", "GetResourcePolicyCommand")
    .sc(GetResourcePolicy)
    .build() {
}

class GetTrustStoreCaCertificatesBundleCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "GetTrustStoreCaCertificatesBundle", {})
    .n("ElasticLoadBalancingV2Client", "GetTrustStoreCaCertificatesBundleCommand")
    .sc(GetTrustStoreCaCertificatesBundle)
    .build() {
}

class GetTrustStoreRevocationContentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "GetTrustStoreRevocationContent", {})
    .n("ElasticLoadBalancingV2Client", "GetTrustStoreRevocationContentCommand")
    .sc(GetTrustStoreRevocationContent)
    .build() {
}

class ModifyCapacityReservationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "ModifyCapacityReservation", {})
    .n("ElasticLoadBalancingV2Client", "ModifyCapacityReservationCommand")
    .sc(ModifyCapacityReservation)
    .build() {
}

class ModifyIpPoolsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "ModifyIpPools", {})
    .n("ElasticLoadBalancingV2Client", "ModifyIpPoolsCommand")
    .sc(ModifyIpPools)
    .build() {
}

class ModifyListenerAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "ModifyListenerAttributes", {})
    .n("ElasticLoadBalancingV2Client", "ModifyListenerAttributesCommand")
    .sc(ModifyListenerAttributes)
    .build() {
}

class ModifyListenerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "ModifyListener", {})
    .n("ElasticLoadBalancingV2Client", "ModifyListenerCommand")
    .sc(ModifyListener)
    .build() {
}

class ModifyLoadBalancerAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "ModifyLoadBalancerAttributes", {})
    .n("ElasticLoadBalancingV2Client", "ModifyLoadBalancerAttributesCommand")
    .sc(ModifyLoadBalancerAttributes)
    .build() {
}

class ModifyRuleCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "ModifyRule", {})
    .n("ElasticLoadBalancingV2Client", "ModifyRuleCommand")
    .sc(ModifyRule)
    .build() {
}

class ModifyTargetGroupAttributesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "ModifyTargetGroupAttributes", {})
    .n("ElasticLoadBalancingV2Client", "ModifyTargetGroupAttributesCommand")
    .sc(ModifyTargetGroupAttributes)
    .build() {
}

class ModifyTargetGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "ModifyTargetGroup", {})
    .n("ElasticLoadBalancingV2Client", "ModifyTargetGroupCommand")
    .sc(ModifyTargetGroup)
    .build() {
}

class ModifyTrustStoreCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "ModifyTrustStore", {})
    .n("ElasticLoadBalancingV2Client", "ModifyTrustStoreCommand")
    .sc(ModifyTrustStore)
    .build() {
}

class RegisterTargetsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "RegisterTargets", {})
    .n("ElasticLoadBalancingV2Client", "RegisterTargetsCommand")
    .sc(RegisterTargets)
    .build() {
}

class RemoveListenerCertificatesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "RemoveListenerCertificates", {})
    .n("ElasticLoadBalancingV2Client", "RemoveListenerCertificatesCommand")
    .sc(RemoveListenerCertificates)
    .build() {
}

class RemoveTagsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "RemoveTags", {})
    .n("ElasticLoadBalancingV2Client", "RemoveTagsCommand")
    .sc(RemoveTags)
    .build() {
}

class RemoveTrustStoreRevocationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "RemoveTrustStoreRevocations", {})
    .n("ElasticLoadBalancingV2Client", "RemoveTrustStoreRevocationsCommand")
    .sc(RemoveTrustStoreRevocations)
    .build() {
}

class SetIpAddressTypeCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "SetIpAddressType", {})
    .n("ElasticLoadBalancingV2Client", "SetIpAddressTypeCommand")
    .sc(SetIpAddressType)
    .build() {
}

class SetRulePrioritiesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "SetRulePriorities", {})
    .n("ElasticLoadBalancingV2Client", "SetRulePrioritiesCommand")
    .sc(SetRulePriorities)
    .build() {
}

class SetSecurityGroupsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "SetSecurityGroups", {})
    .n("ElasticLoadBalancingV2Client", "SetSecurityGroupsCommand")
    .sc(SetSecurityGroups)
    .build() {
}

class SetSubnetsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "SetSubnets", {})
    .n("ElasticLoadBalancingV2Client", "SetSubnetsCommand")
    .sc(SetSubnets)
    .build() {
}

const commands = {
    AddListenerCertificatesCommand,
    AddTagsCommand,
    AddTrustStoreRevocationsCommand,
    CreateListenerCommand,
    CreateLoadBalancerCommand,
    CreateRuleCommand,
    CreateTargetGroupCommand,
    CreateTrustStoreCommand,
    DeleteListenerCommand,
    DeleteLoadBalancerCommand,
    DeleteRuleCommand,
    DeleteSharedTrustStoreAssociationCommand,
    DeleteTargetGroupCommand,
    DeleteTrustStoreCommand,
    DeregisterTargetsCommand,
    DescribeAccountLimitsCommand,
    DescribeCapacityReservationCommand,
    DescribeListenerAttributesCommand,
    DescribeListenerCertificatesCommand,
    DescribeListenersCommand,
    DescribeLoadBalancerAttributesCommand,
    DescribeLoadBalancersCommand,
    DescribeRulesCommand,
    DescribeSSLPoliciesCommand,
    DescribeTagsCommand,
    DescribeTargetGroupAttributesCommand,
    DescribeTargetGroupsCommand,
    DescribeTargetHealthCommand,
    DescribeTrustStoreAssociationsCommand,
    DescribeTrustStoreRevocationsCommand,
    DescribeTrustStoresCommand,
    GetResourcePolicyCommand,
    GetTrustStoreCaCertificatesBundleCommand,
    GetTrustStoreRevocationContentCommand,
    ModifyCapacityReservationCommand,
    ModifyIpPoolsCommand,
    ModifyListenerCommand,
    ModifyListenerAttributesCommand,
    ModifyLoadBalancerAttributesCommand,
    ModifyRuleCommand,
    ModifyTargetGroupCommand,
    ModifyTargetGroupAttributesCommand,
    ModifyTrustStoreCommand,
    RegisterTargetsCommand,
    RemoveListenerCertificatesCommand,
    RemoveTagsCommand,
    RemoveTrustStoreRevocationsCommand,
    SetIpAddressTypeCommand,
    SetRulePrioritiesCommand,
    SetSecurityGroupsCommand,
    SetSubnetsCommand,
};
class ElasticLoadBalancingV2 extends ElasticLoadBalancingV2Client {
}
smithyClient.createAggregatedClient(commands, ElasticLoadBalancingV2);

const paginateDescribeAccountLimits = core.createPaginator(ElasticLoadBalancingV2Client, DescribeAccountLimitsCommand, "Marker", "NextMarker", "");

const paginateDescribeListenerCertificates = core.createPaginator(ElasticLoadBalancingV2Client, DescribeListenerCertificatesCommand, "Marker", "NextMarker", "");

const paginateDescribeListeners = core.createPaginator(ElasticLoadBalancingV2Client, DescribeListenersCommand, "Marker", "NextMarker", "");

const paginateDescribeLoadBalancers = core.createPaginator(ElasticLoadBalancingV2Client, DescribeLoadBalancersCommand, "Marker", "NextMarker", "");

const paginateDescribeRules = core.createPaginator(ElasticLoadBalancingV2Client, DescribeRulesCommand, "Marker", "NextMarker", "");

const paginateDescribeTargetGroups = core.createPaginator(ElasticLoadBalancingV2Client, DescribeTargetGroupsCommand, "Marker", "NextMarker", "");

const paginateDescribeTrustStoreAssociations = core.createPaginator(ElasticLoadBalancingV2Client, DescribeTrustStoreAssociationsCommand, "Marker", "NextMarker", "PageSize");

const paginateDescribeTrustStoreRevocations = core.createPaginator(ElasticLoadBalancingV2Client, DescribeTrustStoreRevocationsCommand, "Marker", "NextMarker", "PageSize");

const paginateDescribeTrustStores = core.createPaginator(ElasticLoadBalancingV2Client, DescribeTrustStoresCommand, "Marker", "NextMarker", "PageSize");

const checkState$4 = async (client, input) => {
    let reason;
    try {
        const result = await client.send(new DescribeLoadBalancersCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                const flat_1 = [].concat(...result.LoadBalancers);
                const projection_3 = flat_1.map((element_2) => {
                    return element_2.State.Code;
                });
                return projection_3;
            };
            let allStringEq_5 = returnComparator().length > 0;
            for (const element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && element_4 == "active";
            }
            if (allStringEq_5) {
                return { state: utilWaiter.WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                const flat_1 = [].concat(...result.LoadBalancers);
                const projection_3 = flat_1.map((element_2) => {
                    return element_2.State.Code;
                });
                return projection_3;
            };
            for (const anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "provisioning") {
                    return { state: utilWaiter.WaiterState.RETRY, reason };
                }
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
        if (exception.name && exception.name == "LoadBalancerNotFoundException") {
            return { state: utilWaiter.WaiterState.RETRY, reason };
        }
    }
    return { state: utilWaiter.WaiterState.RETRY, reason };
};
const waitForLoadBalancerAvailable = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    return utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$4);
};
const waitUntilLoadBalancerAvailable = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    const result = await utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$4);
    return utilWaiter.checkExceptions(result);
};

const checkState$3 = async (client, input) => {
    let reason;
    try {
        const result = await client.send(new DescribeLoadBalancersCommand(input));
        reason = result;
        return { state: utilWaiter.WaiterState.SUCCESS, reason };
    }
    catch (exception) {
        reason = exception;
        if (exception.name && exception.name == "LoadBalancerNotFoundException") {
            return { state: utilWaiter.WaiterState.RETRY, reason };
        }
    }
    return { state: utilWaiter.WaiterState.RETRY, reason };
};
const waitForLoadBalancerExists = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    return utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$3);
};
const waitUntilLoadBalancerExists = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    const result = await utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$3);
    return utilWaiter.checkExceptions(result);
};

const checkState$2 = async (client, input) => {
    let reason;
    try {
        const result = await client.send(new DescribeLoadBalancersCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                const flat_1 = [].concat(...result.LoadBalancers);
                const projection_3 = flat_1.map((element_2) => {
                    return element_2.State.Code;
                });
                return projection_3;
            };
            let allStringEq_5 = returnComparator().length > 0;
            for (const element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && element_4 == "active";
            }
            if (allStringEq_5) {
                return { state: utilWaiter.WaiterState.RETRY, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
        if (exception.name && exception.name == "LoadBalancerNotFoundException") {
            return { state: utilWaiter.WaiterState.SUCCESS, reason };
        }
    }
    return { state: utilWaiter.WaiterState.RETRY, reason };
};
const waitForLoadBalancersDeleted = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    return utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$2);
};
const waitUntilLoadBalancersDeleted = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    const result = await utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$2);
    return utilWaiter.checkExceptions(result);
};

const checkState$1 = async (client, input) => {
    let reason;
    try {
        const result = await client.send(new DescribeTargetHealthCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                const flat_1 = [].concat(...result.TargetHealthDescriptions);
                const projection_3 = flat_1.map((element_2) => {
                    return element_2.TargetHealth.State;
                });
                return projection_3;
            };
            let allStringEq_5 = returnComparator().length > 0;
            for (const element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && element_4 == "unused";
            }
            if (allStringEq_5) {
                return { state: utilWaiter.WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
        if (exception.name && exception.name == "InvalidTargetException") {
            return { state: utilWaiter.WaiterState.SUCCESS, reason };
        }
    }
    return { state: utilWaiter.WaiterState.RETRY, reason };
};
const waitForTargetDeregistered = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    return utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$1);
};
const waitUntilTargetDeregistered = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    const result = await utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState$1);
    return utilWaiter.checkExceptions(result);
};

const checkState = async (client, input) => {
    let reason;
    try {
        const result = await client.send(new DescribeTargetHealthCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                const flat_1 = [].concat(...result.TargetHealthDescriptions);
                const projection_3 = flat_1.map((element_2) => {
                    return element_2.TargetHealth.State;
                });
                return projection_3;
            };
            let allStringEq_5 = returnComparator().length > 0;
            for (const element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && element_4 == "healthy";
            }
            if (allStringEq_5) {
                return { state: utilWaiter.WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
        if (exception.name && exception.name == "InvalidInstance") {
            return { state: utilWaiter.WaiterState.RETRY, reason };
        }
    }
    return { state: utilWaiter.WaiterState.RETRY, reason };
};
const waitForTargetInService = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    return utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState);
};
const waitUntilTargetInService = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    const result = await utilWaiter.createWaiter({ ...serviceDefaults, ...params }, input, checkState);
    return utilWaiter.checkExceptions(result);
};

const AuthenticateCognitoActionConditionalBehaviorEnum = {
    ALLOW: "allow",
    AUTHENTICATE: "authenticate",
    DENY: "deny",
};
const AuthenticateOidcActionConditionalBehaviorEnum = {
    ALLOW: "allow",
    AUTHENTICATE: "authenticate",
    DENY: "deny",
};
const JwtValidationActionAdditionalClaimFormatEnum = {
    SINGLE_STRING: "single-string",
    SPACE_SEPARATED_VALUES: "space-separated-values",
    STRING_ARRAY: "string-array",
};
const RedirectActionStatusCodeEnum = {
    HTTP_301: "HTTP_301",
    HTTP_302: "HTTP_302",
};
const ActionTypeEnum = {
    AUTHENTICATE_COGNITO: "authenticate-cognito",
    AUTHENTICATE_OIDC: "authenticate-oidc",
    FIXED_RESPONSE: "fixed-response",
    FORWARD: "forward",
    JWT_VALIDATION: "jwt-validation",
    REDIRECT: "redirect",
};
const RevocationType = {
    CRL: "CRL",
};
const TargetAdministrativeOverrideReasonEnum = {
    INTERNAL_ERROR: "AdministrativeOverride.Unknown",
    NO_OVERRIDE_ENGAGED: "AdministrativeOverride.NoOverride",
    ZONAL_SHIFT_DELEGATED_TO_DNS: "AdministrativeOverride.ZonalShiftDelegatedToDns",
    ZONAL_SHIFT_ENGAGED: "AdministrativeOverride.ZonalShiftActive",
};
const TargetAdministrativeOverrideStateEnum = {
    NO_OVERRIDE: "no_override",
    UNKNOWN: "unknown",
    ZONAL_SHIFT_ACTIVE: "zonal_shift_active",
    ZONAL_SHIFT_DELEGATED_TO_DNS: "zonal_shift_delegated_to_dns",
};
const AdvertiseTrustStoreCaNamesEnum = {
    off: "off",
    on: "on",
};
const MitigationInEffectEnum = {
    NO: "no",
    YES: "yes",
};
const AnomalyResultEnum = {
    ANOMALOUS: "anomalous",
    NORMAL: "normal",
};
const CapacityReservationStateEnum = {
    FAILED: "failed",
    PENDING: "pending",
    PROVISIONED: "provisioned",
    REBALANCING: "rebalancing",
};
const TrustStoreAssociationStatusEnum = {
    ACTIVE: "active",
    REMOVED: "removed",
};
const ProtocolEnum = {
    GENEVE: "GENEVE",
    HTTP: "HTTP",
    HTTPS: "HTTPS",
    QUIC: "QUIC",
    TCP: "TCP",
    TCP_QUIC: "TCP_QUIC",
    TCP_UDP: "TCP_UDP",
    TLS: "TLS",
    UDP: "UDP",
};
const EnablePrefixForIpv6SourceNatEnum = {
    OFF: "off",
    ON: "on",
};
const IpAddressType = {
    DUALSTACK: "dualstack",
    DUALSTACK_WITHOUT_PUBLIC_IPV4: "dualstack-without-public-ipv4",
    IPV4: "ipv4",
};
const LoadBalancerSchemeEnum = {
    INTERNAL: "internal",
    INTERNET_FACING: "internet-facing",
};
const LoadBalancerTypeEnum = {
    APPLICATION: "application",
    GATEWAY: "gateway",
    NETWORK: "network",
};
const LoadBalancerStateEnum = {
    ACTIVE: "active",
    ACTIVE_IMPAIRED: "active_impaired",
    FAILED: "failed",
    PROVISIONING: "provisioning",
};
const TransformTypeEnum = {
    HOST_HEADER_REWRITE: "host-header-rewrite",
    URL_REWRITE: "url-rewrite",
};
const TargetGroupIpAddressTypeEnum = {
    IPV4: "ipv4",
    IPV6: "ipv6",
};
const TargetTypeEnum = {
    ALB: "alb",
    INSTANCE: "instance",
    IP: "ip",
    LAMBDA: "lambda",
};
const TrustStoreStatus = {
    ACTIVE: "ACTIVE",
    CREATING: "CREATING",
};
const DescribeTargetHealthInputIncludeEnum = {
    ALL: "All",
    ANOMALY: "AnomalyDetection",
};
const TargetHealthReasonEnum = {
    DEREGISTRATION_IN_PROGRESS: "Target.DeregistrationInProgress",
    FAILED_HEALTH_CHECKS: "Target.FailedHealthChecks",
    HEALTH_CHECK_DISABLED: "Target.HealthCheckDisabled",
    INITIAL_HEALTH_CHECKING: "Elb.InitialHealthChecking",
    INTERNAL_ERROR: "Elb.InternalError",
    INVALID_STATE: "Target.InvalidState",
    IP_UNUSABLE: "Target.IpUnusable",
    NOT_IN_USE: "Target.NotInUse",
    NOT_REGISTERED: "Target.NotRegistered",
    REGISTRATION_IN_PROGRESS: "Elb.RegistrationInProgress",
    RESPONSE_CODE_MISMATCH: "Target.ResponseCodeMismatch",
    TIMEOUT: "Target.Timeout",
};
const TargetHealthStateEnum = {
    DRAINING: "draining",
    HEALTHY: "healthy",
    INITIAL: "initial",
    UNAVAILABLE: "unavailable",
    UNHEALTHY: "unhealthy",
    UNHEALTHY_DRAINING: "unhealthy.draining",
    UNUSED: "unused",
};
const RemoveIpamPoolEnum = {
    ipv4: "ipv4",
};
const EnforceSecurityGroupInboundRulesOnPrivateLinkTrafficEnum = {
    off: "off",
    on: "on",
};

Object.defineProperty(exports, "$Command", {
    enumerable: true,
    get: function () { return smithyClient.Command; }
});
Object.defineProperty(exports, "__Client", {
    enumerable: true,
    get: function () { return smithyClient.Client; }
});
exports.ALPNPolicyNotSupportedException = ALPNPolicyNotSupportedException$1;
exports.ActionTypeEnum = ActionTypeEnum;
exports.AddListenerCertificatesCommand = AddListenerCertificatesCommand;
exports.AddTagsCommand = AddTagsCommand;
exports.AddTrustStoreRevocationsCommand = AddTrustStoreRevocationsCommand;
exports.AdvertiseTrustStoreCaNamesEnum = AdvertiseTrustStoreCaNamesEnum;
exports.AllocationIdNotFoundException = AllocationIdNotFoundException$1;
exports.AnomalyResultEnum = AnomalyResultEnum;
exports.AuthenticateCognitoActionConditionalBehaviorEnum = AuthenticateCognitoActionConditionalBehaviorEnum;
exports.AuthenticateOidcActionConditionalBehaviorEnum = AuthenticateOidcActionConditionalBehaviorEnum;
exports.AvailabilityZoneNotSupportedException = AvailabilityZoneNotSupportedException$1;
exports.CaCertificatesBundleNotFoundException = CaCertificatesBundleNotFoundException$1;
exports.CapacityDecreaseRequestsLimitExceededException = CapacityDecreaseRequestsLimitExceededException$1;
exports.CapacityReservationPendingException = CapacityReservationPendingException$1;
exports.CapacityReservationStateEnum = CapacityReservationStateEnum;
exports.CapacityUnitsLimitExceededException = CapacityUnitsLimitExceededException$1;
exports.CertificateNotFoundException = CertificateNotFoundException$1;
exports.CreateListenerCommand = CreateListenerCommand;
exports.CreateLoadBalancerCommand = CreateLoadBalancerCommand;
exports.CreateRuleCommand = CreateRuleCommand;
exports.CreateTargetGroupCommand = CreateTargetGroupCommand;
exports.CreateTrustStoreCommand = CreateTrustStoreCommand;
exports.DeleteAssociationSameAccountException = DeleteAssociationSameAccountException$1;
exports.DeleteListenerCommand = DeleteListenerCommand;
exports.DeleteLoadBalancerCommand = DeleteLoadBalancerCommand;
exports.DeleteRuleCommand = DeleteRuleCommand;
exports.DeleteSharedTrustStoreAssociationCommand = DeleteSharedTrustStoreAssociationCommand;
exports.DeleteTargetGroupCommand = DeleteTargetGroupCommand;
exports.DeleteTrustStoreCommand = DeleteTrustStoreCommand;
exports.DeregisterTargetsCommand = DeregisterTargetsCommand;
exports.DescribeAccountLimitsCommand = DescribeAccountLimitsCommand;
exports.DescribeCapacityReservationCommand = DescribeCapacityReservationCommand;
exports.DescribeListenerAttributesCommand = DescribeListenerAttributesCommand;
exports.DescribeListenerCertificatesCommand = DescribeListenerCertificatesCommand;
exports.DescribeListenersCommand = DescribeListenersCommand;
exports.DescribeLoadBalancerAttributesCommand = DescribeLoadBalancerAttributesCommand;
exports.DescribeLoadBalancersCommand = DescribeLoadBalancersCommand;
exports.DescribeRulesCommand = DescribeRulesCommand;
exports.DescribeSSLPoliciesCommand = DescribeSSLPoliciesCommand;
exports.DescribeTagsCommand = DescribeTagsCommand;
exports.DescribeTargetGroupAttributesCommand = DescribeTargetGroupAttributesCommand;
exports.DescribeTargetGroupsCommand = DescribeTargetGroupsCommand;
exports.DescribeTargetHealthCommand = DescribeTargetHealthCommand;
exports.DescribeTargetHealthInputIncludeEnum = DescribeTargetHealthInputIncludeEnum;
exports.DescribeTrustStoreAssociationsCommand = DescribeTrustStoreAssociationsCommand;
exports.DescribeTrustStoreRevocationsCommand = DescribeTrustStoreRevocationsCommand;
exports.DescribeTrustStoresCommand = DescribeTrustStoresCommand;
exports.DuplicateListenerException = DuplicateListenerException$1;
exports.DuplicateLoadBalancerNameException = DuplicateLoadBalancerNameException$1;
exports.DuplicateTagKeysException = DuplicateTagKeysException$1;
exports.DuplicateTargetGroupNameException = DuplicateTargetGroupNameException$1;
exports.DuplicateTrustStoreNameException = DuplicateTrustStoreNameException$1;
exports.ElasticLoadBalancingV2 = ElasticLoadBalancingV2;
exports.ElasticLoadBalancingV2Client = ElasticLoadBalancingV2Client;
exports.ElasticLoadBalancingV2ServiceException = ElasticLoadBalancingV2ServiceException$1;
exports.EnablePrefixForIpv6SourceNatEnum = EnablePrefixForIpv6SourceNatEnum;
exports.EnforceSecurityGroupInboundRulesOnPrivateLinkTrafficEnum = EnforceSecurityGroupInboundRulesOnPrivateLinkTrafficEnum;
exports.GetResourcePolicyCommand = GetResourcePolicyCommand;
exports.GetTrustStoreCaCertificatesBundleCommand = GetTrustStoreCaCertificatesBundleCommand;
exports.GetTrustStoreRevocationContentCommand = GetTrustStoreRevocationContentCommand;
exports.HealthUnavailableException = HealthUnavailableException$1;
exports.IncompatibleProtocolsException = IncompatibleProtocolsException$1;
exports.InsufficientCapacityException = InsufficientCapacityException$1;
exports.InvalidCaCertificatesBundleException = InvalidCaCertificatesBundleException$1;
exports.InvalidConfigurationRequestException = InvalidConfigurationRequestException$1;
exports.InvalidLoadBalancerActionException = InvalidLoadBalancerActionException$1;
exports.InvalidRevocationContentException = InvalidRevocationContentException$1;
exports.InvalidSchemeException = InvalidSchemeException$1;
exports.InvalidSecurityGroupException = InvalidSecurityGroupException$1;
exports.InvalidSubnetException = InvalidSubnetException$1;
exports.InvalidTargetException = InvalidTargetException$1;
exports.IpAddressType = IpAddressType;
exports.JwtValidationActionAdditionalClaimFormatEnum = JwtValidationActionAdditionalClaimFormatEnum;
exports.ListenerNotFoundException = ListenerNotFoundException$1;
exports.LoadBalancerNotFoundException = LoadBalancerNotFoundException$1;
exports.LoadBalancerSchemeEnum = LoadBalancerSchemeEnum;
exports.LoadBalancerStateEnum = LoadBalancerStateEnum;
exports.LoadBalancerTypeEnum = LoadBalancerTypeEnum;
exports.MitigationInEffectEnum = MitigationInEffectEnum;
exports.ModifyCapacityReservationCommand = ModifyCapacityReservationCommand;
exports.ModifyIpPoolsCommand = ModifyIpPoolsCommand;
exports.ModifyListenerAttributesCommand = ModifyListenerAttributesCommand;
exports.ModifyListenerCommand = ModifyListenerCommand;
exports.ModifyLoadBalancerAttributesCommand = ModifyLoadBalancerAttributesCommand;
exports.ModifyRuleCommand = ModifyRuleCommand;
exports.ModifyTargetGroupAttributesCommand = ModifyTargetGroupAttributesCommand;
exports.ModifyTargetGroupCommand = ModifyTargetGroupCommand;
exports.ModifyTrustStoreCommand = ModifyTrustStoreCommand;
exports.OperationNotPermittedException = OperationNotPermittedException$1;
exports.PriorRequestNotCompleteException = PriorRequestNotCompleteException$1;
exports.PriorityInUseException = PriorityInUseException$1;
exports.ProtocolEnum = ProtocolEnum;
exports.RedirectActionStatusCodeEnum = RedirectActionStatusCodeEnum;
exports.RegisterTargetsCommand = RegisterTargetsCommand;
exports.RemoveIpamPoolEnum = RemoveIpamPoolEnum;
exports.RemoveListenerCertificatesCommand = RemoveListenerCertificatesCommand;
exports.RemoveTagsCommand = RemoveTagsCommand;
exports.RemoveTrustStoreRevocationsCommand = RemoveTrustStoreRevocationsCommand;
exports.ResourceInUseException = ResourceInUseException$1;
exports.ResourceNotFoundException = ResourceNotFoundException$1;
exports.RevocationContentNotFoundException = RevocationContentNotFoundException$1;
exports.RevocationIdNotFoundException = RevocationIdNotFoundException$1;
exports.RevocationType = RevocationType;
exports.RuleNotFoundException = RuleNotFoundException$1;
exports.SSLPolicyNotFoundException = SSLPolicyNotFoundException$1;
exports.SetIpAddressTypeCommand = SetIpAddressTypeCommand;
exports.SetRulePrioritiesCommand = SetRulePrioritiesCommand;
exports.SetSecurityGroupsCommand = SetSecurityGroupsCommand;
exports.SetSubnetsCommand = SetSubnetsCommand;
exports.SubnetNotFoundException = SubnetNotFoundException$1;
exports.TargetAdministrativeOverrideReasonEnum = TargetAdministrativeOverrideReasonEnum;
exports.TargetAdministrativeOverrideStateEnum = TargetAdministrativeOverrideStateEnum;
exports.TargetGroupAssociationLimitException = TargetGroupAssociationLimitException$1;
exports.TargetGroupIpAddressTypeEnum = TargetGroupIpAddressTypeEnum;
exports.TargetGroupNotFoundException = TargetGroupNotFoundException$1;
exports.TargetHealthReasonEnum = TargetHealthReasonEnum;
exports.TargetHealthStateEnum = TargetHealthStateEnum;
exports.TargetTypeEnum = TargetTypeEnum;
exports.TooManyActionsException = TooManyActionsException$1;
exports.TooManyCertificatesException = TooManyCertificatesException$1;
exports.TooManyListenersException = TooManyListenersException$1;
exports.TooManyLoadBalancersException = TooManyLoadBalancersException$1;
exports.TooManyRegistrationsForTargetIdException = TooManyRegistrationsForTargetIdException$1;
exports.TooManyRulesException = TooManyRulesException$1;
exports.TooManyTagsException = TooManyTagsException$1;
exports.TooManyTargetGroupsException = TooManyTargetGroupsException$1;
exports.TooManyTargetsException = TooManyTargetsException$1;
exports.TooManyTrustStoreRevocationEntriesException = TooManyTrustStoreRevocationEntriesException$1;
exports.TooManyTrustStoresException = TooManyTrustStoresException$1;
exports.TooManyUniqueTargetGroupsPerLoadBalancerException = TooManyUniqueTargetGroupsPerLoadBalancerException$1;
exports.TransformTypeEnum = TransformTypeEnum;
exports.TrustStoreAssociationNotFoundException = TrustStoreAssociationNotFoundException$1;
exports.TrustStoreAssociationStatusEnum = TrustStoreAssociationStatusEnum;
exports.TrustStoreInUseException = TrustStoreInUseException$1;
exports.TrustStoreNotFoundException = TrustStoreNotFoundException$1;
exports.TrustStoreNotReadyException = TrustStoreNotReadyException$1;
exports.TrustStoreStatus = TrustStoreStatus;
exports.UnsupportedProtocolException = UnsupportedProtocolException$1;
exports.paginateDescribeAccountLimits = paginateDescribeAccountLimits;
exports.paginateDescribeListenerCertificates = paginateDescribeListenerCertificates;
exports.paginateDescribeListeners = paginateDescribeListeners;
exports.paginateDescribeLoadBalancers = paginateDescribeLoadBalancers;
exports.paginateDescribeRules = paginateDescribeRules;
exports.paginateDescribeTargetGroups = paginateDescribeTargetGroups;
exports.paginateDescribeTrustStoreAssociations = paginateDescribeTrustStoreAssociations;
exports.paginateDescribeTrustStoreRevocations = paginateDescribeTrustStoreRevocations;
exports.paginateDescribeTrustStores = paginateDescribeTrustStores;
exports.waitForLoadBalancerAvailable = waitForLoadBalancerAvailable;
exports.waitForLoadBalancerExists = waitForLoadBalancerExists;
exports.waitForLoadBalancersDeleted = waitForLoadBalancersDeleted;
exports.waitForTargetDeregistered = waitForTargetDeregistered;
exports.waitForTargetInService = waitForTargetInService;
exports.waitUntilLoadBalancerAvailable = waitUntilLoadBalancerAvailable;
exports.waitUntilLoadBalancerExists = waitUntilLoadBalancerExists;
exports.waitUntilLoadBalancersDeleted = waitUntilLoadBalancersDeleted;
exports.waitUntilTargetDeregistered = waitUntilTargetDeregistered;
exports.waitUntilTargetInService = waitUntilTargetInService;
